---
title: "Microbial Response to Rainfall -- Metatranscriptome Analysis"
author: 'Authors: [Karl J. Romanowicz](https://lsa.umich.edu/eeb/people/graduate-students/kjromano.html),
  Byron C. Crump, George W. Kling'
output:
  html_notebook:
    theme: spacelab
    toc: yes
    toc_depth: 5
    toc_float:
      collapsed: yes
      smooth_scroll: yes
  html_document:
    toc: yes
    toc_depth: '5'
    df_print: paged
  pdf_document:
    toc: yes
    toc_depth: '5'
---

__________________________________________________

**R Notebook:** Provides reproducible analysis for metatranscriptome (MT) data in the following manuscript:

**Citation:** Romanowicz, KJ, Crump, BC, Kling, GW. (2021) Rainfall alters permafrost soil redox conditions, but meta-omics show divergent microbial community responses by tundra type in the arctic. Soil Systems 5(1): 17. [https://doi.org/10.3390/soilsystems5010017](https://doi.org/10.3390/soilsystems5010017)

**GitHub Repository:** [https://github.com/kromanowicz/2021-Romanowicz-SoilSystems](https://github.com/kromanowicz/2021-Romanowicz-SoilSystems)

**NCBI BioProject:** [https://www.ncbi.nlm.nih.gov/bioproject/PRJNA666429](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA666429)

**Accepted for Publication:** *Soil Systems* 10 March 2021

# Experiment

This R Notebook provides complete reproducibility of the data analysis in ***"Rainfall alters permafrost soil redox conditions, but meta-omics show divergent microbial community responses by tundra type in the Arctic "*** by Romanowicz, Crump, and Kling. In this experiment, mesocosms containing soil from the active layer of two dominant tundra types were subjected to simulated rainfall to alter redox conditions.  The microbial functional potential (metagenomics) and gene expression (metatranscriptomics) patterns were measured during saturated anoxic redox conditions prior to rainfall and at multiple time points following the simulated rainfall event. Other measurements include soil properties as well as microbial respiration (CO~2~) and methane (CH~4~) production from soil subsamples collected at each sampling time point. The purpose was to determine if rainfall, as a form of soil oxidation, is sufficient to alter the anoxic redox conditions in arctic tundra and enhance the microbial degradation of organic carbon and CH~4~ to CO~2~.

<!-- Display Figure 1 -->
![](R.Images/Concept1.png)
**Conceptual Figure.**  A total of 12 tundra mesocosms (3 replicates x 2 tundra types x 2 sets of response cores) were acclimated initially under anoxic redox conditions to mimic field conditions (T0).  Dissolved oxygen was supplied to soils through the downward flow of oxygenated water during a simulated rainfall event.  Dissolved oxygen will likely change the redox gradient directly following rainfall (T4) as a short-term effect.  Anoxic conditions will likely be re-established after 24 hours (T24) as the pulse of oxygen is consumed through abiotic and biotic soil processes.  Under anoxic redox conditions (T0), microorganisms likely degrade organic carbon through anaerobic and fermentation pathways, producing CH~4~ and reducing Fe(III) to Fe(II).  Rainfall-induced soil oxidation (T4) should stimulate heterotrophic microorganisms that degrade organic carbon and CH~4~ through aerobic metabolic pathways, releasing CO~2~. Soil oxidation should also stimulate aerobic autotrophic iron oxidizing bacteria that oxidize Fe(II) to Fe(III) and convert CO~2~ into microbial biomass.  The long-term response (T24) will likely be a combination of aerobic and anaerobic metabolism as well as a combination of reduction and oxidation iron reactions as dissolved oxygen is consumed.  The predicted redox conditions and predicted redox reactions for coupled Fe(II)/Fe(III) cycling, as well as the microbial-induced release of CO~2~ or CH~4~ at each time point are based on the predicted availability of dissolved oxygen entering tundra soils through simulated rainfall.

<font color="blue">**Soil Sampling for Microbial Gene Expression**</font>

An initial soil sampling event for microbial activity was conducted at the end of the anoxic acclimation period (4-7 days) in all mesocosm replicates, representing sampling time point **T0**.  Mesocosms were then flushed to simulate a rainfall event.  Additional soil sampling events were conducted at **T4** (4-hrs) and **T24** (24-hrs) following the rainfall event to determine the temporal extent of microbial gene expression.  Soil cores (2.54 cm diameter, 30 cm length) were extracted in duplicate from each mesocosm replicate at each sampling time point and homogenized by depth in 10-cm increments.  The 10-20 cm soil increment, composed of organic soil in all mesocosm replicates, was chosen for microbial gene expression analysis and preserved in RNAlater Stabilization Reagent in sterile tubes at 4&deg;C for 18 hours and then stored at -80&deg;C until extraction.

![](R.Images/Buckets.png)

**Field Experiment.** Tundra soil cores were collected from field sites in August 2017 (top left) and placed in buckets to establish the mesocosm experiment (bottom left).  Tussock tundra cores were composed of an organic soil layer overlying a mineral soil layer (top middle) while wet sedge tundra cores were composed entirely of organic soil (bottom middle).  Soil subsampling for microbial activity was taken from the 10-20 cm depth of duplicate soil cores in Tussock (top right) and Wet Sedge (bottom right).

```{r echo=FALSE}
# Set global options for notebook
knitr::opts_knit$set(root.dir = normalizePath("~/Desktop/RainSim2017"))
knitr::opts_chunk$set(fig.width=10, fig.height=8, warning=FALSE, message=FALSE)
```

```{r message=FALSE, results='hide', warning=FALSE}
# Make a vector of required packages
required.packages <- c("ape","cowplot","data.table","devtools","dplyr","DT","edgeR","ggplot2","ggpubr","grid","gridExtra","kableExtra","knitr","pheatmap","png","RColorBrewer","reshape","rstatix","statmod","stringr","tibble","tidyr","tidyverse","vegan","yaml")

# Load required packages
lapply(required.packages, library, character.only = TRUE)
```

# Metatranscriptomes

## 1. RNA Sequencing

Total RNA was extracted from soil samples and treated with RiboZERO rRNA removal (*leaving mRNA as the bulk of the total RNA pool*).  The mRNA pool was sequenced on the Illumina HiSeq 4000 platform (150bp paired-end reads) at the University of Michigan Advanced Genomics Core.

```{r echo=FALSE, results='asis'}
rna.seqcore <-read.table("Table.Data/rna.seqcore.all.table.txt",sep='\t', quote="", fill=TRUE,header = TRUE)
names(rna.seqcore)<- c("ID", "Sample", "Concentration (ng/uL)", "Vol (uL)", "260/280", "260/230", "RIN")
kable(rna.seqcore, caption = "SeqCore RNA sample submission to Illumina HiSeq platform", align = "c") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed")) %>% row_spec(13:18, bold = T, color = "white", background = "#D7261E")
```

**NOTE:** RNA Samples in <font color="red">RED</font> were excluded from sequencing

## 2. RNA Bionformatics

### 2.1. Raw Reads

***Prep and QC the Raw Reads (Hein: Comics -- Omics Prep)***

Raw RNA sequence reads (metatranscriptomics) were processed initially with the same Geomicro Omics container commands as for the raw DNA sequence reads (metagenomics), with several exceptions noted below:
- This prep step assumes the raw data to be available in uncompressed FASTQ format, one file per read direction and each sample's data in its own directory
- The omics prep script will decompress and (if needed) concatenate split fastq files and put them into appropriately names directories

- The `omics qc` script was run for each sample but this command differs from the metagenome workflow by **NOT** dereplicating the RNA reads
- We want to **KEEP** all copies of identical reads to determine the abundance of transcribed functional genes from mRNA

````markdown
`r ''`{Terminal}
cd /RNA_data/work
comics -- omics prep /RNA_data/1083*
comics -- omics qc --no-dereplicate 1083*
````

**RNA.data.prep.qc.pbs** <font color="green">("/RNA_data/work/")</font> 00:15 (hr:min)

- Rename the `.fastq` files within each sample folder in the <font color="green">"RNA_data/work"</font> directory as "**Sample_108379_fwd.fastq**", "**Sample_108379_rev.fastq**", etc.
- Create a <font color="green">"/RNA_data/work/mapping"</font> directory and move over all of the renamed `.fastq` files ("**Sample_108379_fwd.fastq**", "**Sample_108379_rev.fastq**", etc.)

### 2.2. Export Nucleotide Seqs

****Export the nucleotide sequences of the Metagenome Anvi'o gene calls (for mapping MT reads to the MG genes)***

- First, export the nucleotide sequences of the MG Anvi'o gene calls:

````markdown
`r ''`{Terminal}
cd /DNA_data/work/co_assembly
anvi-get-sequences-for-gene-calls -c DNA_data_bbnorm_anvio_contigs.db -o DNA_data_bbnorm_anvio_gene_calls.fna
````
**DNA.data.bbnorm.anvio.export.gene.seqs.pbs** <font color="green">("/RNA_data/work/")</font> 0:16 (hr:min)

- Move the **"DNA_data_bbnorm_anvio_gene_calls.fna"** file to the <font color="green">"/RNA_data/work/mapping"</font> directory  

<font color="blue">**2.2.1**</font> Add the prefix "genecall" to all nucleotide sequences in the **"DNA_data_bbnorm_anvio_gene_calls.fna"** file
````markdown
`r ''`{Terminal}
cd /RNA_data/work/mapping
sed 's/>/>genecall_/g' DNA_data_bbnorm_anvio_gene_calls.fna > DNA_data_bbnorm_anvio_gene_calls_edited.fna
mv DNA_data_bbnorm_anvio_gene_calls_edited.fna DNA_data_bbnorm_anvio_gene_calls.fna
````

<font color="blue">**2.2.2**</font> Use the following script to remove all empty sequences from the **"DNA_data_bbnorm_anvio_gene_calls.fna"** file
````markdown
`r ''`{Terminal}
cd /RNA_data/work/mapping
sed '/^>/ {N; /\n$/d}' DNA_data_bbnorm_anvio_gene_calls.fna > DNA_data_bbnorm_anvio_gene_calls_clean.fna
mv DNA_data_bbnorm_anvio_gene_calls_clean.fna DNA_data_bbnorm_anvio_gene_calls.fna
````

### 2.3. Index MG Genes

***Index the MG Genes and Map Back the MT Reads (Hein: Omics Mapping)***

- Once the sequence headers in the MG genes file were cleaned up, they were indexed as follows:
````markdown
`r ''`{Terminal}
cd /RNA_data/work/mapping
comics -- omics mapping -a DNA_data_bbnorm_anvio_gene_calls.fna --index-only
````
**DNA.data.bbnorm.index.MG.genes.pbs** <font color="green">("/RNA_data/work/")</font> 00:26 (hr:min)

- The output was in a directory called <font color="green">"/bowtie2-index"</font>
- Rename as <font color="green">"/bowtie2-index_bbnorm_MG_genes"</font>


<font color="blue">**2.3.1**</font> Mapping MT QC'd Reads from Each Sample to the Indexed MG Genes

- Individually map the MT QC'd reads from each sample to the indexed MG genes, which will create separate output directories for each sample containing the mapping result files

````markdown
`r ''`{Terminal}
cd /RNA_data/work/mapping
comics -- omics mapping --index-dir /RNA_data/work/mapping/bowtie2-index_bbnorm_MG_genes/ -a DNA_data_bbnorm_anvio_gene_calls.fna -f Sample_108379_fwd.good.fastq -r Sample_108379_rev.good.fastq -o 108379_MT_mapped_reads
comics -- omics mapping --index-dir /RNA_data/work/mapping/bowtie2-index_bbnorm_MG_genes/ -a DNA_data_bbnorm_anvio_gene_calls.fna -f Sample_108381_fwd.good.fastq -r Sample_108381_rev.good.fastq -o 108381_MT_mapped_reads
comics -- omics mapping --index-dir /RNA_data/work/mapping/bowtie2-index_bbnorm_MG_genes/ -a DNA_data_bbnorm_anvio_gene_calls.fna -f Sample_108382_fwd.good.fastq -r Sample_108382_rev.good.fastq -o 108382_MT_mapped_reads
comics -- omics mapping --index-dir /RNA_data/work/mapping/bowtie2-index_bbnorm_MG_genes/ -a DNA_data_bbnorm_anvio_gene_calls.fna -f Sample_108383_fwd.good.fastq -r Sample_108383_rev.good.fastq -o 108383_MT_mapped_reads
comics -- omics mapping --index-dir /RNA_data/work/mapping/bowtie2-index_bbnorm_MG_genes/ -a DNA_data_bbnorm_anvio_gene_calls.fna -f Sample_108385_fwd.good.fastq -r Sample_108385_rev.good.fastq -o 108385_MT_mapped_reads
comics -- omics mapping --index-dir /RNA_data/work/mapping/bowtie2-index_bbnorm_MG_genes/ -a DNA_data_bbnorm_anvio_gene_calls.fna -f Sample_108386_fwd.good.fastq -r Sample_108386_rev.good.fastq -o 108386_MT_mapped_reads
comics -- omics mapping --index-dir /RNA_data/work/mapping/bowtie2-index_bbnorm_MG_genes/ -a DNA_data_bbnorm_anvio_gene_calls.fna -f Sample_108388_fwd.good.fastq -r Sample_108388_rev.good.fastq -o 108388_MT_mapped_reads
comics -- omics mapping --index-dir /RNA_data/work/mapping/bowtie2-index_bbnorm_MG_genes/ -a DNA_data_bbnorm_anvio_gene_calls.fna -f Sample_108390_fwd.good.fastq -r Sample_108390_rev.good.fastq -o 108390_MT_mapped_reads
comics -- omics mapping --index-dir /RNA_data/work/mapping/bowtie2-index_bbnorm_MG_genes/ -a DNA_data_bbnorm_anvio_gene_calls.fna -f Sample_108391_fwd.good.fastq -r Sample_108391_rev.good.fastq -o 108391_MT_mapped_reads
comics -- omics mapping --index-dir /RNA_data/work/mapping/bowtie2-index_bbnorm_MG_genes/ -a DNA_data_bbnorm_anvio_gene_calls.fna -f Sample_108392_fwd.good.fastq -r Sample_108392_rev.good.fastq -o 108392_MT_mapped_reads
comics -- omics mapping --index-dir /RNA_data/work/mapping/bowtie2-index_bbnorm_MG_genes/ -a DNA_data_bbnorm_anvio_gene_calls.fna -f Sample_108394_fwd.good.fastq -r Sample_108394_rev.good.fastq -o 108394_MT_mapped_reads
comics -- omics mapping --index-dir /RNA_data/work/mapping/bowtie2-index_bbnorm_MG_genes/ -a DNA_data_bbnorm_anvio_gene_calls.fna -f Sample_108396_fwd.good.fastq -r Sample_108396_rev.good.fastq -o 108396_MT_mapped_reads
````
**RNA.data.read.mapping.DNA.data.bbnorm.MG.genes.pbs** <font color="green">("/RNA_data/work/")</font> 12:12 (hr:min)  

The results for each individual sample included a mapping directory containing the following files:
<font color="green">"/RNA_data/work/mapping/\*_MT_mapped_reads"</font> *(replace \* with SampleID)*  

- **Sample_RNA_data_bbnorm_anvio_gene_calls.cov** tab-delimited file of the average coverage of each contig in the assembly
- **Sample_RNA_data_bbnorm_anvio_gene_calls.sorted.bam** contains the mapping information sorted by position
- **Sample_RNA_data_bbnorm_anvio_gene_calls.sorted.bam.bai** companion file to the `sorted.bam` file, which contains the index

### 2.4. Convert MT .bam Files

***Convert the MT .bam Files to count tables to be processed statistically***

- Make pile-up file (average read depth per gene)
- BBMap (pileup.sh) **"final.contigs.fixed.bbnorm.sorted.bam"**

````markdown
`r ''`{Terminal}
cd /RNA_data/work/mapping/108379_MT_mapped_reads
pileup.sh in=Sample_108379_RNA_data_bbnorm_anvio_gene_calls.sorted.bam out=Sample_108379_RNA_data_bbnorm_anvio_gene_calls.pileup

#Output
head Sample_108379_RNA_data_bbnorm_anvio_gene_calls.pileup
#ID	Avg_fold	Length	Ref_GC	Covered_percent	Covered_bases	Plus_reads	Minus_reads	Read_GC	Median_fold	Std_Dev
genecall_0	0.0000	300	0.0000	0.0000	0	0	0	0.0000	0	0.00
genecall_1	0.0000	318	0.0000	0.0000	0	0	0	0.0000	0	0.00
genecall_2	0.0000	648	0.0000	0.0000	0	0	0	0.0000	0	0.00
genecall_3	0.0000	375	0.0000	0.0000	0	0	0	0.0000	0	0.00
genecall_4	0.0000	339	0.0000	0.0000	0	0	0	0.0000	0	0.00
genecall_5	0.0000	141	0.0000	0.0000	0	0	0	0.0000	0	0.00
genecall_6	0.0000	132	0.0000	0.0000	0	0	0	0.0000	0	0.00
genecall_7	1.9642	363	0.0000	92.8375	337	4	1	0.7149	1	1.39
genecall_8	0.0000	402	0.0000	0.0000	0	0	0	0.0000	0	0.00
````

- This was done for each Sample `sorted.bam` file to get a table of genes and their counts  

````markdown
`r ''`{Terminal}
cd /RNA_data/work/mapping/108381_MT_mapped_reads
pileup.sh in=Sample_108381_RNA_data_bbnorm_anvio_gene_calls.sorted.bam out=Sample_108381_RNA_data_bbnorm_anvio_gene_calls.pileup
cd /RNA_data/work/mapping/108382_MT_mapped_reads
pileup.sh in=Sample_108382_RNA_data_bbnorm_anvio_gene_calls.sorted.bam out=Sample_108382_RNA_data_bbnorm_anvio_gene_calls.pileup
cd /RNA_data/work/mapping/108383_MT_mapped_reads
pileup.sh in=Sample_108383_RNA_data_bbnorm_anvio_gene_calls.sorted.bam out=Sample_108383_RNA_data_bbnorm_anvio_gene_calls.pileup
cd /RNA_data/work/mapping/108385_MT_mapped_reads
pileup.sh in=Sample_108385_RNA_data_bbnorm_anvio_gene_calls.sorted.bam out=Sample_108385_RNA_data_bbnorm_anvio_gene_calls.pileup
cd /RNA_data/work/mapping/108386_MT_mapped_reads
pileup.sh in=Sample_108386_RNA_data_bbnorm_anvio_gene_calls.sorted.bam out=Sample_108386_RNA_data_bbnorm_anvio_gene_calls.pileup
cd /RNA_data/work/mapping/108388_MT_mapped_reads
pileup.sh in=Sample_108388_RNA_data_bbnorm_anvio_gene_calls.sorted.bam out=Sample_108388_RNA_data_bbnorm_anvio_gene_calls.pileup
cd /RNA_data/work/mapping/108390_MT_mapped_reads
pileup.sh in=Sample_108390_RNA_data_bbnorm_anvio_gene_calls.sorted.bam out=Sample_108390_RNA_data_bbnorm_anvio_gene_calls.pileup
cd /RNA_data/work/mapping/108391_MT_mapped_reads
pileup.sh in=Sample_108391_RNA_data_bbnorm_anvio_gene_calls.sorted.bam out=Sample_108391_RNA_data_bbnorm_anvio_gene_calls.pileup
cd /RNA_data/work/mapping/108392_MT_mapped_reads
pileup.sh in=Sample_108392_RNA_data_bbnorm_anvio_gene_calls.sorted.bam out=Sample_108392_RNA_data_bbnorm_anvio_gene_calls.pileup
cd /RNA_data/work/mapping/108394_MT_mapped_reads
pileup.sh in=Sample_108394_RNA_data_bbnorm_anvio_gene_calls.sorted.bam out=Sample_108394_RNA_data_bbnorm_anvio_gene_calls.pileup
cd /RNA_data/work/mapping/108396_MT_mapped_reads
pileup.sh in=Sample_108396_RNA_data_bbnorm_anvio_gene_calls.sorted.bam out=Sample_108396_RNA_data_bbnorm_anvio_gene_calls.pileup
````
**RNA.data.bam.to.pile.up.pbs** <font color="green">("/RNA_data/work/")</font> 00:01 (hr:min)

- Move all of the **"Sample_\*_RNA_data_bbnorm_anvio_gene_calls.pileup"** files to the <font color="green">"/RNA_data/work/mapping/Pile_Up"</font> directory
- Import these tables into **R** and add up the **"plus read"** and **"minus read"** values for each gene (= total reads of that particular gene)

### 2.5. MT Read Summary

The following table is a summary of the RNA reads by sample and includes **QC Reads**, **Mapped Reads**, and **Average Read Length (bp)**.  All values were derived from the "**Sample_\*.final.contigs.fixed.bbnorm.sorted.bam**" files using `samtools stats`.
```{r echo=FALSE}
rna.informatics<-read.table("Table.Data/rna.bioinformatics.table.txt",sep='\t', quote="", fill=TRUE,header = TRUE)
names(rna.informatics)<-c("ID", "Tundra", "Time", "Replicate", "QC Reads", "Mapped Reads", "Average Length (bp)")
kable(rna.informatics, caption = "Summary statistics for RNA-based metatranscriptome sequencing dataset", format.args = list(big.mark = ","), align = "c") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed"))
```

## 3. Gene Annotations

Following all **Data Bioinformatics** steps above, quality-controlled metatranscriptome reads were mapped to KEGG-annotated coding sequences (CDS) indexed from the metagenome assembly using BBMap to generate pile-up files (average read depth per gene), and SAMtools was used to extract counts and CDS lengths from the BBMap output.

### 3.1. Import Gene Counts
In this section, all `RNA.pileup` files are imported into the R environment and combined into the first **FULL** dataset to be further formatted for downstream statistical analysis.

#### 3.1.1. RNA PileUp Files
The following `RNA.pileup` files (average read depth per gene) for each metatranscriptome sample were generated via the BBMap module.

```{r}
#S108379 -- WS_T0_R1
pileup_108379<-read.table("Pileup.Data/RNA.Pileup/Sample_108379_RNA_data_bbnorm_anvio_gene_calls.pileup", header = TRUE)
pileup_108379$S108379 <- pileup_108379$Plus_reads + pileup_108379$Minus_reads

#S108381 -- WS_T0_R3
pileup_108381<-read.table("Pileup.Data/RNA.Pileup/Sample_108381_RNA_data_bbnorm_anvio_gene_calls.pileup", header = TRUE)
pileup_108381$S108381 <- pileup_108381$Plus_reads + pileup_108381$Minus_reads

#S108385 -- WS_T4_R1
pileup_108385<-read.table("Pileup.Data/RNA.Pileup/Sample_108385_RNA_data_bbnorm_anvio_gene_calls.pileup", header = TRUE)
pileup_108385$S108385 <- pileup_108385$Plus_reads + pileup_108385$Minus_reads

#S108386 -- WS_T4_R2
pileup_108386<-read.table("Pileup.Data/RNA.Pileup/Sample_108386_RNA_data_bbnorm_anvio_gene_calls.pileup", header = TRUE)
pileup_108386$S108386 <- pileup_108386$Plus_reads + pileup_108386$Minus_reads

#S108388 -- WS_T24_R1
pileup_108388<-read.table("Pileup.Data/RNA.Pileup/Sample_108388_RNA_data_bbnorm_anvio_gene_calls.pileup", header = TRUE)
pileup_108388$S108388 <- pileup_108388$Plus_reads + pileup_108388$Minus_reads

#S108390 -- WS_T24_R3
pileup_108390<-read.table("Pileup.Data/RNA.Pileup/Sample_108390_RNA_data_bbnorm_anvio_gene_calls.pileup", header = TRUE)
pileup_108390$S108390 <- pileup_108390$Plus_reads + pileup_108390$Minus_reads

#S108382 -- TUSS_T0_R1
pileup_108382<-read.table("Pileup.Data/RNA.Pileup/Sample_108382_RNA_data_bbnorm_anvio_gene_calls.pileup", header = TRUE)
pileup_108382$S108382 <- pileup_108382$Plus_reads + pileup_108382$Minus_reads

#S108383 -- TUSS_T0_R2
pileup_108383<-read.table("Pileup.Data/RNA.Pileup/Sample_108383_RNA_data_bbnorm_anvio_gene_calls.pileup", header = TRUE)
pileup_108383$S108383 <- pileup_108383$Plus_reads + pileup_108383$Minus_reads

#S108391 -- TUSS_T4_R1
pileup_108391<-read.table("Pileup.Data/RNA.Pileup/Sample_108391_RNA_data_bbnorm_anvio_gene_calls.pileup", header = TRUE)
pileup_108391$S108391 <- pileup_108391$Plus_reads + pileup_108391$Minus_reads

#S108392 -- TUSS_T4_R2
pileup_108392<-read.table("Pileup.Data/RNA.Pileup/Sample_108392_RNA_data_bbnorm_anvio_gene_calls.pileup", header = TRUE)
pileup_108392$S108392 <- pileup_108392$Plus_reads + pileup_108392$Minus_reads

#S108394 -- TUSS_T24_R1
pileup_108394<-read.table("Pileup.Data/RNA.Pileup/Sample_108394_RNA_data_bbnorm_anvio_gene_calls.pileup", header = TRUE)
pileup_108394$S108394 <- pileup_108394$Plus_reads + pileup_108394$Minus_reads

#S108396 -- TUSS_T24_R3
pileup_108396<-read.table("Pileup.Data/RNA.Pileup/Sample_108396_RNA_data_bbnorm_anvio_gene_calls.pileup", header = TRUE)
pileup_108396$S108396 <- pileup_108396$Plus_reads + pileup_108396$Minus_reads
```

#### 3.1.2. Full RNA Dataset

In the previous step, each `RNA.pileup` file was imported and an additional column was added that summed the **"Plus_reads"** and the **Minus_reads**. The sum column in each `RNA.pileup` file was named after the `RNA.pileup` sample name (e.g. **"S108379"**). All `RNA.pileup` files contain identical genecalls in an identical order allowing us to extract the genecall ID and genecall Length columns from a single sample (**S108379**), along with the summed count column of each sample, to create a new dataframe that represents the initial, **FULL** RNA dataset for downstream analyses:

```{r}
cts_rna_all <- data.frame(pileup_108379$ID, pileup_108379$Length, pileup_108379$S108379, pileup_108381$S108381, pileup_108382$S108382, pileup_108383$S108383, pileup_108385$S108385, pileup_108386$S108386, pileup_108388$S108388, pileup_108390$S108390, pileup_108391$S108391, pileup_108392$S108392, pileup_108394$S108394, pileup_108396$S108396)

names(cts_rna_all) <- c("ID", "Length", "S108379", "S108381", "S108382", "S108383", "S108385", "S108386", "S108388", "S108390", "S108391", "S108392", "S108394", "S108396")
```

### 3.2. Import Gene Annotations
The KEGG Functions file (GhostKoala) and KEGG Taxonomy file (GhostKoala) were imported and the functional and taxonomic annotations were matched to their corresponding genecall ID.

#### 3.2.1. KEGG Reference
First, a KEGG reference file was imported with additional KEGG Tier pathways that provide additional, useful, functional categories for downstream analyses.  The Tier II, III, and IV categories were merged to the GhostKoala KEGG functional annotations in subsequent steps.

```{r}
# Import KO_Orthology.txt reference file
KO_ref<-read.table(file='Annotations.Data/KO_Orthology.txt', sep='\t', quote="", fill=TRUE, header = FALSE)

# Rename column headers
names(KO_ref)<-c("Tier_II","Tier_III","Tier_IV","KEGG")

# Separate "KEGG" column into "KO" and "Annotation" columns
KO_ref<-KO_ref %>% separate(KEGG, c("KO_Accession","Annotation"), " ", extra="merge")

# Further separatae "Annotation" column into "Symbol" and "Function" columns
KO_ref<-KO_ref %>% separate(Annotation, c("KO_Symbol","KO_Function"), "; ", extra="merge")
```

#### 3.2.2. KEGG Functions

There are 3,917,616 total unique DNA-based genecalls in the metagenomic dataset.  The GhostKOALA KEGG functional annotations file  ("**DNA_data_bbnorm_anvio_contigs_annotations.tsv**") contains annotations for 1,389,518 unique DNA-based genecalls.  This indicates that **35.5%** of the total unique DNA-based genecalls can be functionally annotated.  That also means that **64.5%** of unique DNA-based genecalls cannot be annotated and will not be included in downstream analyses. Here, the unique genecall functional annotations will be matched to the same unique genecalls in the DNA-based metagenomic dataset.

```{r warning=FALSE, message=FALSE}
# Import the KEGG Functional Annotations file that was exported from the Anvi'o DNA contigs database
KeggAnvio<-read.table(file='Annotations.Data/DNA_data_bbnorm_anvio_contigs_annotations.tsv', sep='\t', quote = "", fill = TRUE, header = TRUE)

# Rename column headers
names(KeggAnvio)<-c("ID","Source","Accession","Function","e-value")

# Separate "Function" column into "Symbol" and "Function" columns
KeggAnvio<-KeggAnvio %>% separate(Function, c("Symbol","Function"), "; ", extra="merge")

# Add "genecall_" to the beginning of each gene ID value to match with downstream analyses
KeggAnvio$ID <- paste("genecall", KeggAnvio$ID, sep="_")

# Add "KO_Symbol" category to the dataset, matched by KO's between the KO_ref file and KeggAnvio
KeggAnvio$KO_Symbol = KO_ref[match(KeggAnvio$Accession, KO_ref$KO_Accession), "KO_Symbol"]

# Add "KO_Function" category to the dataset, matched by KO's between the KO_ref file and KeggAnvio
KeggAnvio$KO_Function = KO_ref[match(KeggAnvio$Accession, KO_ref$KO_Accession), "KO_Function"]

# Add "Tier_II" category to the dataset, matched by KO's between the KO_ref file and KeggAnvio
KeggAnvio$Tier_II = KO_ref[match(KeggAnvio$Accession, KO_ref$KO_Accession), "Tier_II"]

# Add "Tier_III" category to the dataset, matched by KO's between the KO_ref file and KeggAnvio
KeggAnvio$Tier_III = KO_ref[match(KeggAnvio$Accession, KO_ref$KO_Accession), "Tier_III"]

# Add "Tier_IV" category to the dataset, matched by KO's between the KO_ref file and KeggAnvio
KeggAnvio$Tier_IV = KO_ref[match(KeggAnvio$Accession, KO_ref$KO_Accession), "Tier_IV"]

# Keep the Function category from the KO Reference file to potentially fill in gaps in KeggAnvio annotations (many KO's labeled as "None", but the KO should match a known function from the reference file).
KeggFunction<-data.frame(KeggAnvio$ID,KeggAnvio$Accession,KeggAnvio$Symbol,KeggAnvio$KO_Function,KeggAnvio$Tier_II,KeggAnvio$Tier_III,KeggAnvio$Tier_IV)
names(KeggFunction)<-c("ID","KO","Symbol","Function","Tier_II","Tier_III","Tier_IV")
  
# Merge the KO, Symbol, Function, and all Tier columns into single "Combined" column for some downstream analyses
KeggFunction$Combined<-paste(KeggFunction$KO, ":", KeggFunction$Symbol, ":", KeggFunction$Function, ":", KeggFunction$Tier_II, ":", KeggFunction$Tier_III, ":", KeggFunction$Tier_IV)

# Keep just "ID" and "Combined" columns for use with some downstream analyses
KeggData<-data.frame(KeggFunction$ID, KeggFunction$Combined)
names(KeggData)<-c("ID","KEGG")
```

#### 3.2.3. KEGG Taxonomy

There are 3,917,616 total unique DNA-based genecalls in the metagenomic dataset. The GhostKoala KEGG taxonomy file ("**KeggTaxonomy.txt**") contains annotations for 3,912,253 unique DNA-based genecalls.  This indicates that **99.9%** of the total unique DNA-based genecalls can be taxonomically annotated. Here, the unique genecall taxonomic annotations will be matched to the same unique genecalls in the DNA-based metagenomic dataset.

```{r}
#Import KeggTaxonomy.txt from GhostKoala
KeggTaxa<-read.table("Annotations.Data/KeggTaxonomy.txt", header = TRUE, fill = TRUE)
names(KeggTaxa)<-c("ID","Domain","Phylum","Class","Order","Family","Genus","Species")

# Merge the taxonomic classes into a single column
KeggTaxa$Taxonomy <- paste(KeggTaxa$Domain, ":", KeggTaxa$Phylum, ":", KeggTaxa$Class, ":", KeggTaxa$Order, ":", KeggTaxa$Family, ":", KeggTaxa$Genus, ":", KeggTaxa$Species)

# Now keep just "ID" and "Taxonomy"
KeggTaxa <- data.frame(KeggTaxa$ID, KeggTaxa$Taxonomy)
names(KeggTaxa)<-c("ID","Taxonomy")

# Add "genecall_" to the beginning of each gene ID value to match with downstream analyses
KeggTaxa$ID<-paste("genecall", KeggTaxa$ID, sep="_")
```

## 4. Gene Filtering

The **FULL** RNA CTS dataset was assembled as a single dataframe (columns: [1] genecall, [2] length, [3-14] metatranscriptome Samples; rows: 3,917,616 unique "genecalls"), and subsequently subsampled into two new dataframes:

1. **cts_rna_expressed**: retains only those unique genecalls with a value greater than zero in any single sample
    + Indicates that these genes were expressed under at least ONE treatment in the experiment (i.e., actual expression)
    + The count data from these expressed genes will undergo TPM normalization for only those unique genecalls that can be annotated from KEGG
2. **cts_rna_zero**: retains only those unique genecalls that had a value equal to zero in **ALL** samples
    + Indicates that these genes were **NOT** expressed at any point in the experiment (i.e., potential expression only)
    + We retain only those unique genecalls that can be annotated from KEGG to look at which genes were present but not expressed during the experiment

### 4.1. Expressed Genes

Subsample the unique genecalls that have greater than zero raw read counts ("Expressed Genes") across all samples.

```{r}
#cts_rna_expressed
cts_rna_expressed <- subset(cts_rna_all, S108379 > 0 | S108381 > 0 | S108382 > 0 | S108383 > 0 | S108385 > 0 | S108386 > 0 | S108388 > 0 | S108390 > 0 | S108391 > 0 | S108392 > 0 | S108394 > 0 | S108396 > 0, select=c(ID,Length,S108379,S108381,S108382,S108383,S108385,S108386,S108388,S108390,S108391,S108392,S108394,S108396))
```

Add functional and taxonomic annotations to each unique genecall within the "Expressed" subdata.

```{r}
# Expressed Functional and Taxonomic Annotations
cts_rna_expressed$KEGG = KeggData[match(cts_rna_expressed$ID, KeggData$ID), "KEGG"]
cts_rna_expressed$Taxonomy = KeggTaxa[match(cts_rna_expressed$ID, KeggTaxa$ID), "Taxonomy"]

# Expressed Annotated Genes
cts_rna_exp_annotated<-na.omit(cts_rna_expressed, cols="KEGG")
```

### 4.2. Non-Expressed Genes

Also subsample the unique genecalls that have only zero raw read counts ("Non-expressed Genes") across all samples.

```{r}
#cts_rna_zero
cts_rna_zero <- subset(cts_rna_all, S108379 == 0 & S108381 == 0 & S108382 == 0 & S108383 == 0 & S108385 == 0 & S108386 == 0 & S108388 == 0 & S108390 == 0 & S108391 == 0 & S108392 == 0 & S108394 == 0 & S108396 == 0, select=c(ID,Length,S108379,S108381,S108382,S108383,S108385,S108386,S108388,S108390,S108391,S108392,S108394,S108396))
```

Add functional and taxonomic annotations to each unique genecall within the "Non-Expressed" subdata.

```{r}
# Non-Expressed Functional and Taxonomic Annotations
cts_rna_zero$KEGG = KeggData[match(cts_rna_zero$ID, KeggData$ID), "KEGG"]
cts_rna_zero$Taxonomy = KeggTaxa[match(cts_rna_zero$ID, KeggTaxa$ID), "Taxonomy"]

# Non-Expressed Annotated Genes
cts_rna_zero_annotated<-na.omit(cts_rna_zero, cols="KEGG")
```

### 4.3. Expressed Annotated Genes

***From this point***, we  retain only those genecalls that were expressed ***and*** have a KEGG functional annotation.  All remaining genecalls (64.5% of total dataset) are exempt from downstream analyses.  Start by separating the KEGG categories into unique columns. Remove any genecall whose symbol annotation is "None". Further remove any genecall annotated as Tier II "Organismal Systems" or "Human Diseases".

```{r}
# Separate "KEGG" column into "KO", "Symbol", "Function", "Tier_II", "Tier_III", and "Tier_IV" columns
cts_rna_exp_annotated<-cts_rna_exp_annotated %>% separate(KEGG, c("KO","Symbol","Function","Tier_II","Tier_III","Tier_IV"), ": ", extra="merge")

# Remove KO's with "None" as the "Symbol" annotation
cts_rna_exp_annotated<-cts_rna_exp_annotated[!grepl("None", cts_rna_exp_annotated$Symbol),]

# Remove KO's with Tier II category "Organismal Systems"
cts_rna_exp_annotated<-cts_rna_exp_annotated[!grepl("Organismal Systems ", cts_rna_exp_annotated$Tier_II),]

# Remove KO's with Tier II category "Human Diseases"
cts_rna_exp_annotated<-cts_rna_exp_annotated[!grepl("Human Diseases ", cts_rna_exp_annotated$Tier_II),]
```

Separate the Taxonomy categories into unique columns. Remove any genecall whose taxonomy is outside of "Archaea", "Bacteria", or "Fungi".

```{r}
# Separate "Taxonomy" column into "Kingdom", "Phylum", "Class", "Order", "Family", "Genus", and "Species" columns
cts_rna_exp_annotated<-cts_rna_exp_annotated %>% separate(Taxonomy, c("Kingdom","Phylum","Class","Order","Family","Genus","Species"), ": ", extra="merge")

# Remove KO's with "ag" as the "Kingdom" annotation
cts_rna_exp_annotated<-cts_rna_exp_annotated[!grepl("ag", cts_rna_exp_annotated$Kingdom),]

# Remove KO's with "Animals" as the "Kingdom" annotation
cts_rna_exp_annotated<-cts_rna_exp_annotated[!grepl("Animals", cts_rna_exp_annotated$Kingdom),]

# Remove KO's with "Plants" as the "Kingdom" annotation
cts_rna_exp_annotated<-cts_rna_exp_annotated[!grepl("Plants", cts_rna_exp_annotated$Kingdom),]

# Remove KO's with "Protists" as the "Kingdom" annotation
cts_rna_exp_annotated<-cts_rna_exp_annotated[!grepl("Protists", cts_rna_exp_annotated$Kingdom),]

#Re-merge the Taxonomy columns for downstream analysis
cts_rna_exp_annotated$Taxonomy <- paste(cts_rna_exp_annotated$Kingdom, ":", cts_rna_exp_annotated$Phylum, ":", cts_rna_exp_annotated$Class, ":", cts_rna_exp_annotated$Order, ":", cts_rna_exp_annotated$Family, ":", cts_rna_exp_annotated$Genus, ":", cts_rna_exp_annotated$Species)
```

## 5. Gene Normalization

**TPM Normalization:** Converts raw gene counts to ***Transcripts per Million***.  TPM normalization is based on methods described by [Wagner et al. 2012](https://link.springer.com/article/10.1007/s12064-012-0162-3) to normalize transcript abundance data and requires the following two steps:

1. **Calculate Tg**: Tg = Read Count x Average Read Length  / Gene Length
    + ***Read Count***: *number of reads mapped to each unique genecall*
    + ***Average Read Length***: *Average read length (between 120-140 bp) within each sample from sequences that passed QC*
    + ***Gene Length***: *Length of each unique genecall*
2. **Convert Tg into TPM by sample**: TPM = Tg x 1e+06 / sum(Tg)
    + ***The sum of TPM values in each sample will equal 1,000,000***

**At this point, we move forward with only the expressed annotated genecall data for downstream statistical analyses.**

### 5.1. Calculate Tg Values

Calculate Tg values by multiplying the read count of each genecall by the average read length within each sample, then divide by the gene length of each genecall.

```{r}
cts_rna_exp_annotated$S108379_Tg<-(cts_rna_exp_annotated$S108379 * 118) / cts_rna_exp_annotated$Length
cts_rna_exp_annotated$S108381_Tg<-(cts_rna_exp_annotated$S108381 * 129) / cts_rna_exp_annotated$Length
cts_rna_exp_annotated$S108382_Tg<-(cts_rna_exp_annotated$S108382 * 127) / cts_rna_exp_annotated$Length
cts_rna_exp_annotated$S108383_Tg<-(cts_rna_exp_annotated$S108383 * 127) / cts_rna_exp_annotated$Length
cts_rna_exp_annotated$S108385_Tg<-(cts_rna_exp_annotated$S108385 * 129) / cts_rna_exp_annotated$Length
cts_rna_exp_annotated$S108386_Tg<-(cts_rna_exp_annotated$S108386 * 130) / cts_rna_exp_annotated$Length
cts_rna_exp_annotated$S108388_Tg<-(cts_rna_exp_annotated$S108388 * 130) / cts_rna_exp_annotated$Length
cts_rna_exp_annotated$S108390_Tg<-(cts_rna_exp_annotated$S108390 * 129) / cts_rna_exp_annotated$Length
cts_rna_exp_annotated$S108391_Tg<-(cts_rna_exp_annotated$S108391 * 125) / cts_rna_exp_annotated$Length
cts_rna_exp_annotated$S108392_Tg<-(cts_rna_exp_annotated$S108392 * 128) / cts_rna_exp_annotated$Length
cts_rna_exp_annotated$S108394_Tg<-(cts_rna_exp_annotated$S108394 * 127) / cts_rna_exp_annotated$Length
cts_rna_exp_annotated$S108396_Tg<-(cts_rna_exp_annotated$S108396 * 129) / cts_rna_exp_annotated$Length
```

### 5.2. Calculate TPM Values

Now multiple each Tg value within a sample by 1e+06 and divide by the sum of Tg values for that sample to determine TPM.

```{r}
cts_rna_exp_annotated$S108379_TPM<-(cts_rna_exp_annotated$S108379_Tg * 1000000) / sum(cts_rna_exp_annotated$S108379_Tg,na.rm=TRUE)
cts_rna_exp_annotated$S108381_TPM<-(cts_rna_exp_annotated$S108381_Tg * 1000000) / sum(cts_rna_exp_annotated$S108381_Tg,na.rm=TRUE)
cts_rna_exp_annotated$S108382_TPM<-(cts_rna_exp_annotated$S108382_Tg * 1000000) / sum(cts_rna_exp_annotated$S108382_Tg,na.rm=TRUE)
cts_rna_exp_annotated$S108383_TPM<-(cts_rna_exp_annotated$S108383_Tg * 1000000) / sum(cts_rna_exp_annotated$S108383_Tg,na.rm=TRUE)
cts_rna_exp_annotated$S108385_TPM<-(cts_rna_exp_annotated$S108385_Tg * 1000000) / sum(cts_rna_exp_annotated$S108385_Tg,na.rm=TRUE)
cts_rna_exp_annotated$S108386_TPM<-(cts_rna_exp_annotated$S108386_Tg * 1000000) / sum(cts_rna_exp_annotated$S108386_Tg,na.rm=TRUE)
cts_rna_exp_annotated$S108388_TPM<-(cts_rna_exp_annotated$S108388_Tg * 1000000) / sum(cts_rna_exp_annotated$S108388_Tg,na.rm=TRUE)
cts_rna_exp_annotated$S108390_TPM<-(cts_rna_exp_annotated$S108390_Tg * 1000000) / sum(cts_rna_exp_annotated$S108390_Tg,na.rm=TRUE)
cts_rna_exp_annotated$S108391_TPM<-(cts_rna_exp_annotated$S108391_Tg * 1000000) / sum(cts_rna_exp_annotated$S108391_Tg,na.rm=TRUE)
cts_rna_exp_annotated$S108392_TPM<-(cts_rna_exp_annotated$S108392_Tg * 1000000) / sum(cts_rna_exp_annotated$S108392_Tg,na.rm=TRUE)
cts_rna_exp_annotated$S108394_TPM<-(cts_rna_exp_annotated$S108394_Tg * 1000000) / sum(cts_rna_exp_annotated$S108394_Tg,na.rm=TRUE)
cts_rna_exp_annotated$S108396_TPM<-(cts_rna_exp_annotated$S108396_Tg * 1000000) / sum(cts_rna_exp_annotated$S108396_Tg,na.rm=TRUE)
```

### 5.3. Create TPM Dataframe

Create the TPM dataframe by subsetting columns of interest from the cts_exp_annotated dataframe.

```{r}
tpm_all<-data.frame(cts_rna_exp_annotated$ID,cts_rna_exp_annotated$S108379_TPM,cts_rna_exp_annotated$S108381_TPM,cts_rna_exp_annotated$S108382_TPM,cts_rna_exp_annotated$S108383_TPM,cts_rna_exp_annotated$S108385_TPM,cts_rna_exp_annotated$S108386_TPM,cts_rna_exp_annotated$S108388_TPM,cts_rna_exp_annotated$S108390_TPM,cts_rna_exp_annotated$S108391_TPM,cts_rna_exp_annotated$S108392_TPM,cts_rna_exp_annotated$S108394_TPM,cts_rna_exp_annotated$S108396_TPM,cts_rna_exp_annotated$KO,cts_rna_exp_annotated$Symbol,cts_rna_exp_annotated$Function,cts_rna_exp_annotated$Tier_II,cts_rna_exp_annotated$Tier_III,cts_rna_exp_annotated$Tier_IV,cts_rna_exp_annotated$Taxonomy)

names(tpm_all)<-c("ID","S108379_TPM","S108381_TPM","S108382_TPM","S108383_TPM","S108385_TPM","S108386_TPM","S108388_TPM","S108390_TPM","S108391_TPM","S108392_TPM","S108394_TPM","S108396_TPM","KO","Symbol","Function","Tier_II","Tier_III","Tier_IV","Taxonomy")
```

```{r echo=FALSE, eval=FALSE}
write.csv(tpm_all, 'Norm.Results/tpm.all.data.csv')
```

## 6. Gene Datasets

At this point, the **FULL** RNA dataset (`cts_rna_all`) was separated into **Expressed** (`cts_rna_expressed`) and **Non-Expressed** (`cts_rna_zero`) subdata. Then the functional and taxonomic annotations were imported and the **Expressed** and **Non-Expressed** subdata were further separated into **Expressed Annotated** (`cts_rna_exp_annotated`) and **Non-Expressed Annotated** (`cts_rna_zero_annotated`) subdata. The **Non-Expressed Annotated** subdata will be summarized as a list of genes that were NOT expressed at any point during our experiment. Then we moved forward with only the **Expressed Annotated** subdata and normalized the raw read counts using the ***Transcripts per Million*** (TPM) normalization parameters.


Now, we further separate the normalized expressed annotated subdata into their experimental treatments in order to analyze statistical differences in gene expression patterns and differential gene expression between experimental treatments. Pairwise similarities among metatranscriptomes will be calculated using Bray-Curtis similarity values, with differences between treatments assessed via PERMANOVA. Differential gene expresison will be calculated using the `EdgeR` package.

### 6.1. Tussock Gene Data

Separate the **Tussock** samples into their own dataset.  ***Keep only those unique genecalls that have expression in at least 1 Tuss sample.***

```{r}
# Subset the TUSS TPM Normalized Expressed Annotated subdata
tpm_tuss <- subset(tpm_all, select=c(ID,S108382_TPM,S108383_TPM,S108391_TPM,S108392_TPM,S108394_TPM,S108396_TPM,KO,Symbol,Function,Tier_II,Tier_III,Tier_IV, Taxonomy))

names(tpm_tuss)<-c("ID","Tuss1_T0","Tuss2_T0","Tuss1_T4","Tuss2_T4","Tuss1_T24","Tuss3_T24","KO","Symbol","Function","Tier_II","Tier_III","Tier_IV","Taxonomy")

# Remove any non-expressed genes from the Tussock samples
tpm_tuss_expressed <- subset(tpm_tuss, Tuss1_T0 > 0 | Tuss2_T0 > 0 | Tuss1_T4 > 0 | Tuss2_T4 > 0 | Tuss1_T24 > 0 | Tuss3_T24 > 0, select=c(ID,Tuss1_T0,Tuss2_T0,Tuss1_T4,Tuss2_T4,Tuss1_T24,Tuss3_T24,KO,Symbol,Function,Tier_II,Tier_III,Tier_IV,Taxonomy))
```

### 6.2. Wet Sedge Gene Data

Separate the **Wet Sedge** samples into their own dataset.  ***Keep only those unique genecalls that have expression in at least 1 WS sample.***

```{r}
# Subset the WS TPM Normalized Expressed Annotated subdata (that's a mouthful...)
tpm_ws <- subset(tpm_all, select=c(ID,S108379_TPM,S108381_TPM,S108385_TPM,S108386_TPM,S108388_TPM,S108390_TPM,KO,Symbol,Function,Tier_II,Tier_III,Tier_IV,Taxonomy))
names(tpm_ws)<-c("ID","WS1_T0","WS3_T0","WS1_T4","WS2_T4","WS1_T24","WS3_T24","KO","Symbol","Function","Tier_II","Tier_III","Tier_IV","Taxonomy")

# Remove any non-expressed genes from the Wet Sedge samples
tpm_ws_expressed <- subset(tpm_ws, WS1_T0 > 0 | WS3_T0 > 0 | WS1_T4 > 0 | WS2_T4 > 0 | WS1_T24 > 0 | WS3_T24 > 0, select=c(ID,WS1_T0,WS3_T0,WS1_T4,WS2_T4,WS1_T24,WS3_T24,KO,Symbol,Function,Tier_II,Tier_III,Tier_IV,Taxonomy))
```

### 6.3. All Gene Data

We also need the full TUSS and WS Expressed Annotated data together to make comparisons between ecosystems within each sampling timepoint.  Reformat the `tpm_all` table and separate the "KEGG" column into all of its functional categories for downstream use.

```{r}
# Make a new object for the tpm_all table so that you don't overwrite the original
tpm_all_exp_ann<-tpm_all

names(tpm_all_exp_ann)<-c("ID","S108379","S108381","S108382","S108383","S108385","S108386","S108388","S108390","S108391","S108392","S108394","S108396","KO","Symbol","Function","Tier_II","Tier_III","Tier_IV","Taxonomy")
```

## 7. Gene Analysis

### 7.1. Taxonomy
Here, we determine the taxonomic composition of the microbial community by analyzing the relative abundance of gene expression for KEGG tier IV "Ribosomes" (03010 Ribosome PATH: ko03010).  Within the "Ribosome" category, there are annotations for "small subunit ribosomal protein" (ssu) and "large subunit ribosomal protein" (lsu). For Bacteria and Archaea, we use on the ssu annotations (similar to 16S ssu rRNA targeted analysis). For Fungi, we use both the ssu (18S ssu rRNA) and lsu (28S lsu rRNA) annotations.

#### 7.1.1. Bacteria/Archaea SSU
The first analysis pulls out all gene expression for "Ribosomes" and removes lsu annotations such that only ssu annotations are used for determining relative expression of Bacteria and Archaea at each sampling time point in both tussock and wet sedge tundra.

```{r}
# Subset all unique genecalls whose annotation matches "Ribosomes"
taxa_ribo_all<- tpm_all_exp_ann[ which(tpm_all_exp_ann$Tier_IV=='03010 Ribosome [PATH:ko03010]'),]
```

```{r warning=FALSE}
# Make copy of Ribosome dataset to manipulate
taxa_ribo_ssu<-taxa_ribo_all

# Remove "large subunit ribosomal protein" from dataset
taxa_ribo_ssu<-taxa_ribo_ssu[!grepl("large subunit ribosomal protein", taxa_ribo_ssu$Function),]

# Write dataset to file
write.csv(taxa_ribo_ssu, 'Ribo.Results/ALPHA.taxa.rna.ssu.csv')

# Remove "Fungi" from dataset
taxa_ribo_ssu<-taxa_ribo_ssu[!grepl("Fungi", taxa_ribo_ssu$Taxonomy),]

# Separate "Taxonomy" column into all subdivisions
taxa_ribo_ssu<-taxa_ribo_ssu %>% separate(Taxonomy, c("Kingdom","Phylum","Class","Order","Family","Genus","Species"), ": ", extra="merge")

# Keep "Kingdom" and "Phylum"
taxa_ribo_ssu <- data.frame(taxa_ribo_ssu$Kingdom,taxa_ribo_ssu$Phylum, taxa_ribo_ssu$S108379, taxa_ribo_ssu$S108381,taxa_ribo_ssu$S108385,taxa_ribo_ssu$S108386,taxa_ribo_ssu$S108388,taxa_ribo_ssu$S108390,taxa_ribo_ssu$S108382,taxa_ribo_ssu$S108383,taxa_ribo_ssu$S108391,taxa_ribo_ssu$S108392,taxa_ribo_ssu$S108394,taxa_ribo_ssu$S108396)
names(taxa_ribo_ssu)<-c("Kingdom","Phylum","ws1-T0","ws3-T0","ws1-T4","ws2-T4","ws1-T24","ws3-T24","tuss1-T0","tuss2-T0","tuss1-T4","tuss2-T4","tuss1-T24","tuss3-T24")

# Remove "Unclassified" taxa otherwise it will clump unclassified Archaea with unclassified Bacteria (not what we want)
taxa_ribo_ssu<-taxa_ribo_ssu[!grepl("Unclassified", taxa_ribo_ssu$Phylum),]

# Remove "Other" taxa otherwise it will clump "other" Archaea with "other" Bacteria (not what we want)
taxa_ribo_ssu<-taxa_ribo_ssu[!grepl("Other", taxa_ribo_ssu$Phylum),]

# Sort data by Kingdom alphabetically
taxa_ribo_ssu<-taxa_ribo_ssu[order(taxa_ribo_ssu$Kingdom,taxa_ribo_ssu$Phylum),]

# Sum each column by unique Phylum
taxa_ribo_ssu_sum<-taxa_ribo_ssu %>% group_by(Phylum) %>% summarise_at(vars("ws1-T0","ws3-T0","ws1-T4","ws2-T4","ws1-T24","ws3-T24","tuss1-T0","tuss2-T0","tuss1-T4","tuss2-T4","tuss1-T24","tuss3-T24"), sum)
taxa_ribo_ssu_sum<-taxa_ribo_ssu_sum %>% mutate_at(vars('ws1-T0','ws3-T0','ws1-T4','ws2-T4','ws1-T24','ws3-T24','tuss1-T0','tuss2-T0','tuss1-T4','tuss2-T4','tuss1-T24','tuss3-T24'), funs(round(., 0)))
taxa_ribo_ssu_sum<-as.data.frame(taxa_ribo_ssu_sum)

taxa_ribo_ssu_sum
```

```{r echo=FALSE}
# Save taxa_ribo_ssu_sum as .csv file
write.csv(taxa_ribo_ssu_sum, 'Ribo.Results/taxa.ribo.ssu.sum.csv')
```

Repeat the same "Ribosome" analysis as above (at Phylum level), but this time keep all the taxonomy information.
```{r}
# Make copy of taxa_ribo_all to manipulate
taxa_ribo_ssu2<-taxa_ribo_all

# Keep "Taxonomy" column
taxa_ribo_ssu2 <- data.frame(taxa_ribo_ssu2$Function,taxa_ribo_ssu2$Taxonomy,taxa_ribo_ssu2$S108379, taxa_ribo_ssu2$S108381,taxa_ribo_ssu2$S108385,taxa_ribo_ssu2$S108386,taxa_ribo_ssu2$S108388,taxa_ribo_ssu2$S108390,taxa_ribo_ssu2$S108382,taxa_ribo_ssu2$S108383,taxa_ribo_ssu2$S108391,taxa_ribo_ssu2$S108392,taxa_ribo_ssu2$S108394,taxa_ribo_ssu2$S108396)
names(taxa_ribo_ssu2)<-c("Function","Taxonomy","ws1-T0","ws3-T0","ws1-T4","ws2-T4","ws1-T24","ws3-T24","tuss1-T0","tuss2-T0","tuss1-T4","tuss2-T4","tuss1-T24","tuss3-T24")

# Remove "large subunit ribosomal protein" from dataset
taxa_ribo_ssu2<-taxa_ribo_ssu2[!grepl("large subunit ribosomal protein", taxa_ribo_ssu2$Function),]

# Remove "Fungi" from dataset
taxa_ribo_ssu2<-taxa_ribo_ssu2[!grepl("Fungi", taxa_ribo_ssu2$Taxonomy),]
```

```{r}
# Sort data by Kingdom alphabetically
taxa_ribo_ssu2<-taxa_ribo_ssu2[order(taxa_ribo_ssu2$Function,taxa_ribo_ssu2$Taxonomy),]

# Sum each column by unique Taxonomy
taxa_ribo_ssu2_sum<-taxa_ribo_ssu2 %>% group_by(Taxonomy) %>% summarise_at(vars("ws1-T0","ws3-T0","ws1-T4","ws2-T4","ws1-T24","ws3-T24","tuss1-T0","tuss2-T0","tuss1-T4","tuss2-T4","tuss1-T24","tuss3-T24"), sum)
taxa_ribo_ssu2_sum<-taxa_ribo_ssu2_sum %>% mutate_at(vars('ws1-T0','ws3-T0','ws1-T4','ws2-T4','ws1-T24','ws3-T24','tuss1-T0','tuss2-T0','tuss1-T4','tuss2-T4','tuss1-T24','tuss3-T24'), funs(round(., 0)))
taxa_ribo_ssu2_sum<-as.data.frame(taxa_ribo_ssu2_sum)

taxa_ribo_ssu2_sum
```

```{r echo=FALSE}
# Save taxa_ribo_all_sum as .csv file
write.csv(taxa_ribo_ssu2_sum, 'Ribo.Results/taxa_ribo_ssu2_sum.csv')
```

#### 7.1.2. Fungi SSU/LSU
The second analysis pulls out all gene expression for "Ribosomes" and removes Bacteria and Archaea annotations while retaining just Fungi with both ssu and lsu annotations to determine relative expression of Fungi at each sampling time point in both tussock and wet sedge tundra.

```{r}
# Make copy of Ribosome dataset to manipulate
taxa_ribo_fungi<-taxa_ribo_all

# Remove "Bacteria" from dataset
taxa_ribo_fungi<-taxa_ribo_fungi[!grepl("Bacteria", taxa_ribo_fungi$Taxonomy),]

# Remove "Archaea" from dataset
taxa_ribo_fungi<-taxa_ribo_fungi[!grepl("Archaea", taxa_ribo_fungi$Taxonomy),]

# Separate "Taxonomy" column into all subdivisions
taxa_ribo_fungi<-taxa_ribo_fungi %>% separate(Taxonomy, c("Kingdom","Phylum","Class","Order","Family","Genus","Species"), ": ", extra="merge")

# Keep "Kingdom" and "Phylum"
taxa_ribo_fungi <- data.frame(taxa_ribo_fungi$Kingdom,taxa_ribo_fungi$Phylum, taxa_ribo_fungi$S108379, taxa_ribo_fungi$S108381,taxa_ribo_fungi$S108385,taxa_ribo_fungi$S108386,taxa_ribo_fungi$S108388,taxa_ribo_fungi$S108390,taxa_ribo_fungi$S108382,taxa_ribo_fungi$S108383,taxa_ribo_fungi$S108391,taxa_ribo_fungi$S108392,taxa_ribo_fungi$S108394,taxa_ribo_fungi$S108396)
names(taxa_ribo_fungi)<-c("Kingdom","Phylum","ws1-T0","ws3-T0","ws1-T4","ws2-T4","ws1-T24","ws3-T24","tuss1-T0","tuss2-T0","tuss1-T4","tuss2-T4","tuss1-T24","tuss3-T24")

# Remove "Unclassified" taxa
taxa_ribo_fungi<-taxa_ribo_fungi[!grepl("Unclassified", taxa_ribo_fungi$Phylum),]

# Remove "Other" taxa
taxa_ribo_fungi<-taxa_ribo_fungi[!grepl("Other", taxa_ribo_fungi$Phylum),]

# Sort data by Kingdom alphabetically
taxa_ribo_fungi<-taxa_ribo_fungi[order(taxa_ribo_fungi$Kingdom,taxa_ribo_fungi$Phylum),]

# Sum each column by unique Phylum
taxa_ribo_fungi_sum<-taxa_ribo_fungi %>% group_by(Phylum) %>% summarise_at(vars("ws1-T0","ws3-T0","ws1-T4","ws2-T4","ws1-T24","ws3-T24","tuss1-T0","tuss2-T0","tuss1-T4","tuss2-T4","tuss1-T24","tuss3-T24"), sum)
taxa_ribo_fungi_sum<-taxa_ribo_fungi_sum %>% mutate_at(vars('ws1-T0','ws3-T0','ws1-T4','ws2-T4','ws1-T24','ws3-T24','tuss1-T0','tuss2-T0','tuss1-T4','tuss2-T4','tuss1-T24','tuss3-T24'), funs(round(., 0)))
taxa_ribo_fungi_sum<-as.data.frame(taxa_ribo_fungi_sum)

taxa_ribo_fungi_sum
```

```{r echo=FALSE}
# Save taxa_ribo_ssu_sum as .csv file
write.csv(taxa_ribo_fungi_sum, 'Ribo.Results/taxa.ribo.fungi.sum.csv')
```

Repeat the same "Ribosome" analysis as above (at Phylum level), but this time keep all the taxonomy information.
```{r}
# Make copy of taxa_ribo_all to manipulate
taxa_ribo_fungi2<-taxa_ribo_all

# Keep "Taxonomy" column
taxa_ribo_fungi2 <- data.frame(taxa_ribo_fungi2$Function,taxa_ribo_fungi2$Taxonomy,taxa_ribo_fungi2$S108379, taxa_ribo_fungi2$S108381,taxa_ribo_fungi2$S108385,taxa_ribo_fungi2$S108386,taxa_ribo_fungi2$S108388,taxa_ribo_fungi2$S108390,taxa_ribo_fungi2$S108382,taxa_ribo_fungi2$S108383,taxa_ribo_fungi2$S108391,taxa_ribo_fungi2$S108392,taxa_ribo_fungi2$S108394,taxa_ribo_fungi2$S108396)
names(taxa_ribo_fungi2)<-c("Function","Taxonomy","ws1-T0","ws3-T0","ws1-T4","ws2-T4","ws1-T24","ws3-T24","tuss1-T0","tuss2-T0","tuss1-T4","tuss2-T4","tuss1-T24","tuss3-T24")

# Remove "Bacteria" from dataset
taxa_ribo_fungi2<-taxa_ribo_fungi2[!grepl("Bacteria", taxa_ribo_fungi2$Taxonomy),]

# Remove "Archaea" from dataset
taxa_ribo_fungi2<-taxa_ribo_fungi2[!grepl("Archaea", taxa_ribo_fungi2$Taxonomy),]

```

```{r}
# Sort data by Kingdom alphabetically
taxa_ribo_fungi2<-taxa_ribo_fungi2[order(taxa_ribo_fungi2$Function,taxa_ribo_fungi2$Taxonomy),]

# Sum each column by unique Taxonomy
taxa_ribo_fungi2_sum<-taxa_ribo_fungi2 %>% group_by(Taxonomy) %>% summarise_at(vars("ws1-T0","ws3-T0","ws1-T4","ws2-T4","ws1-T24","ws3-T24","tuss1-T0","tuss2-T0","tuss1-T4","tuss2-T4","tuss1-T24","tuss3-T24"), sum)
taxa_ribo_fungi2_sum<-taxa_ribo_fungi2_sum %>% mutate_at(vars('ws1-T0','ws3-T0','ws1-T4','ws2-T4','ws1-T24','ws3-T24','tuss1-T0','tuss2-T0','tuss1-T4','tuss2-T4','tuss1-T24','tuss3-T24'), funs(round(., 0)))
taxa_ribo_fungi2_sum<-as.data.frame(taxa_ribo_fungi2_sum)

taxa_ribo_fungi2_sum
```

```{r echo=FALSE}
# Save taxa_ribo_all_sum as .csv file
write.csv(taxa_ribo_fungi2_sum, 'Ribo.Results/taxa.ribo.fungi2.sum.csv')
```

#### 7.1.3. Plotting Taxonomy

Plot the relative abundance of taxa by all replicates within tussock and wet sedge tundra as a stackplot.
```{r echo=FALSE}
taxa.MT.all.chart<-read.csv("Plot.Data/taxa.MT.all.stackplot.csv")
```

```{r}
# Place taxa in order for plotting
taxa.MT.all.chart$Species<-factor(taxa.MT.all.chart$Species,levels = c("Acidobacteria","Actinobacteria","Alphaproteobacteria","Betaproteobacteria","Deltaproteobacteria","Gammaproteobacteria","Proteobacteria Unclassified","Bacteroidetes","Chloroflexi","Firmicutes","Planctomycetes","Verrucomicrobia","Bacteria Unclassified","Bacteria Other","Archaea","Fungi"))

taxa.MT.all.chart$Species<-fct_rev(taxa.MT.all.chart$Species)

taxa.MT.all.chart$Sample<-factor(taxa.MT.all.chart$Sample,levels = c("Tuss1-MT-T0","Tuss2-MT-T0","Tuss1-MT-T4","Tuss2-MT-T4","Tuss1-MT-T24","Tuss3-MT-T24","WS1-MT-T0","WS3-MT-T0","WS1-MT-T4","WS2-MT-T4","WS1-MT-T24","WS3-MT-T24"))
```

```{r}
colourCount = length(unique(taxa.MT.all.chart$Species))
getPalette = colorRampPalette(brewer.pal(12, "Paired"))

taxa.MT.all.plot<-ggplot(taxa.MT.all.chart, aes(fill=Species, y=Value, x=Sample)) + geom_bar(position = "stack", stat="identity", color="black") + ylab(expression(atop("Relative Taxon", paste("Expression (%)")))) + theme_minimal() + theme(axis.text=element_text(size=10),axis.title=element_text(size=12),axis.title.x=element_blank()) + theme(legend.position = "right", legend.title=element_blank(), legend.text=element_text(size=8), legend.key.size = unit(0.75,"line"), panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_blank(), panel.border = element_blank()) + scale_x_discrete(labels = function(Sample) str_wrap(Sample, width = 8)) + scale_size(guide=FALSE) + scale_fill_manual(values = rev(getPalette(colourCount))) + scale_x_discrete(labels = function(Sample) str_wrap(Sample, width = 15)) + theme(axis.text.x = element_text(angle = 270, hjust=0)) 
```

```{r}
taxa.MT.all.plot
```

```{r, echo=FALSE, eval=FALSE}
# Save the Ribosome Taxonomy plot as an individual figure
setEPS()
postscript("Fig.S2.Ribo.Stackplot.eps", width=7, height = 5.0)
ggarrange(taxa.ribo.plot, heights=c(3,3),
          ncol = 1, nrow = 1)
dev.off()
```

#### 4.2.2 Mean Time Point

Plot the mean relative abundance of taxa by sampling time point within tussock and wet sedge tundra as a stackplot.
```{r echo=FALSE}
taxa.MT.time.mean.chart<-read.csv("Plot.Data/taxa.MT.time.mean.stackplot.csv")
```

```{r}
# Place taxa in order for plotting
taxa.MT.time.mean.chart$Species<-factor(taxa.MT.time.mean.chart$Species,levels = c("Acidobacteria","Actinobacteria","Alphaproteobacteria","Betaproteobacteria","Deltaproteobacteria","Gammaproteobacteria","Proteobacteria Unclassified","Bacteroidetes","Chloroflexi","Firmicutes","Planctomycetes","Verrucomicrobia","Bacteria Unclassified","Bacteria Other","Archaea","Fungi"))

taxa.MT.time.mean.chart$Species<-fct_rev(taxa.MT.time.mean.chart$Species)

taxa.MT.time.mean.chart$Sample<-factor(taxa.MT.time.mean.chart$Sample,levels = c("Tuss-MT-T0","Tuss-MT-T4","Tuss-MT-T24","WS-MT-T0","WS-MT-T4","WS-MT-T24"))
```

```{r}
colourCount = length(unique(taxa.MT.time.mean.chart$Species))
getPalette = colorRampPalette(brewer.pal(12, "Paired"))

taxa.MT.time.mean.plot<-ggplot(taxa.MT.time.mean.chart, aes(fill=Species, y=Value, x=Sample)) + geom_bar(position = "stack", stat="identity", color="black") + ylab(expression(atop("Relative Taxon", paste("Expression (%)")))) + theme_minimal() + theme(axis.text=element_text(size=10),axis.title=element_text(size=12),axis.title.x=element_blank()) + theme(legend.position = "right", legend.title=element_blank(), legend.text=element_text(size=8), legend.key.size = unit(0.75,"line"), panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_blank(), panel.border = element_blank()) + scale_x_discrete(labels = function(Sample) str_wrap(Sample, width = 8)) + scale_size(guide=FALSE) + scale_fill_manual(values = rev(getPalette(colourCount))) + scale_x_discrete(labels = function(Sample) str_wrap(Sample, width = 15)) + theme(axis.text.x = element_text(angle = 270, hjust=0)) 
```

```{r}
taxa.MT.time.mean.plot
```

#### 4.2.3 Mean Tundra

Plot the mean relative abundance of taxa by within tussock and wet sedge tundra as a stackplot.
```{r echo=FALSE}
taxa.MT.tundra.mean.chart<-read.csv("Plot.Data/taxa.MT.tundra.mean.stackplot.csv")
```

```{r}
# Place taxa in order for plotting
taxa.MT.tundra.mean.chart$Species<-factor(taxa.MT.tundra.mean.chart$Species,levels = c("Acidobacteria","Actinobacteria","Alphaproteobacteria","Betaproteobacteria","Deltaproteobacteria","Gammaproteobacteria","Proteobacteria Unclassified","Bacteroidetes","Chloroflexi","Firmicutes","Planctomycetes","Verrucomicrobia","Bacteria Unclassified","Bacteria Other","Archaea","Fungi"))

taxa.MT.tundra.mean.chart$Species<-fct_rev(taxa.MT.tundra.mean.chart$Species)

taxa.MT.tundra.mean.chart$Sample<-factor(taxa.MT.tundra.mean.chart$Sample,levels = c("Tuss-MT","WS-MT"))
```

```{r}
colourCount = length(unique(taxa.MT.tundra.mean.chart$Species))
getPalette = colorRampPalette(brewer.pal(12, "Paired"))

taxa.MT.tundra.mean.plot<-ggplot(taxa.MT.tundra.mean.chart, aes(fill=Species, y=Value, x=Sample)) + geom_bar(position = "stack", stat="identity", color="black") + ylab(expression(atop("Relative Taxon", paste("Expression (%)")))) + theme_minimal() + theme(axis.text=element_text(size=10),axis.title=element_text(size=12),axis.title.x=element_blank()) + theme(legend.position = "right", legend.title=element_blank(), legend.text=element_text(size=8), legend.key.size = unit(0.75,"line"), panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_blank(), panel.border = element_blank()) + scale_x_discrete(labels = function(Sample) str_wrap(Sample, width = 8)) + scale_size(guide=FALSE) + scale_fill_manual(values = rev(getPalette(colourCount))) + scale_x_discrete(labels = function(Sample) str_wrap(Sample, width = 15)) + theme(axis.text.x = element_text(angle = 270, hjust=0)) 
```

```{r}
taxa.MT.tundra.mean.plot
```























Plot the relative expression of ribosomes as a stackplot with the MEAN of samples by time point
```{r echo=FALSE}
taxa.mean.ribo.chart<-read.csv("Plot.Data/taxa.mean.ribo.stackplot.csv")
```

```{r}
# Place taxa in order for plotting
taxa.mean.ribo.chart$Species<-factor(taxa.mean.ribo.chart$Species,levels = c("Acidobacteria","Actinobacteria","Alphaproteobacteria","Betaproteobacteria","Deltaproteobacteria","Gammaproteobacteria","Proteobacteria Unclassified","Bacteroidetes","Chloroflexi","Firmicutes","Planctomycetes","Verrucomicrobia","Bacteria Other","Archaea","Fungi"))

taxa.mean.ribo.chart$Species<-fct_rev(taxa.mean.ribo.chart$Species)

taxa.mean.ribo.chart$Sample<-factor(taxa.mean.ribo.chart$Sample,levels = c("Tuss-16S","Tuss-MG","Tuss-MT-T0","Tuss-MT-T4","Tuss-MT-T24","WS-16S","WS-MG","WS-MT-T0","WS-MT-T4","WS-MT-T24"))
```

```{r warning=FALSE, message=FALSE}
colourCount = length(unique(taxa.mean.ribo.chart$Species))
getPalette = colorRampPalette(brewer.pal(12, "Paired"))

taxa.mean.ribo.plot<-ggplot(taxa.mean.ribo.chart, aes(fill=Species, y=Value, x=Sample)) + geom_bar(position = "stack", stat="identity", color="black") + ylab(" ") + theme_minimal() + theme(axis.text=element_text(size=10),axis.title=element_text(size=12),axis.title.x=element_blank()) + theme(legend.position = "right", legend.title=element_blank(), legend.text=element_text(size=8), legend.key.size = unit(0.75,"line"), panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_blank(), panel.border = element_blank()) + scale_x_discrete(labels = function(Sample) str_wrap(Sample, width = 8)) + scale_size(guide=FALSE) + guides(shape = guide_legend(override.aes = list(size = 1))) + scale_fill_manual(values = rev(getPalette(colourCount)), guide=guide_legend(reverse=FALSE)) + scale_x_discrete(labels = function(Sample) str_wrap(Sample, width = 15)) + theme(axis.text.x = element_text(angle = 270, hjust=0))
```

```{r}
taxa.mean.ribo.plot
```

```{r, echo=FALSE, eval=FALSE}
# Save the Ribosome Taxonomy plot as an individual figure
setEPS()
postscript("Fig.1.Taxa.MG.MT.eps", width=8, height = 5.0)
ggarrange(taxa.mean.ribo.plot, heights=c(4,4),
          ncol = 1, nrow = 1)
dev.off()
```

Plot the relative expression of TUSS ribosomes as a stackplot by MEAN of TIMEPOINT
```{r echo=FALSE}
taxa.tuss.mean.ribo.chart<-read.csv("Plot.Data/taxa.tuss.mean.ribo.stackplot.csv")
```

```{r}
# Place taxa in order for plotting
taxa.tuss.mean.ribo.chart$Species<-factor(taxa.tuss.mean.ribo.chart$Species,levels = c("Acidobacteria","Actinobacteria","Alphaproteobacteria","Betaproteobacteria","Deltaproteobacteria","Gammaproteobacteria","Proteobacteria Unclassified","Bacteroidetes","Chloroflexi","Firmicutes","Planctomycetes","Verrucomicrobia","Bacteria Other","Archaea","Fungi"))

taxa.tuss.mean.ribo.chart$Species<-fct_rev(taxa.tuss.mean.ribo.chart$Species)

taxa.tuss.mean.ribo.chart$Sample<-factor(taxa.tuss.mean.ribo.chart$Sample,levels = c("Tuss-16S","Tuss-MG","Tuss-MT-T0","Tuss-MT-T4","Tuss-MT-T24"))
```

```{r warning=FALSE, message=FALSE}
colourCount = length(unique(taxa.tuss.mean.ribo.chart$Species))
getPalette = colorRampPalette(brewer.pal(12, "Paired"))

taxa.tuss.mean.ribo.plot<-ggplot(taxa.tuss.mean.ribo.chart, aes(fill=Species, y=Value, x=Sample)) + geom_bar(position = "stack", stat="identity", color="black") + ylab(expression(atop("Community Composition", paste("Relative Abundance (%)")))) + scale_fill_manual(values = rev(getPalette(colourCount)), guide=guide_legend(reverse=FALSE)) + theme_classic() + theme(axis.title.x=element_blank(), panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.border = element_blank(), axis.ticks = element_blank(), axis.text.x = element_text(angle = 270, hjust=0), axis.text=element_text(size=10), axis.title=element_text(size=12), legend.position = "none", legend.title=element_blank(), legend.text=element_text(size=10), legend.key.size = unit(0.75,"line")) + scale_y_continuous(expand = c(0, 0), limits = c(0, 101))
```

```{r, eval=FALSE}
taxa.tuss.mean.ribo.plot
```

Plot the relative expression of WS ribosomes as a stackplot by MEAN of TIMEPOINT
```{r echo=FALSE}
taxa.ws.mean.ribo.chart<-read.csv("Plot.Data/taxa.ws.mean.ribo.stackplot.csv")
```

```{r}
# Place taxa in order for plotting
taxa.ws.mean.ribo.chart$Species<-factor(taxa.ws.mean.ribo.chart$Species,levels = c("Acidobacteria","Actinobacteria","Alphaproteobacteria","Betaproteobacteria","Deltaproteobacteria","Gammaproteobacteria","Proteobacteria Unclassified","Bacteroidetes","Chloroflexi","Firmicutes","Planctomycetes","Verrucomicrobia","Bacteria Other","Archaea","Fungi"))

taxa.ws.mean.ribo.chart$Species<-fct_rev(taxa.ws.mean.ribo.chart$Species)

taxa.ws.mean.ribo.chart$Sample<-factor(taxa.ws.mean.ribo.chart$Sample,levels = c("WS-16S","WS-MG","WS-MT-T0","WS-MT-T4","WS-MT-T24"))
```

```{r warning=FALSE, message=FALSE}
colourCount = length(unique(taxa.ws.mean.ribo.chart$Species))
getPalette = colorRampPalette(brewer.pal(12, "Paired"))

taxa.ws.mean.ribo.plot<-ggplot(taxa.ws.mean.ribo.chart, aes(fill=Species, y=Value, x=Sample)) + geom_bar(position = "stack", stat="identity", color="black ") + ylab(expression(atop("Community Composition", paste("Relative Abundance (%)")))) + scale_fill_manual(values = rev(getPalette(colourCount)), guide=guide_legend(reverse=FALSE)) + theme_classic() + theme(axis.title.x=element_blank(), panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.border = element_blank(), axis.ticks = element_blank(), axis.text.x = element_text(angle = 270, hjust=0), axis.text=element_text(size=10), axis.title=element_text(size=12), legend.position = "right", legend.title=element_blank(), legend.text=element_text(size=10), legend.key.size = unit(0.75,"line")) + scale_y_continuous(expand = c(0, 0), limits = c(0, 101))
```

```{r}
taxa.ws.mean.ribo.plot
```

```{r echo=FALSE, eval=FALSE}
# Save the Ribosome Taxonomy plot as an individual figure
setEPS()
postscript("Fig.1.16S.MG.MT.Taxa.eps", width=10.0, height = 5.0)
div.plot<-plot_grid(taxa.tuss.mean.ribo.plot, taxa.ws.mean.ribo.plot,rel_widths = c(2,3))
plot_grid(div.plot,ncol = 1)
dev.off()
```

#### 7.1.4. Taxonomy Statistics

To determine if there are significant differences in mean relative abundance of dominant taxa between tundra ecosystems, we calculate the mean (SD) of each phylum (dominant phylum; >1%) within tussock tundra and within wet sedge tundra throughout the experiment and compare to each other.

```{r echo=FALSE}
taxa.table<-read.csv("Table.Data/taxa.table.csv")

# Rename column headings
colnames(taxa.table)<-c("Taxonomy","Tuss Relative Abundance (%)","WS Relative Abundance (%)","Mean Difference", "Paired t-test (p-value)")

kable(taxa.table, caption = "Relative abundance (mean (SD)) of microbial taxa within tussock tundra and wet sedge tundra with mean difference (%) and significance (p-value).", format.args = list(big.mark = ","), align = "l") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed"))
```

Test for differences in mean relative abundance of dominant phyla between tundra ecosystems.

```{r echo=FALSE}
taxa.phylum.t.test<-read.csv("Stats.Data/taxa.phylum.t.test.csv")
```

```{r}
# Pairwise comparisons between tundra ecosystems for each microbial taxonomic class
taxa.phylum.t.test.stats <- taxa.phylum.t.test %>%
  group_by(Phylum) %>%
  pairwise_t_test(
    Abundance ~ Tundra, paired = TRUE, 
    p.adjust.method = "bonferroni"
    ) %>%
  select(-.y., -n1, -n2, -df, -statistic, -p) # Remove details
taxa.phylum.t.test.stats
```

To determine if there are significant differences in mean relative abundance of dominant taxa between sampling time points within each tundra ecosystem, we calculate the mean (SD) of each phylum (dominant phylum; >1%) at each time point within tussock tundra or within wet sedge tundra throughout the experiment and compare to each other.

Click on the **Show/Hide** button to see the statistics.

<button class="btn btn-primary" data-toggle="collapse" data-target="#BlockName96"> Show/Hide </button>  
<div id="BlockName96" class="collapse">

Tussock Tundra Taxonomy (Phylum) ANOVA
```{r echo=FALSE}
tuss_mt_taxa_stats<-read.csv("Stats.Data/tuss.mt.taxa.anova.stats.csv")
```

```{r}
# Subset response variables for MANOVA
tuss_mt_taxa_stats$response <- as.matrix(tuss_mt_taxa_stats[, 2:12])
```

```{r}
# MANOVA test
tuss_mt_taxa_stats_manova <- manova(response ~ Timepoint, data=tuss_mt_taxa_stats)
summary.aov(tuss_mt_taxa_stats_manova)

## Run ANOVA for each category of interest (significant in MANOVA)

# Acidobacteria
tuss_mt_taxa_stats1<-aov(Acidobacteria~Timepoint,data=tuss_mt_taxa_stats)
summary.aov(tuss_mt_taxa_stats1)
TukeyHSD(tuss_mt_taxa_stats1)

# Actinobacteria
tuss_mt_taxa_stats2<-aov(Actinobacteria~Timepoint,data=tuss_mt_taxa_stats)
summary.aov(tuss_mt_taxa_stats2)
TukeyHSD(tuss_mt_taxa_stats2)
```

Wet Sedge Tundra Taxonomy (Phylum) ANOVA
```{r echo=FALSE}
ws_mt_taxa_stats<-read.csv("Stats.Data/ws.mt.taxa.anova.stats.csv")
```

```{r}
# Subset response variables for MANOVA
ws_mt_taxa_stats$response <- as.matrix(ws_mt_taxa_stats[, 2:12])
```

```{r}
# MANOVA test
ws_mt_taxa_stats_manova <- manova(response ~ Timepoint, data=ws_mt_taxa_stats)
summary.aov(ws_mt_taxa_stats_manova)

## Run ANOVA for each category of interest (significant in MANOVA)

# Acidobacteria
ws_mt_taxa_stats1<-aov(Acidobacteria~Timepoint,data=ws_mt_taxa_stats)
summary.aov(ws_mt_taxa_stats1)
TukeyHSD(ws_mt_taxa_stats1)
```
</div>

We further investigate which classes of taxa are driving these significant differences in relative taxon abundance observed at the phylum level (see above).  We retain only those classes of each dominant phylum where at least 1 sample (out of 12) had > 1.0% relative abundance (to make biologically-relevant statistical comparisons of the data).

Here, we test for differences in mean relative abundance of biologically-relevant microbial classes within each dominant phylum (identified above) between tundra ecosystems.

```{r echo=FALSE}
taxa.class.t.test<-read.csv("Stats.Data/taxa.class.t.test.csv")
```

```{r}
# Pairwise comparisons between tundra ecosystems for each microbial taxonomic class
taxa.class.t.test.stats <- taxa.class.t.test %>%
  group_by(Class) %>%
  pairwise_t_test(
    Abundance ~ Tundra, paired = TRUE, 
    p.adjust.method = "bonferroni"
    ) %>%
  select(-.y., -group1, -group2, -n1, -n2, -df, -statistic, -p)
taxa.class.t.test.stats
```

Here, we separate these taxa by tundra ecosystem and look for differences between sampling time point over the course of the study.

Click on the **Show/Hide** button to see the statistics.

<button class="btn btn-primary" data-toggle="collapse" data-target="#BlockName97"> Show/Hide </button>  
<div id="BlockName97" class="collapse">

Tussock Tundra Taxonomy (All Levels) ANOVA
```{r echo=FALSE}
tuss_mt_all_taxa_stats<-read.csv("Stats.Data/tuss.mt.taxa.all.anova.stats.csv")
```

```{r}
# Subset response variables for MANOVA
tuss_mt_all_taxa_stats$response <- as.matrix(tuss_mt_all_taxa_stats[, 2:44])
```

```{r}
# MANOVA test
tuss_mt_all_taxa_stats_manova <- manova(response ~ Timepoint, data=tuss_mt_all_taxa_stats)
summary.aov(tuss_mt_all_taxa_stats_manova)

## Run ANOVA for each category of interest (significant in MANOVA)

# Acidobacteria
#tuss_mt_taxa_stats1<-aov(Acidobacteria~Timepoint,data=tuss_mt_taxa_stats)
#summary.aov(tuss_mt_taxa_stats1)
#TukeyHSD(tuss_mt_taxa_stats1)

# Actinobacteria
#tuss_mt_taxa_stats2<-aov(Actinobacteria~Timepoint,data=tuss_mt_taxa_stats)
#summary.aov(tuss_mt_taxa_stats2)
#TukeyHSD(tuss_mt_taxa_stats2)
```

Wet Sedge Tundra Taxonomy (Phylum) ANOVA
```{r echo=FALSE}
ws_mt_all_taxa_stats<-read.csv("Stats.Data/ws.mt.taxa.all.anova.stats.csv")
```

```{r}
# Subset response variables for MANOVA
ws_mt_all_taxa_stats$response <- as.matrix(ws_mt_all_taxa_stats[, 2:44])
```

```{r}
# MANOVA test
ws_mt_all_taxa_stats_manova <- manova(response ~ Timepoint, data=ws_mt_all_taxa_stats)
summary.aov(ws_mt_all_taxa_stats_manova)

## Run ANOVA for each category of interest (significant in MANOVA)

# Acidobacteria
#ws_mt_taxa_stats1<-aov(Acidobacteria~Timepoint,data=ws_mt_taxa_stats)
#summary.aov(ws_mt_taxa_stats1)
#TukeyHSD(ws_mt_taxa_stats1)
```
</div>

Alpha Diversity
```{r echo=FALSE}
alpha.rna.taxa<-read.csv("Stats.Data/ALPHA.taxa.rna.ssu.csv")

rownames(alpha.rna.taxa) <- alpha.rna.taxa$ID
alpha.rna.taxa[1] <- NULL
```

```{r}
# Richness: Sum up the number of non-zero entries per row
alpha.rna.taxa.rich <- apply(alpha.rna.taxa>0,1,sum)
write.csv(alpha.rna.taxa.rich, 'Ribo.Results/alpha.rna.taxa.rich.csv')

# Abundance: sum up the number of non-zero entries per row (1)
alpha.rna.taxa.abund <- apply(alpha.rna.taxa,1,sum)
write.csv(alpha.rna.taxa.abund, 'Ribo.Results/alpha.rna.taxa.abund.csv')

# Diversity: Shannon-Wiener Index (H')
alpha.rna.taxa.div <- diversity(alpha.rna.taxa, index="shannon")
write.csv(alpha.rna.taxa.div, 'Ribo.Results/alpha.rna.taxa.div.csv')
```

```{r echo=FALSE}
alpha.rna.taxa.t.test<-read.csv("Stats.Data/alpha.rna.taxa.t.test.csv")
```

Run paired t-test to determine significance.
```{r}
t.test(alpha.rna.taxa.t.test$Tuss,alpha.rna.taxa.t.test$WS,paired=TRUE)
```

Beta Diversity

Calculate the Bray-Curtis dissimilarity matrix to use with `adonis`

```{r echo=FALSE}
beta.rna.taxa<-read.csv("Stats.Data/ALPHA.taxa.rna.ssu.csv")

rownames(beta.rna.taxa) <- beta.rna.taxa$ID
beta.rna.taxa[1] <- NULL
```

```{r}
# Make tpm_all_groups object
beta.rna.taxa_sample<-c("ws1-T0","ws3-T0","tuss1-T0","tuss2-T0","ws1-T4","ws2-T4","ws1-T24","ws3-T24","tuss1-T4","tuss2-T4","tuss1-T24","tuss3-T24")
beta.rna.taxa_veg<-c("WS","WS","TUSS","TUSS","WS","WS","WS","WS","TUSS","TUSS","TUSS","TUSS")
beta.rna.taxa_time<-c("T0","T0","T0","T0","T4","T4","T24","T24","T4","T4","T24","T24")
beta.rna.taxa_groups<-data.frame(beta.rna.taxa_sample,beta.rna.taxa_veg,beta.rna.taxa_time)

# Convert the values in Column 1 ("tpm_sample") into row names
rownames(beta.rna.taxa_groups)<-beta.rna.taxa_groups[,1]
beta.rna.taxa_groups<-beta.rna.taxa_groups[,-1]
```

```{r}
# Bray-Curtis Dissimilarity Matrix for tpm_all_taxa_beta
rna.tpm.taxa.bc.dist<-vegdist(beta.rna.taxa, method = "bray")
```

Run PERMANOVA using `adonis` function of the **Vegan** package

PERMANOVA - MT-TPM-Taxa - Bray-Curtis
```{r}
adonis(rna.tpm.taxa.bc.dist~beta.rna.taxa_veg*beta.rna.taxa_time, data=beta.rna.taxa_groups)
```

### 7.2. Gene Diversity

Alpha diversity of each expressed, annotated metatranscriptome was assessed by the Shannon index. For beta diversity, pairwise similarities among metatranscriptomes were calculated using Bray-Curtis similarity values and visualized with principle coordinates analysis. The difference between treatments was assessed with PERMANOVA.

#### 7.2.1. Alpha Diversity

We calculated alpha diversity based on raw gene counts.  Measurements include **Richness**, **Abundance**, and **Shannon-Wiener Diversity Index (H')**.

```{r}
# Subset the tpm_all_exp_ann file to prepare for beta diversity analysis
cts_rna_all_alpha<-subset(cts_rna_exp_annotated, select=c(S108379,S108381,S108382,S108383,S108385,S108386,S108388,S108390,S108391,S108392,S108394,S108396))
names(cts_rna_all_alpha)<-c("WS1-T0","WS3-T0","Tuss1-T0","Tuss2-T0","WS1-T4","WS2-T4","WS1-T24","WS3-T24","Tuss1-T4","Tuss2-T4","Tuss1-T24","Tuss3-T24")

# Transpose all but the first column ("ID")
cts_rna_all_alpha<-as.data.frame(t(cts_rna_all_alpha))
```

**Richness:** Determine the number of unique genecalls that have more than one count recorded within each sample.

```{r}
# Sum up the number of non-zero entries per row
apply(cts_rna_all_alpha>0,1,sum)
```

**Abundance:** Determine the total abundance of individual genecalls

```{r}
# sum up the number of non-zero entries per row (1)
apply(cts_rna_all_alpha,1,sum)
```

**Shannon-Wiener Diversity Index (H'):** An information index that quantifies the uncertainty associated with predicting the identity of a new genecall given the total number of genecalls and the evenness in count abundances within each genecall.

```{r}
diversity(cts_rna_all_alpha, index="shannon")
```

The following table is a summary of the **Alpha Diversity** on the raw gene counts for the **Expressed Annotated Genes** (*expressed annotated genes for downstream analyses*) by sample.

```{r echo=FALSE}
rna.experiment<-read.table("Table.Data/rna.alpha.summary.table.txt",sep='\t', quote="", fill=TRUE,header = TRUE)
names(rna.experiment)<-c("ID", "Tundra", "Time", "Replicate", "Gene Richness","Gene Abundance","Gene Diversity (H')")
kable(rna.experiment, caption = "Alpha Diversity for Expressed Annotated Genecalls in the RNA-based metatranscriptome sequencing dataset", format.args = list(big.mark = ","), align = "c") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed"))
```

```{r echo=FALSE}
alpha.t.test<-read.csv("Stats.Data/alpha.t.test.csv")
```

Run paired t-test to determine significance.
```{r}
t.test(alpha.t.test$Tuss,alpha.t.test$WS,paired=TRUE)
```

Plot the Alpha Diversity summary.

```{r echo=FALSE}
alpha<-read.csv("Plot.Data/alpha.boxplot.csv")
```

```{r}
# Re-arrange samples for plotting PCoA
alpha$Tundra <- factor(alpha$Tundra, levels=c("Tuss-MT-T0","Tuss-MT-T4","Tuss-MT-T24","WS-MT-T0","WS-MT-T4","WS-MT-T24"))

alpha.boxplot<-ggplot(alpha, aes(x=Tundra, y=H, fill=Tundra)) + 
    geom_boxplot() + ylab(expression(atop("Shannon-Wiener", paste("Diversity Index (H')")))) + theme_classic() + theme(axis.title.x=element_blank(), axis.text=element_text(size=10), axis.text.x = element_text(angle = 270, hjust=0, vjust=0.5), axis.title=element_text(size=12), legend.position = "none", legend.title=element_blank(), legend.text=element_text(size=10)) + ylim(9,12) + annotate(geom="text", x=3.5, y=11.9, label="Paired t-test") + annotate("text", x=3.5, y=11.7, size=3, label = "Tundra~(italic(p) < 0.001)", parse = TRUE) + scale_fill_manual(values=c("palegreen", "green3", "darkgreen", "lightskyblue1", "dodgerblue1", "midnightblue"), breaks=c("Tuss-MG-T0", "WS-MG-T0", "Tuss-MG-T4", "WS-MG-T4", "Tuss-MG-T24", "WS-MG-T24"), labels=c("Tuss-MG-T0", "WS-MG-T0", "Tuss-MG-T4", "WS-MG-T4", "Tuss-MG-T24", "WS-MG-T24"))
```

```{r}
alpha.boxplot
```

#### 7.2.2. Beta Diversity

We calculated beta diversity as the variation in gene expression between our two tundra ecosystems using Bray-Curtis similarity values.

##### 7.2.2.1. Subset Data

First, subset the columns of interest from the **tpm_all_exp_ann** object (ID and all samples) to analyze the beta diversity of both ecosystems together. Create a **tpm_groups** object with columns for sample ID, vegetation type, and timepoint to be used for calculating pairwise similiarities among samples via PERMANOVA (*`adonis` from **Vegan** package*)

```{r}
# Subset the tpm_all_exp_ann file to prepare for beta diversity analysis
tpm_all_beta<-subset(tpm_all_exp_ann, select=c(ID,S108379,S108381,S108382,S108383,S108385,S108386,S108388,S108390,S108391,S108392,S108394,S108396))
names(tpm_all_beta)<-c("ID","WS1-T0","WS3-T0","Tuss1-T0","Tuss2-T0","WS1-T4","WS2-T4","WS1-T24","WS3-T24","Tuss1-T4","Tuss2-T4","Tuss1-T24","Tuss3-T24")

# Remember "ID" column as non-numeric values
tpm_all_beta_ID<-tpm_all_beta$ID

# Transpose all but the first column ("ID")
tpm_all_beta<-as.data.frame(t(tpm_all_beta[,-1]))
colnames(tpm_all_beta)<-tpm_all_beta_ID

# Make tpm_all_groups object
tpm_all_sample<-c("WS1-T0","WS3-T0","Tuss1-T0","Tuss2-T0","WS1-T4","WS2-T4","WS1-T24","WS3-T24","Tuss1-T4","Tuss2-T4","Tuss1-T24","Tuss3-T24")
tpm_all_veg<-c("WS","WS","TUSS","TUSS","WS","WS","WS","WS","TUSS","TUSS","TUSS","TUSS")
tpm_all_time<-c("T0","T0","T0","T0","T4","T4","T24","T24","T4","T4","T24","T24")
tpm_all_groups<-data.frame(tpm_all_sample,tpm_all_veg,tpm_all_time)

# Convert the values in Column 1 ("tpm_sample") into row names
rownames(tpm_all_groups)<-tpm_all_groups[,1]
tpm_all_groups<-tpm_all_groups[,-1]
```

Second, subset the columns of interest from the **tpm_ws_exp_ann** object (ID and all samples) to analyze the beta diversity of the Wet Sedge ecosystem

```{r}
# Subset the tpm_ws_exp_ann file to prepare for beta diversity analysis
tpm_ws_beta<-subset(tpm_all_exp_ann, select=c(ID,S108379,S108381,S108385,S108386,S108388,S108390))
names(tpm_ws_beta)<-c("ID","WS1-T0","WS3-T0","WS1-T4","WS2-T4","WS1-T24","WS3-T24")

# Remember "ID" column as non-numeric values
tpm_ws_beta_ID<-tpm_ws_beta$ID

# Transpose all but the first column ("ID")
tpm_ws_beta<-as.data.frame(t(tpm_ws_beta[,-1]))
colnames(tpm_ws_beta)<-tpm_ws_beta_ID

# Make tpm_ws_groups object
tpm_ws_sample<-c("WS1-T0","WS3-T0","WS1-T4","WS2-T4","WS1-T24","WS3-T24")
tpm_ws_veg<-c("WS","WS","WS","WS","WS","WS")
tpm_ws_time<-c("T0","T0","T4","T4","T24","T24")
tpm_ws_groups<-data.frame(tpm_ws_sample,tpm_ws_veg,tpm_ws_time)

# Convert the values in Column 1 ("tpm_sample") into row names
rownames(tpm_ws_groups)<-tpm_ws_groups[,1]
tpm_ws_groups<-tpm_ws_groups[,-1]
```

Third, subset the columns of interest from the **tpm_tuss_exp_ann** object (ID and all samples) to analyze the beta diversity of the Tussock ecosystem

```{r}
# Subset the tpm_tuss_exp_ann file to prepare for beta diversity analysis
tpm_tuss_beta<-subset(tpm_all_exp_ann, select=c(ID,S108382,S108383,S108391,S108392,S108394,S108396))
names(tpm_tuss_beta)<-c("ID","Tuss1-T0","Tuss2-T0","Tuss1-T4","Tuss2-T4","Tuss1-T24","Tuss3-T24")

# Remember "ID" column as non-numeric values
tpm_tuss_beta_ID<-tpm_tuss_beta$ID

# Transpose all but the first column ("ID")
tpm_tuss_beta<-as.data.frame(t(tpm_tuss_beta[,-1]))
colnames(tpm_tuss_beta)<-tpm_tuss_beta_ID

# Make tpm_tuss_groups object
tpm_tuss_sample<-c("Tuss1-T0","Tuss2-T0","Tuss1-T4","Tuss2-T4","Tuss1-T24","Tuss3-T24")
tpm_tuss_veg<-c("TUSS","TUSS","TUSS","TUSS","TUSS","TUSS")
tpm_tuss_time<-c("T0","T0","T4","T4","T24","T24")
tpm_tuss_groups<-data.frame(tpm_tuss_sample,tpm_tuss_veg,tpm_tuss_time)

# Convert the values in Column 1 ("tpm_sample") into row names
rownames(tpm_tuss_groups)<-tpm_tuss_groups[,1]
tpm_tuss_groups<-tpm_tuss_groups[,-1]
```

##### 7.2.2.2. Bray-Curtis

Calculate the Bray-Curtis dissimilarity matrix to use with `adonis`

```{r}
# Bray-Curtis Dissimilarity Matrix for tpm_all_beta
tpm.all.bc.dist<-vegdist(tpm_all_beta, method = "bray")

# Bray-Curtis Dissimilarity Matrix for tpm_ws_beta
tpm.ws.bc.dist<-vegdist(tpm_ws_beta, method = "bray")

# Bray-Curtis Dissimilarity Matrix for tpm_tuss_beta
tpm.tuss.bc.dist<-vegdist(tpm_tuss_beta, method = "bray")
```

##### 7.2.2.3. PCoA Plots

Principal coordinates analysis (PCoA, also known as metric multidimensional scaling) attempts to represent the distances between samples in a low-dimensional, Euclidean space. In particular, it maximizes the linear correlation between the distances in the distance matrix, and the distances in a space of low dimension (typically, 2 or 3 axes are selected).

The first step of a PCoA is the construction of a (dis)similarity matrix. While PCA (principal component analysis) is based on Euclidean distances, PCoA can handle (dis)similarity matrices calculated from quantitative, semi-quantitative, qualitative, and mixed variables. As always, the choice of (dis)similarity measure is critical and must be suitable to the data in question. For count data, Bray-Curtis distance is recommended. You can use Jaccard index for presence/absence data. When the distance metric is Euclidean, PCoA is equivalent to PCA.

**The first PCoA compares similarity between tundra ecosystems by incorporating the combined dataset.**

```{r}
# Run PCoA analysis on the Bray-Curtis dissimilarity matrix
tpm.all.bc.pcoa<-pcoa(tpm.all.bc.dist)
```

Extract the eigenvalues and calculate the variation explained by each axis.

```{r}
pcoa.all<-cmdscale(tpm.all.bc.dist, eig=TRUE, k=2)
pcoa.all

# Calculate the percent variation explained by Axis 1
pcoa.all.exp.var1<-round(pcoa.all$eig[1] / sum(pcoa.all$eig), 2) * 100
pcoa.all.exp.var1

# Calculate the percent variation explained by Axis 2
pcoa.all.exp.var2<-round(pcoa.all$eig[2] / sum(pcoa.all$eig), 2) * 100
pcoa.all.exp.var2

# Sum the percent variation explained by both axes
pcoa.all.sum.eig<-sum(pcoa.all.exp.var1, pcoa.all.exp.var2)
pcoa.all.sum.eig
```

Plot the PCoA.

```{r}
# Extract the plot scores from first two PCoA axes
tpm.all.bc.pcoa.axes <- tpm.all.bc.pcoa$vectors[,c(1,2)]

# Create new object with PCoA axes
all.bc.pcoa<-data.frame(tpm.all.bc.pcoa.axes)

# Add Timepoint variable to the dataframe
all.timepoint<-c('WS-MT-T0','WS-MT-T0','Tuss-MT-T0','Tuss-MT-T0','WS-MT-T4','WS-MT-T4','WS-MT-T24','WS-MT-T24','Tuss-MT-T4','Tuss-MT-T4','Tuss-MT-T24','Tuss-MT-T24')
all.pcoa.1<-all.bc.pcoa$Axis.1
all.pcoa.2<-all.bc.pcoa$Axis.2
all.bc.pcoa<-data.frame(all.timepoint,all.pcoa.1,all.pcoa.2)

# Rename column headings
colnames(all.bc.pcoa)<-c("Timepoint","PCoA_1","PCoA_2")

# Re-arrange samples for plotting PCoA
all.bc.pcoa$Timepoint <- factor(all.bc.pcoa$Timepoint, levels=c("Tuss-MT-T0","WS-MT-T0","Tuss-MT-T4","WS-MT-T4","Tuss-MT-T24","WS-MT-T24"))

# Plot the PCoA
all.pcoa<-ggplot(data=all.bc.pcoa, aes(x=PCoA_1, y=PCoA_2, group=Timepoint, color=Timepoint)) + geom_point(aes(shape=Timepoint, size=0.75, fill=Timepoint)) + scale_fill_manual(values=c("palegreen","lightskyblue1","green3","dodgerblue1","darkgreen","midnightblue")) + scale_shape_manual(values=c(21,21,22,22,24,24)) + scale_color_manual(values=c("black","black","black","black","black","black")) + xlab("PCoA 1 (57%)") + ylab("PCoA 2 (10%)") + theme_classic() + theme(axis.text=element_text(size=10), axis.title=element_text(size=12), legend.position = "bottom", legend.title=element_blank(), legend.text=element_text(size=10)) + guides(shape = guide_legend(override.aes = list(size = 3))) + scale_size(guide="none") + geom_hline(aes(yintercept = 0), color="gray", linetype="dashed") + geom_vline(aes(xintercept=0), color="gray", linetype="dashed") + ylim(-0.6,0.6) + xlim(-0.6,0.6) + annotate(geom="text", x=0.0, y=0.59, label="PERMANOVA")   + annotate("text", x = 0, y = 0.52, size=3, label = "Tundra~(italic(p) == 0.001)", parse = TRUE) + annotate("text", x = 0, y = 0.45, size=3, label = "Time~Point~(italic(p) == 0.171)", parse = TRUE) + annotate("text", x = 0, y = 0.38, size=3, label = "Tundra:Time~Point~(italic(p) == 0.226)", parse = TRUE)
```

```{r}
all.pcoa
```

```{r echo=FALSE, eval=FALSE}
# Save the ALPHA DIVERSITY and BETA DIVERSITY plots together
setEPS()
postscript("Fig.S2.MT.Diversity.eps", width=6.0, height = 10.0)
ggarrange(alpha.boxplot, all.pcoa, heights=c(3,3),
          labels = c("A","B"),
          ncol = 1, nrow = 2, align="v")
dev.off()
```

**The second PCoA compares similarity between sampling time points within the Tussock tundra ecosystem.**

```{r}
# Run PCoA analysis on the Bray-Curtis dissimilarity matrix
tpm.tuss.bc.pcoa<-pcoa(tpm.tuss.bc.dist)
```

Extract the eigenvalues and calculate the variation explained by each axis.

```{r}
pcoa.tuss<-cmdscale(tpm.tuss.bc.dist, eig=TRUE, k=2)
pcoa.tuss

# Calculate the percent variation explained by Axis 1
pcoa.tuss.exp.var1<-round(pcoa.tuss$eig[1] / sum(pcoa.tuss$eig), 2) * 100
pcoa.tuss.exp.var1

# Calculate the percent variation explained by Axis 2
pcoa.tuss.exp.var2<-round(pcoa.tuss$eig[2] / sum(pcoa.tuss$eig), 2) * 100
pcoa.tuss.exp.var2

# Sum the percent variation explained by both axes
pcoa.tuss.sum.eig<-sum(pcoa.tuss.exp.var1, pcoa.tuss.exp.var2)
pcoa.tuss.sum.eig
```

Plot the PCoA.

```{r}
# Extract the plot scores from first two PCoA axes
tpm.tuss.bc.pcoa.axes <- tpm.tuss.bc.pcoa$vectors[,c(1,2)]

# Create new object with PCoA axes
tuss.bc.pcoa<-data.frame(tpm.tuss.bc.pcoa.axes)

# Add Timepoint variable to the dataframe
tuss.timepoint<-c('Tuss-T0','Tuss-T0','Tuss-T4','Tuss-T4','Tuss-T24','Tuss-T24')
tuss.pcoa.1<-tuss.bc.pcoa$Axis.1
tuss.pcoa.2<-tuss.bc.pcoa$Axis.2
tuss.bc.pcoa<-data.frame(tuss.timepoint,tuss.pcoa.1,tuss.pcoa.2)

# Rename column headings
colnames(tuss.bc.pcoa)<-c("Timepoint","PCoA_1","PCoA_2")
#tuss.bc.pcoa

# Re-arrange samples for plotting PCoA
tuss.bc.pcoa$Timepoint<-factor(tuss.bc.pcoa$Timepoint, levels=c("Tuss-T0","Tuss-T4","Tuss-T24"))

# Plot the PCoA
tuss.pcoa<-ggplot(data=tuss.bc.pcoa, aes(x=PCoA_1, y=PCoA_2, group=Timepoint, color=Timepoint)) + geom_point(aes(shape=Timepoint, size=0.75, fill=Timepoint)) + scale_fill_manual(values=c("palegreen","green3","darkgreen")) + scale_shape_manual(values=c(21,24,22)) + scale_color_manual(values=c("black","black","black")) + xlab("PCoA 1 (39%)") + ylab("PCoA 2 (26%)") + theme_minimal() + theme(axis.text=element_text(size=12),axis.title=element_text(size=14))+ theme(legend.position = "bottom", legend.title=element_blank(), legend.text=element_text(size=12), panel.background = element_blank(), panel.grid.major=element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "gray"), panel.border = element_rect(colour = "gray", fill=NA, size=1)) + scale_size(guide=FALSE) + guides(shape = guide_legend(override.aes = list(size = 3))) + geom_hline(aes(yintercept = 0), color="gray", linetype="dashed") + geom_vline(aes(xintercept=0), color="gray", linetype="dashed") + ylim(-0.5,0.5) + xlim(-0.5,0.5) + annotate(geom="text", x=0.0, y=0.49, label="PERMANOVA") + annotate("text", x = 0, y = 0.42, size=3, label = "Time~Point~(italic(p) == 0.067)", parse = TRUE)
```

```{r}
tuss.pcoa
```

**The third PCoA compares similarity between sampling time points within the Wet Sedge tundra ecosystem.**

```{r}
# Run PCoA analysis on the Bray-Curtis dissimilarity matrix
tpm.ws.bc.pcoa<-pcoa(tpm.ws.bc.dist)
```

Extract the eigenvalues and calculate the variation explained by each axis.

```{r}
pcoa.ws<-cmdscale(tpm.ws.bc.dist, eig=TRUE, k=2)
pcoa.ws

# Calculate the percent variation explained by Axis 1
pcoa.ws.exp.var1<-round(pcoa.ws$eig[1] / sum(pcoa.ws$eig), 2) * 100
pcoa.ws.exp.var1

# Calculate the percent variation explained by Axis 2
pcoa.ws.exp.var2<-round(pcoa.ws$eig[2] / sum(pcoa.ws$eig), 2) * 100
pcoa.ws.exp.var2

# Sum the percent variation explained by both axes
pcoa.ws.sum.eig<-sum(pcoa.ws.exp.var1, pcoa.ws.exp.var2)
pcoa.ws.sum.eig
```

Plot the PCoA.

```{r}
# Extract the plot scores from first two PCoA axes
tpm.ws.bc.pcoa.axes <- tpm.ws.bc.pcoa$vectors[,c(1,2)]

# Create new object with PCoA axes
ws.bc.pcoa<-data.frame(tpm.ws.bc.pcoa.axes)

# Add Timepoint variable to the dataframe
ws.timepoint<-c('WS-T0','WS-T0','WS-T4','WS-T4','WS-T24','WS-T24')
ws.pcoa.1<-ws.bc.pcoa$Axis.1
ws.pcoa.2<-ws.bc.pcoa$Axis.2
ws.bc.pcoa<-data.frame(ws.timepoint,ws.pcoa.1,ws.pcoa.2)

# Rename column headings
colnames(ws.bc.pcoa)<-c("Timepoint","PCoA_1","PCoA_2")
#ws.bc.pcoa

# Re-arrange samples for plotting PCoA
ws.bc.pcoa$Timepoint<-factor(ws.bc.pcoa$Timepoint, levels=c("WS-T0","WS-T4","WS-T24"))

# Plot the PCoA
ws.pcoa<-ggplot(data=ws.bc.pcoa, aes(x=PCoA_1, y=PCoA_2, group=Timepoint, color=Timepoint)) + geom_point(aes(shape=Timepoint, size=0.75, fill=Timepoint)) + scale_fill_manual(values=c("lightskyblue1","dodgerblue1","midnightblue")) + scale_shape_manual(values=c(21,24,22)) + scale_color_manual(values=c("black","black","black")) + xlab("PCoA 1 (44%)") + ylab("PCoA 2 (19%)") + theme_minimal() + theme(axis.text=element_text(size=12),axis.title=element_text(size=14))+ theme(legend.position = "bottom", legend.title=element_blank(), legend.text=element_text(size=12), panel.background = element_blank(), panel.grid.major=element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "gray"), panel.border = element_rect(colour = "gray", fill=NA, size=1)) + scale_size(guide=FALSE) + guides(shape = guide_legend(override.aes = list(size = 3))) + geom_hline(aes(yintercept = 0), color="gray", linetype="dashed") + geom_vline(aes(xintercept=0), color="gray", linetype="dashed") + ylim(-0.5,0.5) + xlim(-0.5,0.5) + annotate(geom="text", x=0.0, y=0.49, label="PERMANOVA") + annotate("text", x = 0, y = 0.42, size=3, label = "Time~Point~(italic(p) == 0.333)", parse = TRUE)
```

```{r}
ws.pcoa
```

```{r echo=FALSE, eval=FALSE}
setEPS()
postscript("Fig.2.PCoA.MG.MT.eps", width=8.0, height = 8.0)
ggarrange(tuss.dna.pcoa, tuss.pcoa, ws.dna.pcoa, ws.pcoa, heights=c(4,4),
          labels = c("A","B","C","D"),
          ncol = 2, nrow = 2, align="v")
dev.off()
```

##### 7.2.2.4. PERMANOVA

- Run PERMANOVA using `adonis` function of the **Vegan** package

PERMANOVA - TPM-All - Bray-Curtis
```{r}
adonis(tpm.all.bc.dist~tpm_all_veg*tpm_all_time, data=tpm_all_groups)
```

PERMANOVA - TPM-Tuss - Bray-Curtis
```{r}
adonis(tpm.tuss.bc.dist~tpm_tuss_time, data=tpm_tuss_groups)
```

PERMANOVA - TPM-WS - Bray-Curtis
```{r}
adonis(tpm.ws.bc.dist~tpm_ws_time, data=tpm_ws_groups)
```

### 7.3. KEGG Tier Patterns

Look for differences in gene expression at multiple KEGG tier levels (II-IV) using TPM-normalized values.

#### 7.3.1. Tussock Tundra

Look at differences in gene expression patterns for multiple Tier levels within the Tussock tundra between sampling time points.

##### 7.3.1.1. Tier II Expression
```{r}
# Calculate TPM-normalized gene count sums for each Tuss sample in each Tier II category
tuss_tII_sum<-tpm_all_exp_ann %>% group_by(Tier_II) %>% summarise_at(vars("S108382","S108383","S108391","S108392","S108394","S108396"), sum)

tuss_tII_sum<-tuss_tII_sum %>% mutate_at(vars(S108382,S108383,S108391,S108392,S108394,S108396), funs(round(., 0)))

tuss_tII_sum<-as.data.frame(tuss_tII_sum)

tuss_tII_sum<-tuss_tII_sum[order(-tuss_tII_sum$S108382),]
```

Heatmap for Tier II categories by sampling timepoints
```{r}
# Create copy of "ws_tII_sum" object to prep for heatmap
tuss_tII_heatmap<-tuss_tII_sum

# Convert the first column (Tier categories) into rownames
rownames(tuss_tII_heatmap) <- tuss_tII_heatmap$Tier_II
tuss_tII_heatmap<-as.data.frame(tuss_tII_heatmap[-1])

# Rename columns to sample IDs
colnames(tuss_tII_heatmap)<-c("Tuss1-T0","Tuss2-T0","Tuss1-T4","Tuss2-T4","Tuss1-T24","Tuss3-T24")

# Remove rotuss for "Organismal Systems" and "Human Diseases"
#tuss_tII_heatmap <- tuss_tII_heatmap[-c(5,6),]

# Convert dataframe into a matrix for heatmap
tuss_tII_heatmap<-as.matrix(tuss_tII_heatmap)

# Scale matrix values to generate Z-scores
tuss_tII_heatmap<-scale(t(tuss_tII_heatmap))

tuss_tII_heatmap<-t(tuss_tII_heatmap)

# Specify RColorBrewer custom color palette
col <- colorRampPalette(brewer.pal(10, "RdYlBu"))(256)

# Pheatmap
pheatmap(tuss_tII_heatmap, treeheight_row = 0, treeheight_col = 0,cluster_rows = FALSE, cluster_cols = FALSE)
```

Summed values by replicate for each KEGG tier II category.
```{r echo=FALSE}
kable(tuss_tII_sum, caption = "Tussock Tier II KEGG category summary by sample.", format.args = list(big.mark=",")) %>% kable_styling(bootstrap_options = c("striped","hover","condensed"))
```

Calculate mean values from the replicate sums
```{r}
# Calculate mean values from the replicate sums
tuss_tII_sum$meantuss_T0<-apply(tuss_tII_sum[,2:3], 1, mean)
tuss_tII_sum$meantuss_T4<-apply(tuss_tII_sum[,4:5], 1, mean)
tuss_tII_sum$meantuss_T24<-apply(tuss_tII_sum[,6:7], 1, mean)
tuss_tII_sum<-tuss_tII_sum %>% mutate_at(vars(meantuss_T0,meantuss_T4,meantuss_T24), funs(round(., 0)))

# Calculate sd values from the replicate sums
tuss_tII_sum$sdtuss_T0<-apply(tuss_tII_sum[,2:3], 1, sd)
tuss_tII_sum$sdtuss_T4<-apply(tuss_tII_sum[,4:5], 1, sd)
tuss_tII_sum$sdtuss_T24<-apply(tuss_tII_sum[,6:7], 1, sd)
tuss_tII_sum<-tuss_tII_sum %>% mutate_at(vars(sdtuss_T0,sdtuss_T4,sdtuss_T24), funs(round(., 0)))

# Subset Mean values
tuss_tII_mean<-subset(tuss_tII_sum, select=c(Tier_II,meantuss_T0,meantuss_T4,meantuss_T24))
names(tuss_tII_mean)<-c("Tier II", "Tuss-T0", "Tuss-T4", "Tuss-T24")

# Subset SD values
tuss_tII_sd<-subset(tuss_tII_sum, select=c(Tier_II,sdtuss_T0,sdtuss_T4,sdtuss_T24))
names(tuss_tII_sd)<-c("Tier II", "Tuss-T0", "Tuss-T4", "Tuss-T24")

# Remember "Tier II" as non-numeric values
tuss_tII_mean_TierII<-tuss_tII_mean$`Tier II`
tuss_tII_sd_TierII<-tuss_tII_sd$`Tier II`

# Transpose all but first column (Tier II)
tuss_tII_mean<-as.data.frame(t(tuss_tII_mean[,-1]))
colnames(tuss_tII_mean)<-tuss_tII_mean_TierII
tuss_tII_sd<-as.data.frame(t(tuss_tII_sd[,-1]))
colnames(tuss_tII_sd)<-tuss_tII_sd_TierII

# Combine mean and sd into single column with ± divider
tuss_tII_table<-as.data.frame(do.call(cbind, lapply(1:ncol(tuss_tII_mean), function(i) paste0(tuss_tII_mean[ , i], " ± ", tuss_tII_sd[ , i]))))

# Transpose the table back
tuss_tII_table<-t(tuss_tII_table)

# Rename columns to Sites
colnames(tuss_tII_table)<-c("Tussock (T0)", "Tussock (T4)", "Tussock (T24)")

rownames(tuss_tII_table)<-tuss_tII_mean_TierII

kable(tuss_tII_table, caption = "Tussock Tier II KEGG category averages (± standard deviation) by sample.", format.args = list(big.mark=","), align = "l") %>% kable_styling(bootstrap_options = c("striped","hover","condensed"))
```

Run statistical tests to determine if significant differences exist between sampling timepoints for each Tier KEGG category. Click on the **Show/Hide** button to see the statistics.

<button class="btn btn-primary" data-toggle="collapse" data-target="#BlockName6"> Show/Hide </button>  
<div id="BlockName6" class="collapse">
```{r}
# Subset the "sum" values for each sample
tuss_tII_stats<-subset(tuss_tII_sum, select=c(Tier_II,S108382,S108383,S108391,S108392,S108394,S108396))

rownames(tuss_tII_stats) <- tuss_tII_stats$Tier_II
tuss_tII_stats<-as.data.frame(t(tuss_tII_stats[-1]))

# Create a column with timepoint variable names
Timepoint<-c("T0","T0","T4","T4","T24","T24")
Timepoint<-data.frame(Timepoint)

# Add the timepoint column to the sum table
tuss_tII_stats<-data.frame(Timepoint,tuss_tII_stats)

# Subset response variables for MANOVA
tuss_tII_stats$response <- as.matrix(tuss_tII_stats[, 2:5])

# MANOVA test
tuss_tII_manova <- manova(response ~ Timepoint, data=tuss_tII_stats)
summary.aov(tuss_tII_manova)
```

Run ANOVA for each category of interest (significant in MANOVA)
```{r}
# Metabolism
tuss_tII_aov1<-aov(Metabolism.~Timepoint,data=tuss_tII_stats)
summary.aov(tuss_tII_aov1)
TukeyHSD(tuss_tII_aov1)

# Genetic Information Processing
tuss_tII_aov2<-aov(Genetic.Information.Processing.~Timepoint,data=tuss_tII_stats)
summary.aov(tuss_tII_aov2)
TukeyHSD(tuss_tII_aov2)

# Environmental Information Processing
tuss_tII_aov3<-aov(Environmental.Information.Processing.~Timepoint,data=tuss_tII_stats)
summary.aov(tuss_tII_aov3)
TukeyHSD(tuss_tII_aov3)

# Cellular Processing
tuss_tII_aov4<-aov(Cellular.Processes.~Timepoint,data=tuss_tII_stats)
summary.aov(tuss_tII_aov4)
TukeyHSD(tuss_tII_aov4)
```
</div>

Plot the KEGG tier II categories as a barchart.
```{r, echo=FALSE}
tuss_tII_bardata<-read.csv("Plot.Data/tuss_tII_barplot.csv")
```

```{r}
# Place the KEGG tier II categories in the preferred order for plotting
tuss_tII_bardata$Tier.II <- factor(tuss_tII_bardata$Tier.II,levels = c("Metabolism", "Genetic Information Processing", "Environmental Information Processing", "Cellular Processes"))

tuss_tII_bardata$Sample <- factor(tuss_tII_bardata$Sample,levels=c("Tuss-T0","Tuss-T4","Tuss-T24"))

tuss_tII_barplot<-ggplot(tuss_tII_bardata, aes(x = Tier.II, y = Mean, fill = Sample)) + geom_bar(stat = "identity", position = "dodge", color="black") + geom_errorbar(aes(ymin = Mean - SD, ymax = Mean + SD), width = 0.2, position = position_dodge(0.9)) + ylab(expression(atop("KEGG Tier II Categories", paste("Transcript Counts (TPM)")))) + theme_classic() + theme(axis.text=element_text(size=10), axis.title=element_text(size=12), axis.title.x=element_blank()) + theme(legend.position = "bottom", legend.title=element_blank(), legend.text=element_text(size=8), panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.border = element_blank()) + scale_size(guide=FALSE) + scale_fill_manual(values=c("palegreen", "green3", "darkgreen")) + scale_x_discrete(labels = function(Tier.II) str_wrap(Tier.II, width = 10)) + scale_y_continuous(expand = c(0, 0), limits = c(0, 800000)) + annotate(geom="text", x=0.70, y=750000, label="a") + annotate(geom="text", x=1.0, y=475000, label="b") + annotate(geom="text", x=1.3, y=475000, label="b") + annotate(geom="text", x=1.7, y=350000, label="a") + annotate(geom="text", x=2.0, y=600000, label="b") + annotate(geom="text", x=2.3, y=600000, label="b") + annotate(geom="text", x=3.0, y=150000, label="italic(N.S.)", parse=TRUE) + annotate(geom="text", x=3.7, y=100000, label="a") + annotate(geom="text", x=4.0, y=150000, label="b") + annotate(geom="text", x=4.3, y=120000, label="a")
```

```{r}
tuss_tII_barplot
```

Plot the combined METAGENOME and METATRANSCRIPTOME KEGG tier II categories as a barchart.
```{r, echo=FALSE}
tuss_mg_mt_tII_bardata<-read.csv("Plot.Data/tuss_mg_mt_tII_barplot.csv")
```

```{r}
# Place the KEGG tier II categories in the preferred order for plotting
tuss_mg_mt_tII_bardata$Tier.II <- factor(tuss_mg_mt_tII_bardata$Tier.II,levels = c("Metabolism", "Genetic Information Processing", "Environmental Information Processing", "Cellular Processes"))

tuss_mg_mt_tII_bardata$Sample <- factor(tuss_mg_mt_tII_bardata$Sample,levels=c("Tuss-MG","Tuss-MT-T0","Tuss-MT-T4","Tuss-MT-T24"))

tuss_mg_mt_tII_barplot<-ggplot(tuss_mg_mt_tII_bardata, aes(x = Tier.II, y = Mean, fill = Sample)) + geom_bar(stat = "identity", position = "dodge", color="black") + geom_errorbar(aes(ymin = Mean - SD, ymax = Mean + SD), width = 0.2, position = position_dodge(0.9)) + ylab(expression(atop("KEGG Tier II Categories", paste("Gene Counts")))) + theme_classic() + theme(axis.text=element_text(size=10), axis.title=element_text(size=12), axis.title.x=element_blank()) + theme(legend.position = "bottom", legend.title=element_blank(), legend.text=element_text(size=8), panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.border = element_blank()) + scale_size(guide=FALSE) + scale_fill_manual(values=c("gray83","palegreen", "green3", "darkgreen")) + scale_x_discrete(labels = function(Tier.II) str_wrap(Tier.II, width = 10)) + scale_y_continuous(expand = c(0, 0), limits = c(0, 800000)) + annotate(geom="text", x=0.90, y=750000, label="a") + annotate(geom="text", x=1.12, y=475000, label="b") + annotate(geom="text", x=1.35, y=475000, label="b") + annotate(geom="text", x=1.9, y=350000, label="a") + annotate(geom="text", x=2.12, y=600000, label="b") + annotate(geom="text", x=2.35, y=600000, label="b") + annotate(geom="text", x=3.12, y=150000, label="italic(N.S.)", parse=TRUE) + annotate(geom="text", x=3.9, y=100000, label="a") + annotate(geom="text", x=4.12, y=150000, label="b") + annotate(geom="text", x=4.35, y=120000, label="a")
```

```{r}
tuss_mg_mt_tII_barplot
```

##### 7.3.1.2. Tier III Expression

Calculate TPM-normalized gene count sums for each Tuss sample in each Tier III category.
```{r}
# Calculate TPM-normalized gene count sums for each Tuss sample in each Tier III category
tuss_tIII_sum<-tpm_all_exp_ann %>% group_by(Tier_III) %>% summarise_at(vars("S108382","S108383","S108391","S108392","S108394","S108396"), sum)
tuss_tIII_sum<-tuss_tIII_sum %>% mutate_at(vars(S108382,S108383,S108391,S108392,S108394,S108396), funs(round(., 0)))
tuss_tIII_sum<-as.data.frame(tuss_tIII_sum)
tuss_tIII_sum<-tuss_tIII_sum[order(-tuss_tIII_sum$S108382),]
```

```{r}
# Calculate mean values from the replicate sums
tuss_tIII_sum$meantuss_T0<-apply(tuss_tIII_sum[,2:3], 1, mean)
tuss_tIII_sum$meantuss_T4<-apply(tuss_tIII_sum[,4:5], 1, mean)
tuss_tIII_sum$meantuss_T24<-apply(tuss_tIII_sum[,6:7], 1, mean)
tuss_tIII_sum<-tuss_tIII_sum %>% mutate_at(vars(meantuss_T0,meantuss_T4,meantuss_T24), funs(round(., 0)))

# Calculate sd values from the replicate sums
tuss_tIII_sum$sdtuss_T0<-apply(tuss_tIII_sum[,2:3], 1, sd)
tuss_tIII_sum$sdtuss_T4<-apply(tuss_tIII_sum[,4:5], 1, sd)
tuss_tIII_sum$sdtuss_T24<-apply(tuss_tIII_sum[,6:7], 1, sd)
tuss_tIII_sum<-tuss_tIII_sum %>% mutate_at(vars(sdtuss_T0,sdtuss_T4,sdtuss_T24), funs(round(., 0)))

# Write data to .csv file
write.csv(tuss_tIII_sum, 'Norm.Results/tuss.rna.tIII.mean.sd.csv')

# Subset Mean values
tuss_tIII_mean<-subset(tuss_tIII_sum, select=c(Tier_III,meantuss_T0,meantuss_T4,meantuss_T24))
names(tuss_tIII_mean)<-c("Tier III", "Tuss-T0", "Tuss-T4", "Tuss-T24")

# Subset SD values
tuss_tIII_sd<-subset(tuss_tIII_sum, select=c(Tier_III,sdtuss_T0,sdtuss_T4,sdtuss_T24))
names(tuss_tIII_sd)<-c("Tier III", "Tuss-T0", "Tuss-T4", "Tuss-T24")

# Remember "Tier III" as non-numeric values
tuss_tIII_mean_TierIII<-tuss_tIII_mean$`Tier III`
tuss_tIII_sd_TierIII<-tuss_tIII_sd$`Tier III`

# Transpose all but first column (Tier II)
tuss_tIII_mean<-as.data.frame(t(tuss_tIII_mean[,-1]))
colnames(tuss_tIII_mean)<-tuss_tIII_mean_TierIII
tuss_tIII_sd<-as.data.frame(t(tuss_tIII_sd[,-1]))
colnames(tuss_tIII_sd)<-tuss_tIII_sd_TierIII

# Combine mean and sd into single column with ± divider
tuss_tIII_table<-as.data.frame(do.call(cbind, lapply(1:ncol(tuss_tIII_mean), function(i) paste0(tuss_tIII_mean[ , i], " ± ", tuss_tIII_sd[ , i]))))

# Transpose the table back
tuss_tIII_table<-t(tuss_tIII_table)

# Rename columns to Sites
colnames(tuss_tIII_table)<-c("Tussock (T0)", "Tussock (T4)", "Tussock (T24)")

rownames(tuss_tIII_table)<-tuss_tIII_mean_TierIII
```

Run statistical tests to determine if significant differences exist between sampling timepoints for each Tier KEGG category. Click on the **Show/Hide** button to see the statistics.

<button class="btn btn-primary" data-toggle="collapse" data-target="#BlockName7"> Show/Hide </button>  
<div id="BlockName7" class="collapse">
```{r}
# Subset the "sum" values for each sample
tuss_tIII_stats<-subset(tuss_tIII_sum, select=c(Tier_III,S108382,S108383,S108391,S108392,S108394,S108396))

rownames(tuss_tIII_stats) <- tuss_tIII_stats$Tier_III
tuss_tIII_stats<-as.data.frame(t(tuss_tIII_stats[-1]))

# Add the timepoint column to the sum table
tuss_tIII_stats<-data.frame(Timepoint,tuss_tIII_stats)

# Subset response variables for MANOVA
tuss_tIII_stats$response <- as.matrix(tuss_tIII_stats[, 2:26])

# MANOVA test
tuss_tIII_manova <- manova(response ~ Timepoint, data=tuss_tIII_stats)
summary.aov(tuss_tIII_manova)
```

Run ANOVA for each category of interest (significant in MANOVA)
```{r}
# Overview
tuss_tIII_aov1<-aov(Overview.~Timepoint,data=tuss_tIII_stats)
summary.aov(tuss_tIII_aov1)
TukeyHSD(tuss_tIII_aov1)

# Carbohydrate Metabolism
tuss_tIII_aov2<-aov(Carbohydrate.metabolism.~Timepoint,data=tuss_tIII_stats)
summary.aov(tuss_tIII_aov2)
TukeyHSD(tuss_tIII_aov2)

# Translation
tuss_tIII_aov3<-aov(Translation.~Timepoint,data=tuss_tIII_stats)
summary.aov(tuss_tIII_aov3)
TukeyHSD(tuss_tIII_aov3)

# Folding, Sorting, and Degradation
tuss_tIII_aov4<-aov(Folding..sorting.and.degradation.~Timepoint,data=tuss_tIII_stats)
summary.aov(tuss_tIII_aov4)
TukeyHSD(tuss_tIII_aov4)

# Metabolism of Cofactors and Vitamins
tuss_tIII_aov5<-aov(Metabolism.of.cofactors.and.vitamins.~Timepoint,data=tuss_tIII_stats)
summary.aov(tuss_tIII_aov5)
TukeyHSD(tuss_tIII_aov5)

# Amino Acid Metabolism
tuss_tIII_aov6<-aov(Amino.acid.metabolism.~Timepoint,data=tuss_tIII_stats)
summary.aov(tuss_tIII_aov6)
TukeyHSD(tuss_tIII_aov6)

# Cell Growth and Death
tuss_tIII_aov7<-aov(Cell.growth.and.death.~Timepoint,data=tuss_tIII_stats)
summary.aov(tuss_tIII_aov7)
TukeyHSD(tuss_tIII_aov7)

# Xenobiotics Biodegradation and Metabolism
tuss_tIII_aov8<-aov(Xenobiotics.biodegradation.and.metabolism.~Timepoint,data=tuss_tIII_stats)
summary.aov(tuss_tIII_aov8)
TukeyHSD(tuss_tIII_aov8)
```
</div>

##### 7.3.1.3. Tier IV Expression

Calculate TPM-normalized gene count sums for each Tuss sample in each Tier IV category.
```{r}
# Calculate TPM-normalized gene count sums for each Tuss sample in each Tier IV category
tuss_tIV_sum<-tpm_all_exp_ann %>% group_by(Tier_IV) %>% summarise_at(vars("S108382","S108383","S108391","S108392","S108394","S108396"), sum)
tuss_tIV_sum<-tuss_tIV_sum %>% mutate_at(vars(S108382,S108383,S108391,S108392,S108394,S108396), funs(round(., 0)))
tuss_tIV_sum<-as.data.frame(tuss_tIV_sum)
tuss_tIV_sum<-tuss_tIV_sum[order(-tuss_tIV_sum$S108382),]
```

```{r}
# Calculate mean values from the replicate sums
tuss_tIV_sum$meantuss_T0<-apply(tuss_tIV_sum[,2:3], 1, mean)
tuss_tIV_sum$meantuss_T4<-apply(tuss_tIV_sum[,4:5], 1, mean)
tuss_tIV_sum$meantuss_T24<-apply(tuss_tIV_sum[,6:7], 1, mean)
tuss_tIV_sum<-tuss_tIV_sum %>% mutate_at(vars(meantuss_T0,meantuss_T4,meantuss_T24), funs(round(., 0)))

# Calculate sd values from the replicate sums
tuss_tIV_sum$sdtuss_T0<-apply(tuss_tIV_sum[,2:3], 1, sd)
tuss_tIV_sum$sdtuss_T4<-apply(tuss_tIV_sum[,4:5], 1, sd)
tuss_tIV_sum$sdtuss_T24<-apply(tuss_tIV_sum[,6:7], 1, sd)
tuss_tIV_sum<-tuss_tIV_sum %>% mutate_at(vars(sdtuss_T0,sdtuss_T4,sdtuss_T24), funs(round(., 0)))

# Write data to .csv file
write.csv(tuss_tIV_sum, 'Norm.Results/tuss.rna.tIV.mean.sd.csv')

# Subset Mean values
tuss_tIV_mean<-subset(tuss_tIV_sum, select=c(Tier_IV,meantuss_T0,meantuss_T4,meantuss_T24))
names(tuss_tIV_mean)<-c("Tier IV", "Tuss-T0", "Tuss-T4", "Tuss-T24")

# Subset SD values
tuss_tIV_sd<-subset(tuss_tIV_sum, select=c(Tier_IV,sdtuss_T0,sdtuss_T4,sdtuss_T24))
names(tuss_tIV_sd)<-c("Tier IV", "Tuss-T0", "Tuss-T4", "Tuss-T24")

# Remember "Tier IV" as non-numeric values
tuss_tIV_mean_TierIV<-tuss_tIV_mean$`Tier IV`
tuss_tIV_sd_TierIV<-tuss_tIV_sd$`Tier IV`

# Transpose all but first column (Tier IV)
tuss_tIV_mean<-as.data.frame(t(tuss_tIV_mean[,-1]))
colnames(tuss_tIV_mean)<-tuss_tIV_mean_TierIV
tuss_tIV_sd<-as.data.frame(t(tuss_tIV_sd[,-1]))
colnames(tuss_tIV_sd)<-tuss_tIV_sd_TierIV

# Combine mean and sd into single column with ± divider
tuss_tIV_table<-as.data.frame(do.call(cbind, lapply(1:ncol(tuss_tIV_mean), function(i) paste0(tuss_tIV_mean[ , i], " ± ", tuss_tIV_sd[ , i]))))

# Transpose the table back
tuss_tIV_table<-t(tuss_tIV_table)

# Rename columns to Sites
colnames(tuss_tIV_table)<-c("Tussock (T0)", "Tussock (T4)", "Tussock (T24)")

rownames(tuss_tIV_table)<-tuss_tIV_mean_TierIV

#tuss_tIV_table
```

Run statistical tests to determine if significant differences exist between sampling timepoints for each Tier KEGG category. Click on the **Show/Hide** button to see the statistics.

<button class="btn btn-primary" data-toggle="collapse" data-target="#BlockName8"> Show/Hide </button>  
<div id="BlockName8" class="collapse">
```{r}
# Subset the "sum" values for each sample
tuss_tIV_stats<-subset(tuss_tIV_sum, select=c(Tier_IV,S108382,S108383,S108391,S108392,S108394,S108396))

rownames(tuss_tIV_stats) <- tuss_tIV_stats$Tier_IV
tuss_tIV_stats<-as.data.frame(t(tuss_tIV_stats[-1]))

# Add the timepoint column to the sum table
tuss_tIV_stats<-data.frame(Timepoint,tuss_tIV_stats)

# Subset response variables for MANOVA
tuss_tIV_stats$response <- as.matrix(tuss_tIV_stats[, 2:227])

# MANOVA test
tuss_tIV_manova <- manova(response ~ Timepoint, data=tuss_tIV_stats)
summary.aov(tuss_tIV_manova)
```

Run ANOVA for each category of interest (significant in MANOVA)
```{r}
# Carbon Metabolism
tuss_tIV_aov1<-aov(X01200.Carbon.metabolism..PATH.ko01200.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov1)
TukeyHSD(tuss_tIV_aov1)

# Ribosome
tuss_tIV_aov2<-aov(X03010.Ribosome..PATH.ko03010.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov2)
TukeyHSD(tuss_tIV_aov2)

# RNA Degradation
tuss_tIV_aov3<-aov(X03018.RNA.degradation..PATH.ko03018.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov3)
TukeyHSD(tuss_tIV_aov3)

# Fatty Acid Metabolism
tuss_tIV_aov4<-aov(X01212.Fatty.acid.metabolism..PATH.ko01212.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov4)
TukeyHSD(tuss_tIV_aov4)

# Butonaote Metabolism
tuss_tIV_aov5<-aov(X00650.Butanoate.metabolism..PATH.ko00650.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov5)
TukeyHSD(tuss_tIV_aov5)

# Glycolysis Gluconeogenesis
tuss_tIV_aov6<-aov(X00010.Glycolysis...Gluconeogenesis..PATH.ko00010.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov6)
TukeyHSD(tuss_tIV_aov6)

# Caulobacter Cell Cycle
tuss_tIV_aov7<-aov(X04112.Cell.cycle...Caulobacter..PATH.ko04112.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov7)
TukeyHSD(tuss_tIV_aov7)

# Nicotinate and Nicotinamide Metabolism
tuss_tIV_aov8<-aov(X00760.Nicotinate.and.nicotinamide.metabolism..PATH.ko00760.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov8)
TukeyHSD(tuss_tIV_aov8)

# Degradation of Aromatic Compounds
tuss_tIV_aov9<-aov(X01220.Degradation.of.aromatic.compounds..PATH.ko01220.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov9)
TukeyHSD(tuss_tIV_aov9)

# Starch and Sucrose Metabolism
tuss_tIV_aov10<-aov(X00500.Starch.and.sucrose.metabolism..PATH.ko00500.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov10)
TukeyHSD(tuss_tIV_aov10)

# Porphyrin and Chlorophyll Metabolism
tuss_tIV_aov11<-aov(X00860.Porphyrin.and.chlorophyll.metabolism..PATH.ko00860.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov11)
TukeyHSD(tuss_tIV_aov11)

# Pyruvate Metabolism
tuss_tIV_aov12<-aov(X00620.Pyruvate.metabolism..PATH.ko00620.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov12)
TukeyHSD(tuss_tIV_aov12)

# Nitrotoluene Degradation
tuss_tIV_aov13<-aov(X00633.Nitrotoluene.degradation..PATH.ko00633.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov13)
TukeyHSD(tuss_tIV_aov13)

# Thiamine Metabolism
tuss_tIV_aov14<-aov(X00730.Thiamine.metabolism..PATH.ko00730.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov14)
TukeyHSD(tuss_tIV_aov14)

# Pantothenate and CoA Biosynthesis
tuss_tIV_aov15<-aov(X00770.Pantothenate.and.CoA.biosynthesis..PATH.ko00770.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov15)
TukeyHSD(tuss_tIV_aov15)

# Glycerolipid Metabolism
tuss_tIV_aov16<-aov(X00561.Glycerolipid.metabolism..PATH.ko00561.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov16)
TukeyHSD(tuss_tIV_aov16)

# Arginine and Proline Metabolism
tuss_tIV_aov17<-aov(X00330.Arginine.and.proline.metabolism..PATH.ko00330.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov17)
TukeyHSD(tuss_tIV_aov17)

# Sulfur Relay System
tuss_tIV_aov19<-aov(X04122.Sulfur.relay.system..PATH.ko04122.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov19)
TukeyHSD(tuss_tIV_aov19)

# Histidine Metabolism
tuss_tIV_aov20<-aov(X00340.Histidine.metabolism..PATH.ko00340.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov20)
TukeyHSD(tuss_tIV_aov20)

# Steroid Hormone Biosynthesis
tuss_tIV_aov23<-aov(X00140.Steroid.hormone.biosynthesis..PATH.ko00140.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov23)
TukeyHSD(tuss_tIV_aov23)

# Sesquiterpenoid and Triterpenoid Biosynthesis
tuss_tIV_aov24<-aov(X00909.Sesquiterpenoid.and.triterpenoid.biosynthesis..PATH.ko00909.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov24)
TukeyHSD(tuss_tIV_aov24)

# Benzoate Degradation
tuss_tIV_aov25<-aov(X00362.Benzoate.degradation..PATH.ko00362.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov25)
TukeyHSD(tuss_tIV_aov25)

# Chloroalkane and Chloroalkene Degradation
tuss_tIV_aov26<-aov(X00625.Chloroalkane.and.chloroalkene.degradation..PATH.ko00625.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov26)
TukeyHSD(tuss_tIV_aov26)

# Biotin Metabolism
tuss_tIV_aov27<-aov(X00780.Biotin.metabolism..PATH.ko00780.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov27)
TukeyHSD(tuss_tIV_aov27)

# Lysine Biosynthesis
tuss_tIV_aov28<-aov(X00300.Lysine.biosynthesis..PATH.ko00300.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov28)
TukeyHSD(tuss_tIV_aov28)

# MAPK Signaling Pathway
tuss_tIV_aov29<-aov(X04016.MAPK.signaling.pathway...plant..PATH.ko04016.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov29)
TukeyHSD(tuss_tIV_aov29)

# Arginine Biosynthesis
tuss_tIV_aov30<-aov(X00220.Arginine.biosynthesis..PATH.ko00220.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov30)
TukeyHSD(tuss_tIV_aov30)

# Staurosporine Biosynthesis
tuss_tIV_aov32<-aov(X00404.Staurosporine.biosynthesis..PATH.ko00404.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov32)
TukeyHSD(tuss_tIV_aov32)

# Ferroptosis
tuss_tIV_aov33<-aov(X04216.Ferroptosis..PATH.ko04216.~Timepoint,data=tuss_tIV_stats)
summary.aov(tuss_tIV_aov33)
TukeyHSD(tuss_tIV_aov33)
```
</div>

Plot the significant Tuss-MT KEGG tier IV categories by timepoint as a single barchart.
```{r, echo=FALSE}
tuss_tIV_sub_bardata<-read.csv("Plot.Data/tuss.tIV.sub.bardata.csv")
```

```{r}
tuss_tIV_sub_bardata$Sample <- factor(tuss_tIV_sub_bardata$Sample, levels=c("Tuss-MT-T24","Tuss-MT-T4","Tuss-MT-T0"))

tuss_tIV_sub_bardata$Tier.IV <- factor (tuss_tIV_sub_bardata$Tier.IV, levels=c("Cell cycle - Caulobacter (K04112)","RNA degradation (K03018)","Ribosome (K03010)","Degradation of aromatic compounds (K01220)","Fatty acid metabolism (K01212)","Carbon metabolism (K01200)","Nicotinate and nicotinamide metabolism (K00760)","Butanoate metabolism (K006650)","Glycolysis-Gluconeogenesis (K00010)"))

tuss_tIV_sub_MT_barplot<-ggplot(tuss_tIV_sub_bardata, aes(x = Tier.IV, y=Mean, fill=Sample)) + geom_bar(stat = "identity", position = "dodge", color = "black") + geom_errorbar(aes(ymin = Mean - SD, ymax = Mean +SD), width = 0.2, position = position_dodge(0.9)) + coord_flip() + scale_y_continuous(expand = c(0, 0), limits = c(0, 30), position="bottom") + ylab ("Relative Abundance") + theme_classic() + theme(axis.title.y=element_blank(), axis.text.y=element_blank(), plot.margin = margin(10, 20, 10, 10), legend.position = "bottom", legend.title=element_blank(), legend.text=element_text(size=8)) + scale_fill_manual(values=c("darkgreen","green3","palegreen"), guide=guide_legend(reverse=TRUE))

### Saving as .eps file with R Viewer
# Re-size the viewer to the following dimensions before saving as .eps file
# In Console, type "tuss_ws_dna_tIII_pheatmap"
# In Plot Viewer, click "Export" --> "Save as Image"
# Width: 600  Height: 600
# Click "Update Preview" --> save as "EPS" file type --> rename as Fig.4.Bar
```

```{r}
tuss_tIV_sub_MT_barplot
```

Plot the combined Tuss-MG and WS-MG KEGG tier III categories as a single barchart.
```{r, echo=FALSE, eval=FALSE}
tuss_tIV_bardata<-read.csv("Plot.Data/tuss.tIV.bardata.csv")
```

```{r, echo=FALSE, eval=FALSE}
tuss_tIV_bardata$Sample <- factor(tuss_tIV_bardata$Sample, levels=c("Tuss-MT-T0", "Tuss-MT-T4", "Tuss-MT-T24"))

tuss_tIV_bardata$Tier.IV <- factor (tuss_tIV_bardata$Tier.IV, levels=c("Arginine biosynthesis (K00220)","Lysine biosynthesis (K00300)","Arginine and proline metabolism (K00330)","Histidine metabolism (K00340)","Staurosporine biosynthesis (K00404)","Glycolysis-Gluconeogenesis (K00010)","Starch and sucrose metabolism (K00500)","Pyruvate metabolism (K00620)","Butanoate metabolism (K006650)","Steroid hormone biosynthesis (K00140)","Glycerolipid metabolism (K00561)","Thiamine metabolism (K00730)","Nicotinate and nicotinamide metabolism (K00760)","Pantothenate and CoA biosynthesis (K00770)","Biotin metabolism (K00780)","Porphyrin and chlorophyll metabolism (K00860)","Terpenoid backbone biosynthesis (K00900)","Sesquiterpenoid and triterpenoid biosynthesis (K00909)","Carbon metabolism (K01200)","Fatty acid metabolism (K01212)","Degradation of aromatic compounds (K01220)","Benzoate degradation (K00362)","Chloroalkane and chloroalkene degradation (K00625)","Nitrotoluene degradation (K00633)","RNA degradation (K03018)","Sulfur relay system (K04122)","Ribosome (K03010)","Cell cycle - Caulobacter (K04112)","Ferroptosis (K04216)"))

tuss_tIV_MT_barplot<-ggplot(tuss_tIV_bardata, aes(x = Tier.IV, y=Mean, fill=Sample)) + geom_bar(stat = "identity", position = "dodge", color = "black") + geom_errorbar(aes(ymin = Mean - SD, ymax = Mean +SD), width = 0.2, position = position_dodge(0.9)) + coord_flip() + scale_y_continuous(expand = c(0, 0), limits = c(0, 30), position="bottom") + ylab ("Relative Abundance") + theme_classic() + theme(axis.title.y=element_blank(), plot.margin = margin(10, 20, 10, 10), legend.position = "bottom", legend.title=element_blank(), legend.text=element_text(size=8)) + scale_fill_manual(values=c("palegreen", "green3", "darkgreen"), guide=guide_legend(reverse=TRUE))

### Saving as .eps file with R Viewer
# Re-size the viewer to the following dimensions before saving as .eps file
# In Console, type "tuss_ws_dna_tIII_pheatmap"
# In Plot Viewer, click "Export" --> "Save as Image"
# Width: 500  Height: 700
# Click "Update Preview" --> save as "EPS" file type --> rename as Fig.2.Bar
```

```{r, echo=FALSE, eval=FALSE}
tuss_tIV_MT_barplot
```

Heatmap for Tier IV categories by sampling timepoints
```{r}
# Create copy of "tuss_tIV_sum" object to prep for heatmap
tuss_tIV_sub_heatmap<-tuss_tIV_sum

# Keep the first 7 columns
tuss_tIV_sub_heatmap <- tuss_tIV_sub_heatmap[,c(1,8,9,10)]  #This code uses the means of timepoint

# Keep only those categories with significant differences (from ANOVA)

tuss_tIV_sub_heatmap <- subset(tuss_tIV_sub_heatmap, Tier_IV=="00010 Glycolysis / Gluconeogenesis [PATH:ko00010]" | Tier_IV=="00650 Butanoate metabolism [PATH:ko00650]" | Tier_IV=="00760 Nicotinate and nicotinamide metabolism [PATH:ko00760]" | Tier_IV=="01200 Carbon metabolism [PATH:ko01200]" | Tier_IV=="01212 Fatty acid metabolism [PATH:ko01212]" | Tier_IV=="01220 Degradation of aromatic compounds [PATH:ko01220]" | Tier_IV=="03018 RNA degradation [PATH:ko03018]" | Tier_IV=="03010 Ribosome [PATH:ko03010]" | Tier_IV=="04112 Cell cycle - Caulobacter [PATH:ko04112]", select=Tier_IV:meantuss_T24)

colnames(tuss_tIV_sub_heatmap)<-c("Tier_IV","Tuss-MT-T0","Tuss-MT-T4","Tuss-MT-T24")

# Create a vector with categores in the desired order
tuss_tIV_sub_x <- c("00010 Glycolysis / Gluconeogenesis [PATH:ko00010]",
"00650 Butanoate metabolism [PATH:ko00650]",
"00760 Nicotinate and nicotinamide metabolism [PATH:ko00760]",
"01200 Carbon metabolism [PATH:ko01200]",
"01212 Fatty acid metabolism [PATH:ko01212]",
"01220 Degradation of aromatic compounds [PATH:ko01220]",
"03010 Ribosome [PATH:ko03010]",
"03018 RNA degradation [PATH:ko03018]",
"04112 Cell cycle - Caulobacter [PATH:ko04112]")

# Re-sort the data in the desired order
tuss_tIV_sub_heatmap<-tuss_tIV_sub_heatmap %>%
  slice(match(tuss_tIV_sub_x, Tier_IV))

# Convert the first column (Tier categories) into rownames
rownames(tuss_tIV_sub_heatmap) <- tuss_tIV_sub_heatmap$Tier_IV
tuss_tIV_sub_heatmap<-as.data.frame(tuss_tIV_sub_heatmap[-1])

# Rename columns to sample IDs
colnames(tuss_tIV_sub_heatmap)<-c("Tuss-MT-T0","Tuss-MT-T4","Tuss-MT-T24")

# Convert dataframe into a matrix for heatmap
tuss_tIV_sub_heatmap<-as.matrix(tuss_tIV_sub_heatmap)

# Scale matrix values to generate Z-scores
tuss_tIV_sub_heatmap<-scale(t(tuss_tIV_sub_heatmap))

tuss_tIV_sub_heatmap<-t(tuss_tIV_sub_heatmap)

# Specify RColorBrewer custom color palette
col <- colorRampPalette(brewer.pal(10, "RdYlBu"))(256)

### Saving as .eps file with R Viewer
# Re-size the viewer to the following dimensions before saving as .eps file
# In Console, type "tuss_tIV_pheatmap"
# In Plot Viewer, click "Export" --> "Save as Image"
# Width: 600  Height: 600
# Click "Update Preview" --> save as "EPS" file type --> rename as Fig.2.2.Tuss.MT.Heatmap

# Pheatmap
tuss_tIV_sub_pheatmap<-pheatmap(tuss_tIV_sub_heatmap, treeheight_row = 0, treeheight_col = 0,cluster_rows = FALSE, cluster_cols = FALSE)
```



```{r, eval=FALSE, echo=FALSE}
# Create copy of "tuss_tIV_sum" object to prep for heatmap
tuss_tIV_heatmap<-tuss_tIV_sum

# Keep the first 7 columns
tuss_tIV_heatmap <- tuss_tIV_heatmap[,c(1,8,9,10)]  #This code uses the means of timepoint

# Keep only those categories with significant differences (from ANOVA)

tuss_tIV_heatmap <- subset(tuss_tIV_heatmap, Tier_IV=="00220 Arginine biosynthesis [PATH:ko00220]" | Tier_IV=="00300 Lysine biosynthesis [PATH:ko00300]" | Tier_IV=="00330 Arginine and proline metabolism [PATH:ko00330]" | Tier_IV=="00340 Histidine metabolism [PATH:ko00340]" | Tier_IV=="00404 Staurosporine biosynthesis [PATH:ko00404]" | Tier_IV=="00010 Glycolysis / Gluconeogenesis [PATH:ko00010]" | Tier_IV=="00500 Starch and sucrose metabolism [PATH:ko00500]" | Tier_IV=="00620 Pyruvate metabolism [PATH:ko00620]" | Tier_IV=="00650 Butanoate metabolism [PATH:ko00650]" | Tier_IV=="00140 Steroid hormone biosynthesis [PATH:ko00140]" | Tier_IV=="00561 Glycerolipid metabolism [PATH:ko00561]" | Tier_IV=="00730 Thiamine metabolism [PATH:ko00730]" | Tier_IV=="00760 Nicotinate and nicotinamide metabolism [PATH:ko00760]" | Tier_IV=="00770 Pantothenate and CoA biosynthesis [PATH:ko00770]" | Tier_IV=="00780 Biotin metabolism [PATH:ko00780]" | Tier_IV=="00860 Porphyrin and chlorophyll metabolism [PATH:ko00860]" | Tier_IV=="00900 Terpenoid backbone biosynthesis [PATH:ko00900]" | Tier_IV=="00909 Sesquiterpenoid and triterpenoid biosynthesis [PATH:ko00909]" | Tier_IV=="01200 Carbon metabolism [PATH:ko01200]" | Tier_IV=="01212 Fatty acid metabolism [PATH:ko01212]" | Tier_IV=="01220 Degradation of aromatic compounds [PATH:ko01220]" | Tier_IV=="00362 Benzoate degradation [PATH:ko00362]" | Tier_IV=="00625 Chloroalkane and chloroalkene degradation [PATH:ko00625]" | Tier_IV=="00633 Nitrotoluene degradation [PATH:ko00633]" | Tier_IV=="03018 RNA degradation [PATH:ko03018]" | Tier_IV=="04122 Sulfur relay system [PATH:ko04122]" | Tier_IV=="03010 Ribosome [PATH:ko03010]" | Tier_IV=="04112 Cell cycle - Caulobacter [PATH:ko04112]" | Tier_IV=="04216 Ferroptosis [PATH:ko04216]", select=Tier_IV:meantuss_T24)

colnames(tuss_tIV_heatmap)<-c("Tier_IV","Tuss-MT-T0","Tuss-MT-T4","Tuss-MT-T24")

# Create a vector with categores in the desired order
tuss_tIV_x <- c("00220 Arginine biosynthesis [PATH:ko00220]",
"00300 Lysine biosynthesis [PATH:ko00300]",
"00330 Arginine and proline metabolism [PATH:ko00330]",
"00340 Histidine metabolism [PATH:ko00340]",
"00404 Staurosporine biosynthesis [PATH:ko00404]",
"00010 Glycolysis / Gluconeogenesis [PATH:ko00010]",
"00500 Starch and sucrose metabolism [PATH:ko00500]",
"00620 Pyruvate metabolism [PATH:ko00620]",
"00650 Butanoate metabolism [PATH:ko00650]",
"00140 Steroid hormone biosynthesis [PATH:ko00140]",
"00561 Glycerolipid metabolism [PATH:ko00561]",
"00730 Thiamine metabolism [PATH:ko00730]",
"00760 Nicotinate and nicotinamide metabolism [PATH:ko00760]",
"00770 Pantothenate and CoA biosynthesis [PATH:ko00770]",
"00780 Biotin metabolism [PATH:ko00780]",
"00860 Porphyrin and chlorophyll metabolism [PATH:ko00860]",
"00900 Terpenoid backbone biosynthesis [PATH:ko00900]",
"00909 Sesquiterpenoid and triterpenoid biosynthesis [PATH:ko00909]",
"01200 Carbon metabolism [PATH:ko01200]",
"01212 Fatty acid metabolism [PATH:ko01212]",
"01220 Degradation of aromatic compounds [PATH:ko01220]",
"00362 Benzoate degradation [PATH:ko00362]",
"00625 Chloroalkane and chloroalkene degradation [PATH:ko00625]",
"00633 Nitrotoluene degradation [PATH:ko00633]",
"03018 RNA degradation [PATH:ko03018]",
"04122 Sulfur relay system [PATH:ko04122]",
"03010 Ribosome [PATH:ko03010]",
"04112 Cell cycle - Caulobacter [PATH:ko04112]",
"04216 Ferroptosis [PATH:ko04216]")

# Re-sort the data in the desired order
tuss_tIV_heatmap<-tuss_tIV_heatmap %>%
  slice(match(tuss_tIV_x, Tier_IV))

# Convert the first column (Tier categories) into rownames
rownames(tuss_tIV_heatmap) <- tuss_tIV_heatmap$Tier_IV
tuss_tIV_heatmap<-as.data.frame(tuss_tIV_heatmap[-1])

# Rename columns to sample IDs
colnames(tuss_tIV_heatmap)<-c("Tuss-MT-T0","Tuss-MT-T4","Tuss-MT-T24")

# Convert dataframe into a matrix for heatmap
tuss_tIV_heatmap<-as.matrix(tuss_tIV_heatmap)

# Scale matrix values to generate Z-scores
tuss_tIV_heatmap<-scale(t(tuss_tIV_heatmap))

tuss_tIV_heatmap<-t(tuss_tIV_heatmap)

# Specify RColorBrewer custom color palette
col <- colorRampPalette(brewer.pal(10, "RdYlBu"))(256)

### Saving as .eps file with R Viewer
# Re-size the viewer to the following dimensions before saving as .eps file
# In Console, type "tuss_tIV_pheatmap"
# In Plot Viewer, click "Export" --> "Save as Image"
# Width: 600  Height: 650
# Click "Update Preview" --> save as "EPS" file type --> rename as Fig.2.2.Tuss.MT.Heatmap

# Pheatmap
tuss_tIV_pheatmap<-pheatmap(tuss_tIV_heatmap, treeheight_row = 0, treeheight_col = 0,cluster_rows = FALSE, cluster_cols = FALSE)
```

#### 7.3.2. Wet Sedge Tundra

Look at differences in gene expression patterns for multiple Tier levels within the Wet Sedge tundra between sampling time points.

##### 7.3.2.1. Tier II Expression

Calculate TPM-normalized gene count sums for each WS sample in each Tier II category.
```{r}
# Calculate TPM-normalized gene count sums for each WS sample in each Tier II category
ws_tII_sum<-tpm_all_exp_ann %>% group_by(Tier_II) %>% summarise_at(vars("S108379","S108381","S108385","S108386","S108388","S108390"), sum)

ws_tII_sum<-ws_tII_sum %>% mutate_at(vars(S108379,S108381,S108385,S108386,S108388,S108390), funs(round(., 0)))

ws_tII_sum<-as.data.frame(ws_tII_sum)

ws_tII_sum<-ws_tII_sum[order(-ws_tII_sum$S108379),]
```

```{r echo=FALSE}
kable(ws_tII_sum, caption = "Wet Sedge Tier II KEGG category summary by sample.", format.args = list(big.mark=",")) %>% kable_styling(bootstrap_options = c("striped","hover","condensed"))
```

Heatmap for Tier II categories by sampling timepoints
```{r}
# Create copy of "ws_tII_sum" object to prep for heatmap
ws_tII_heatmap<-ws_tII_sum

# Convert the first column (Tier categories) into rownames
rownames(ws_tII_heatmap) <- ws_tII_heatmap$Tier_II
ws_tII_heatmap<-as.data.frame(ws_tII_heatmap[-1])

# Rename columns to sample IDs
colnames(ws_tII_heatmap)<-c("WS1-T0","WS3-T0","WS1-T4","WS2-T4","WS1-T24","WS3-T24")

# Remove rows for "Organismal Systems" and "Human Diseases"
#ws_tII_heatmap <- ws_tII_heatmap[-c(5,6),]

# Convert dataframe into a matrix for heatmap
ws_tII_heatmap<-as.matrix(ws_tII_heatmap)

# Scale matrix values to generate Z-scores
ws_tII_heatmap<-scale(t(ws_tII_heatmap))

ws_tII_heatmap<-t(ws_tII_heatmap)

# Specify RColorBrewer custom color palette
col <- colorRampPalette(brewer.pal(10, "RdYlBu"))(256)

# Pheatmap
pheatmap(ws_tII_heatmap, treeheight_row = 0, treeheight_col = 0,cluster_rows = FALSE, cluster_cols = FALSE)
```

Calculate mean values from the replicate sums.
```{r}
# Calculate mean values from the replicate sums
ws_tII_sum$meanWS_T0<-apply(ws_tII_sum[,2:3], 1, mean)
ws_tII_sum$meanWS_T4<-apply(ws_tII_sum[,4:5], 1, mean)
ws_tII_sum$meanWS_T24<-apply(ws_tII_sum[,6:7], 1, mean)
ws_tII_sum<-ws_tII_sum %>% mutate_at(vars(meanWS_T0,meanWS_T4,meanWS_T24), funs(round(., 0)))

# Calculate sd values from the replicate sums
ws_tII_sum$sdWS_T0<-apply(ws_tII_sum[,2:3], 1, sd)
ws_tII_sum$sdWS_T4<-apply(ws_tII_sum[,4:5], 1, sd)
ws_tII_sum$sdWS_T24<-apply(ws_tII_sum[,6:7], 1, sd)
ws_tII_sum<-ws_tII_sum %>% mutate_at(vars(sdWS_T0,sdWS_T4,sdWS_T24), funs(round(., 0)))

# Subset Mean values
ws_tII_mean<-subset(ws_tII_sum, select=c(Tier_II,meanWS_T0,meanWS_T4,meanWS_T24))
names(ws_tII_mean)<-c("Tier II", "WS - T0", "WS - T4", "WS - T24")

# Subset SD values
ws_tII_sd<-subset(ws_tII_sum, select=c(Tier_II,sdWS_T0,sdWS_T4,sdWS_T24))
names(ws_tII_sd)<-c("Tier II", "WS - T0", "WS - T4", "WS - T24")

# Remember "Tier II" as non-numeric values
ws_tII_mean_TierII<-ws_tII_mean$`Tier II`
ws_tII_sd_TierII<-ws_tII_sd$`Tier II`

# Transpose all but first column (Tier II)
ws_tII_mean<-as.data.frame(t(ws_tII_mean[,-1]))
colnames(ws_tII_mean)<-ws_tII_mean_TierII
ws_tII_sd<-as.data.frame(t(ws_tII_sd[,-1]))
colnames(ws_tII_sd)<-ws_tII_sd_TierII

# Combine mean and sd into single column with ± divider
ws_tII_table<-as.data.frame(do.call(cbind, lapply(1:ncol(ws_tII_mean), function(i) paste0(ws_tII_mean[ , i], " ± ", ws_tII_sd[ , i]))))

# Transpose the table back
ws_tII_table<-t(ws_tII_table)

# Rename columns to Sites
colnames(ws_tII_table)<-c("Wet Sedge (T0)", "Wet Sedge (T4)", "Wet Sedge (T24)")

rownames(ws_tII_table)<-ws_tII_mean_TierII

kable(ws_tII_table, caption = "Wet Sedge Tier II KEGG category averages (± standard deviation) by sample.", format.args = list(big.mark=","), align = "l") %>% kable_styling(bootstrap_options = c("striped","hover","condensed"))
```

Run statistical tests to determine if significant differences exist between sampling timepoints for each Tier KEGG category. Click on the **Show/Hide** button to see the statistics.

<button class="btn btn-primary" data-toggle="collapse" data-target="#BlockName9"> Show/Hide </button>  
<div id="BlockName9" class="collapse">
```{r}
# Subset the "sum" values for each sample
ws_tII_stats<-subset(ws_tII_sum, select=c(Tier_II,S108379,S108381,S108385,S108386,S108388,S108390))

rownames(ws_tII_stats) <- ws_tII_stats$Tier_II
ws_tII_stats<-as.data.frame(t(ws_tII_stats[-1]))

# Add the timepoint column to the sum table
ws_tII_stats<-data.frame(Timepoint,ws_tII_stats)

# Subset response variables for MANOVA
ws_tII_stats$response <- as.matrix(ws_tII_stats[, 2:5])

# MANOVA test
ws_tII_manova <- manova(response ~ Timepoint, data=ws_tII_stats)
summary.aov(ws_tII_manova)
```
</div>

Plot a barchart for Tier II categories by sampling timepoints.
```{r echo=FALSE}
ws_tII_bardata<-read.csv("Plot.Data/ws_tII_barplot.csv")
```

```{r}
# Place Tier II categories in the preferred order for plotting
ws_tII_bardata$Tier.II <- factor(ws_tII_bardata$Tier.II,levels = c("Metabolism", "Genetic Information Processing", "Environmental Information Processing", "Cellular Processes"))

ws_tII_bardata$Sample <- factor(ws_tII_bardata$Sample,levels=c("WS-T0","WS-T4","WS-T24"))

ws_tII_barplot<-ggplot(ws_tII_bardata, aes(x = Tier.II, y = Mean, fill = Sample)) + geom_bar(stat = "identity", position = "dodge", color="black") + geom_errorbar(aes(ymin = Mean - SD, ymax = Mean + SD), width = 0.2, position = position_dodge(0.9)) + ylab(expression(atop("KEGG Tier II Categories", paste("Transcript Counts (TPM)")))) + theme_classic() + theme(axis.text=element_text(size=10), axis.title=element_text(size=12), axis.title.x=element_blank()) + theme(legend.position = "bottom", legend.title=element_blank(), legend.text=element_text(size=8), panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.border = element_blank()) + scale_size(guide=FALSE) + scale_fill_manual(values=c("lightskyblue1", "dodgerblue1", "midnightblue")) + scale_x_discrete(labels = function(Tier.II) str_wrap(Tier.II, width = 10)) + scale_y_continuous(expand = c(0, 0), limits = c(0, 800000)) + annotate(geom="text", x=1.0, y=700000, label="italic(N.S.)", parse=TRUE) + annotate(geom="text", x=2.0, y=400000, label="italic(N.S.)", parse=TRUE) + annotate(geom="text", x=3.0, y=250000, label="italic(N.S.)", parse=TRUE) + annotate(geom="text", x=4.0, y=150000, label="italic(N.S.)", parse=TRUE)


```

```{r}
ws_tII_barplot
```

```{r echo=FALSE}
ws_mg_mt_tII_bardata<-read.csv("Plot.Data/ws_mg_mt_tII_barplot.csv")
```

```{r}
# Place Tier II categories in the preferred order for plotting
ws_mg_mt_tII_bardata$Tier.II <- factor(ws_mg_mt_tII_bardata$Tier.II,levels = c("Metabolism", "Genetic Information Processing", "Environmental Information Processing", "Cellular Processes"))

# Place Samples in the preferred order for plotting
ws_mg_mt_tII_bardata$Sample <- factor(ws_mg_mt_tII_bardata$Sample,levels=c("WS-MG","WS-MT-T0","WS-MT-T4","WS-MT-T24"))

ws_mg_mt_tII_barplot<-ggplot(ws_mg_mt_tII_bardata, aes(x = Tier.II, y = Mean, fill = Sample)) + geom_bar(stat = "identity", position = "dodge", color="black") + geom_errorbar(aes(ymin = Mean - SD, ymax = Mean + SD), width = 0.2, position = position_dodge(0.9)) + ylab(expression(atop("KEGG Tier II Categories", paste("Gene Counts")))) + theme_classic() + theme(axis.text=element_text(size=10), axis.title=element_text(size=12), axis.title.x=element_blank()) + theme(legend.position = "bottom", legend.title=element_blank(), legend.text=element_text(size=8), panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.border = element_blank()) + scale_size(guide=FALSE) + scale_fill_manual(values=c("black", "lightskyblue1", "dodgerblue1", "midnightblue")) + scale_x_discrete(labels = function(Tier.II) str_wrap(Tier.II, width = 10)) + scale_y_continuous(expand = c(0, 0), limits = c(0, 800000)) + annotate(geom="text", x=1.12, y=700000, label="italic(N.S.)", parse=TRUE) + annotate(geom="text", x=2.12, y=400000, label="italic(N.S.)", parse=TRUE) + annotate(geom="text", x=3.12, y=250000, label="italic(N.S.)", parse=TRUE) + annotate(geom="text", x=4.12, y=150000, label="italic(N.S.)", parse=TRUE)
```

```{r}
ws_mg_mt_tII_barplot
```

```{r echo=FALSE, eval=FALSE}
# Save the Ribosome Taxonomy plot as an individual figure
setEPS()
postscript("Fig.2.MT.Taxa.KEGG.eps", width=10.0, height = 10.0)
div.plot<-plot_grid(taxa.tuss.mean.ribo.plot, taxa.ws.mean.ribo.plot,labels=c('A','B'),rel_widths = c(4.5,6))
div.plot2<-plot_grid(tuss_mg_mt_tII_barplot, ws_mg_mt_tII_barplot, labels=c('C','D'), rel_widths = c(5,5))
plot_grid(div.plot,div.plot2,ncol = 1)
dev.off()
```

```{r echo=FALSE, eval=FALSE}
# Plot the TUSS and WS MG-MT-Tier-II-Barplots
setEPS()
postscript("Fig.2.1.1.Tuss.Taxa.eps", width=5.0, height = 5.0)
ggarrange(taxa.tuss.mean.ribo.plot, ncol = 1, nrow = 1)
dev.off()
```

```{r echo=FALSE, eval=FALSE}
# Plot the TUSS and WS MG-MT-Tier-II-Barplots
setEPS()
postscript("Fig.2.1.2.WS.Taxa.eps", width=7.0, height = 5.0)
ggarrange(taxa.ws.mean.ribo.plot, ncol = 1, nrow = 1)
dev.off()
```

```{r echo=FALSE, eval=FALSE}
# Plot the TUSS and WS MG-MT-Tier-II-Barplots
setEPS()
postscript("Fig.2.2.MT.KEGG.eps", width=10.0, height = 5.0)
ggarrange(tuss_mg_mt_tII_barplot, ws_mg_mt_tII_barplot, ncol = 2, nrow = 1, labels=c('A','B'), align="v")
dev.off()
```

##### 7.3.2.2. Tier III Expression

Calculate TPM-normalized gene count sums for each WS sample in each Tier III category.
```{r}
# Calculate TPM-normalized gene count sums for each WS sample in each Tier III category
ws_tIII_sum<-tpm_all_exp_ann %>% group_by(Tier_III) %>% summarise_at(vars("S108379","S108381","S108385","S108386","S108388","S108390"), sum)
ws_tIII_sum<-ws_tIII_sum %>% mutate_at(vars(S108379,S108381,S108385,S108386,S108388,S108390), funs(round(., 0)))
ws_tIII_sum<-as.data.frame(ws_tIII_sum)
ws_tIII_sum<-ws_tIII_sum[order(-ws_tIII_sum$S108379),]
```

```{r}
# Calculate mean values from the replicate sums
ws_tIII_sum$meanWS_T0<-apply(ws_tIII_sum[,2:3], 1, mean)
ws_tIII_sum$meanWS_T4<-apply(ws_tIII_sum[,4:5], 1, mean)
ws_tIII_sum$meanWS_T24<-apply(ws_tIII_sum[,6:7], 1, mean)
ws_tIII_sum<-ws_tIII_sum %>% mutate_at(vars(meanWS_T0,meanWS_T4,meanWS_T24), funs(round(., 0)))

# Calculate sd values from the replicate sums
ws_tIII_sum$sdWS_T0<-apply(ws_tIII_sum[,2:3], 1, sd)
ws_tIII_sum$sdWS_T4<-apply(ws_tIII_sum[,4:5], 1, sd)
ws_tIII_sum$sdWS_T24<-apply(ws_tIII_sum[,6:7], 1, sd)
ws_tIII_sum<-ws_tIII_sum %>% mutate_at(vars(sdWS_T0,sdWS_T4,sdWS_T24), funs(round(., 0)))

# Write data to .csv file
write.csv(ws_tIII_sum, 'Norm.Results/ws.rna.tIII.mean.sd.csv')

# Subset Mean values
ws_tIII_mean<-subset(ws_tIII_sum, select=c(Tier_III,meanWS_T0,meanWS_T4,meanWS_T24))
names(ws_tIII_mean)<-c("Tier III", "WS - T0", "WS - T4", "WS - T24")

# Subset SD values
ws_tIII_sd<-subset(ws_tIII_sum, select=c(Tier_III,sdWS_T0,sdWS_T4,sdWS_T24))
names(ws_tIII_sd)<-c("Tier III", "WS - T0", "WS - T4", "WS - T24")

# Remember "Tier III" as non-numeric values
ws_tIII_mean_TierIII<-ws_tIII_mean$`Tier III`
ws_tIII_sd_TierIII<-ws_tIII_sd$`Tier III`

# Transpose all but first column (Tier II)
ws_tIII_mean<-as.data.frame(t(ws_tIII_mean[,-1]))
colnames(ws_tIII_mean)<-ws_tIII_mean_TierIII
ws_tIII_sd<-as.data.frame(t(ws_tIII_sd[,-1]))
colnames(ws_tIII_sd)<-ws_tIII_sd_TierIII

# Combine mean and sd into single column with ± divider
ws_tIII_table<-as.data.frame(do.call(cbind, lapply(1:ncol(ws_tIII_mean), function(i) paste0(ws_tIII_mean[ , i], " ± ", ws_tIII_sd[ , i]))))

# Transpose the table back
ws_tIII_table<-t(ws_tIII_table)

# Rename columns to Sites
colnames(ws_tIII_table)<-c("Wet Sedge (T0)", "Wet Sedge (T4)", "Wet Sedge (T24)")

rownames(ws_tIII_table)<-ws_tIII_mean_TierIII
```

Run statistical tests to determine if significant differences exist between sampling timepoints for each Tier KEGG category. Click on the **Show/Hide** button to see the statistics.

<button class="btn btn-primary" data-toggle="collapse" data-target="#BlockName10"> Show/Hide </button>  
<div id="BlockName10" class="collapse">
```{r}
# Subset the "sum" values for each sample
ws_tIII_stats<-subset(ws_tIII_sum, select=c(Tier_III,S108379,S108381,S108385,S108386,S108388,S108390))

rownames(ws_tIII_stats) <- ws_tIII_stats$Tier_III
ws_tIII_stats<-as.data.frame(t(ws_tIII_stats[-1]))

# Add the timepoint column to the sum table
ws_tIII_stats<-data.frame(Timepoint,ws_tIII_stats)

# Subset response variables for MANOVA
ws_tIII_stats$response <- as.matrix(ws_tIII_stats[, 2:26])

# MANOVA test
ws_tIII_manova <- manova(response ~ Timepoint, data=ws_tIII_stats)
summary.aov(ws_tIII_manova)
```

Run ANOVA for each category of interest (significant in MANOVA)
```{r}
# Carbohydrate.metabolism.
ws_tIII_aov_carbs<-aov(Carbohydrate.metabolism.~Timepoint,data=ws_tIII_stats)
TukeyHSD(ws_tIII_aov_carbs)

# Signaling Molecules and Interactions
ws_tIII_aov_trans<-aov(Signaling.molecules.and.interaction.~Timepoint,data=ws_tIII_stats)
TukeyHSD(ws_tIII_aov_trans)
```
</div>

##### 7.3.2.3. Tier IV Expression
Calculate TPM-normalized gene count sums for each WS sample in each Tier III category.
```{r}
# Calculate TPM-normalized gene count sums for each WS sample in each Tier III category
ws_tIV_sum<-tpm_all_exp_ann %>% group_by(Tier_IV) %>% summarise_at(vars("S108379","S108381","S108385","S108386","S108388","S108390"), sum)
ws_tIV_sum<-ws_tIV_sum %>% mutate_at(vars(S108379,S108381,S108385,S108386,S108388,S108390), funs(round(., 0)))
ws_tIV_sum<-as.data.frame(ws_tIV_sum)
ws_tIV_sum<-ws_tIV_sum[order(-ws_tIV_sum$S108379),]
```

```{r}
# Calculate mean values from the replicate sums
ws_tIV_sum$meanWS_T0<-apply(ws_tIV_sum[,2:3], 1, mean)
ws_tIV_sum$meanWS_T4<-apply(ws_tIV_sum[,4:5], 1, mean)
ws_tIV_sum$meanWS_T24<-apply(ws_tIV_sum[,6:7], 1, mean)
ws_tIV_sum<-ws_tIV_sum %>% mutate_at(vars(meanWS_T0,meanWS_T4,meanWS_T24), funs(round(., 0)))

# Calculate sd values from the replicate sums
ws_tIV_sum$sdWS_T0<-apply(ws_tIV_sum[,2:3], 1, sd)
ws_tIV_sum$sdWS_T4<-apply(ws_tIV_sum[,4:5], 1, sd)
ws_tIV_sum$sdWS_T24<-apply(ws_tIV_sum[,6:7], 1, sd)
ws_tIV_sum<-ws_tIV_sum %>% mutate_at(vars(sdWS_T0,sdWS_T4,sdWS_T24), funs(round(., 0)))

# Write data to .csv file
write.csv(ws_tIV_sum, 'Norm.Results/ws.rna.tIV.mean.sd.csv')

# Subset Mean values
ws_tIV_mean<-subset(ws_tIV_sum, select=c(Tier_IV,meanWS_T0,meanWS_T4,meanWS_T24))
names(ws_tIV_mean)<-c("Tier IV", "WS - T0", "WS - T4", "WS - T24")

# Subset SD values
ws_tIV_sd<-subset(ws_tIV_sum, select=c(Tier_IV,sdWS_T0,sdWS_T4,sdWS_T24))
names(ws_tIV_sd)<-c("Tier IV", "WS - T0", "WS - T4", "WS - T24")

# Remember "Tier IV" as non-numeric values
ws_tIV_mean_TierIV<-ws_tIV_mean$`Tier IV`
ws_tIV_sd_TierIV<-ws_tIV_sd$`Tier IV`

# Transpose all but first column (Tier IV)
ws_tIV_mean<-as.data.frame(t(ws_tIV_mean[,-1]))
colnames(ws_tIV_mean)<-ws_tIV_mean_TierIV
ws_tIV_sd<-as.data.frame(t(ws_tIV_sd[,-1]))
colnames(ws_tIV_sd)<-ws_tIV_sd_TierIV

# Combine mean and sd into single column with ± divider
ws_tIV_table<-as.data.frame(do.call(cbind, lapply(1:ncol(ws_tIV_mean), function(i) paste0(ws_tIV_mean[ , i], " ± ", ws_tIV_sd[ , i]))))

# Transpose the table back
ws_tIV_table<-t(ws_tIV_table)

# Rename columns to Sites
colnames(ws_tIV_table)<-c("Wet Sedge (T0)", "Wet Sedge (T4)", "Wet Sedge (T24)")

rownames(ws_tIV_table)<-ws_tIV_mean_TierIV

#ws_tIV_table
```

Run statistical tests to determine if significant differences exist between sampling timepoints for each Tier KEGG category. Click on the **Show/Hide** button to see the statistics.

<button class="btn btn-primary" data-toggle="collapse" data-target="#BlockName11"> Show/Hide </button>  
<div id="BlockName11" class="collapse">
```{r}
# Subset the "sum" values for each sample
ws_tIV_stats<-subset(ws_tIV_sum, select=c(Tier_IV,S108379,S108381,S108385,S108386,S108388,S108390))

rownames(ws_tIV_stats) <- ws_tIV_stats$Tier_IV
ws_tIV_stats<-as.data.frame(t(ws_tIV_stats[-1]))

# Add the timepoint column to the sum table
ws_tIV_stats<-data.frame(Timepoint,ws_tIV_stats)

# Subset response variables for MANOVA
ws_tIV_stats$response <- as.matrix(ws_tIV_stats[, 2:227])

# MANOVA test
ws_tIV_manova <- manova(response ~ Timepoint, data=ws_tIV_stats)
summary.aov(ws_tIV_manova)
```

Run ANOVA for each category of interest (significant in MANOVA)
```{r eval=FALSE}
## Glycolysis
ws_tIV_stat1<-aov(X00010.Glycolysis...Gluconeogenesis..PATH.ko00010.~Timepoint,data=ws_tIV_stats)
summary.aov(ws_tIV_stat1)
TukeyHSD(ws_tIV_stat1)

## Fatty Acid Metabolism
ws_tIV_stat2<-aov(X01212.Fatty.acid.metabolism..PATH.ko01212.~Timepoint,data=ws_tIV_stats)
summary.aov(ws_tIV_stat2)
TukeyHSD(ws_tIV_stat2)

## Pentose and Glucuronate Interconversions
ws_tIV_stat3<-aov(X00040.Pentose.and.glucuronate.interconversions..PATH.ko00040.~Timepoint,data=ws_tIV_stats)
summary.aov(ws_tIV_stat3)
TukeyHSD(ws_tIV_stat3)

## Steroid Degradation
ws_tIV_stat4<-aov(X00984.Steroid.degradation..PATH.ko00984.~Timepoint,data=ws_tIV_stats)
summary.aov(ws_tIV_stat4)
TukeyHSD(ws_tIV_stat4)

## Bacterial Chemotaxis
ws_tIV_stat5<-aov(X02030.Bacterial.chemotaxis..PATH.ko02030.~Timepoint,data=ws_tIV_stats)
summary.aov(ws_tIV_stat5)
TukeyHSD(ws_tIV_stat5)

## Phosphonate and Phoshinate Metabolism
ws_tIV_stat6<-aov(X00440.Phosphonate.and.phosphinate.metabolism..PATH.ko00440.~Timepoint,data=ws_tIV_stats)
summary.aov(ws_tIV_stat6)
TukeyHSD(ws_tIV_stat6)

## Carbon Fixation Pathways in Prokaryotes
ws_tIV_stat7<-aov(X00720.Carbon.fixation.pathways.in.prokaryotes..PATH.ko00720.~Timepoint,data=ws_tIV_stats)
summary.aov(ws_tIV_stat7)
TukeyHSD(ws_tIV_stat7)

## Mucin Type O Glycan Biosynthesis
ws_tIV_stat8<-aov(X00512.Mucin.type.O.glycan.biosynthesis..PATH.ko00512.~Timepoint,data=ws_tIV_stats)
summary.aov(ws_tIV_stat8)
TukeyHSD(ws_tIV_stat8)
```
</div>

### 7.4. Differential Expression

Determine which genes were differentially expressed between experimental treatments using the Bioconductor package **edgeR** for differential expression analysis on TPM-normalized read counts.  The **edgeR** package implements statistical methods based on generalized linear models (glms), suitable for multifactor experiments of any complexity.  A particular feature of **edgeR** functionality are empirical Bayes methods that permit the estimation of gene-specific biological variation, even for experiments with minimal levels of biological replication.

#### 7.4.1. DGE Objective

We have two objectives for our differential expression analysis. 

1. Detect genes that are differentially expressed between experimental timepoints, adjusting for any differences between mesocosms that were sampled, within Wet Sedge or Tussock mesocosms.
    + *Determine how gene expression changed between anoxic (T0) to oxic (T4) to mixed-redox (T24) timepoints, specific to the Wet Sedge ecosystem or the Tussock ecosystem*
2. Detect genes that are differentially expressed between ecosystems at similar timepoints (similar redox conditions).
    + *Determine if the same (or different) genes were expressed between ecosystems when the soil redox conditions were similar*

It is possible (and likely) that there are unique genecall ID's that share identical annotations to KO and Taxonomy, but are still treated as unique entries with separate count values that are factored into DE analysis improperly.
    + *Here, we aggregate (sum) genecall ID counts by shared KO and Taxonomy to reduce duplicates in the dataset and improve the DE analysis results to be biologically relevant.*

#### 7.4.2. DGE Within Tundra

Our first set of DE analyses focus on determining significant differentially expressed genes between sampling timepoints within Tussock or Wet Sedge mesocosms.

##### 7.4.2.1. Tuss T4vsT0

Subset data from the full tpm_expressed dataset. Click on the **Show/Hide** button to see the steps.

<button class="btn btn-primary" data-toggle="collapse" data-target="#BlockName12"> Show/Hide </button>  
<div id="BlockName12" class="collapse">
```{r}
# Subset data from the full tpm_expressed dataset
tuss_T4vsT0<-tpm_tuss_expressed[,c(2:5,8:14)]

# Aggregate (sum) genecall ID counts by shared KO and Taxonomy to reduce duplicates in the dataset and improve the DE analysis results to be biologically relevant.
tuss_T4vsT0<-aggregate(. ~ KO + Symbol + Function + Tier_II + Tier_III + Tier_IV + Taxonomy, data=tuss_T4vsT0,FUN=sum)

## Reading in the data
# Create DGEList for Tuss T4 vs T0 subdata
tuss_T4vsT0<-DGEList(counts=tuss_T4vsT0[,8:11], genes=tuss_T4vsT0[,1:7])
rownames(tuss_T4vsT0$counts)<-rownames(tuss_T4vsT0$genes)<-tuss_T4vsT0$genes$ID
tuss_T4vsT0$genes$ID<-NULL

## Normalization
# Tell edgeR that you don't want to normalize the subdata (already TPM normalized)
tuss_T4vsT0<-calcNormFactors(tuss_T4vsT0, method="none")
tuss_T4vsT0$samples

## Data Exploration
# The plotMDS function produces a plot in which distances between samples correspond to leading biological coefficient of variation (BCV) between those samples
plotMDS(tuss_T4vsT0, col=rep(1:2, each=2))

## The Design Matrix
# Before we fit negative binomial GLMs, we need to define our design matrix based on the experimental design. Here, we want to test for differential expression between T4 and T0 timepoints within mesocosms, i.e., adjusting for differences between mesocosms. In statistics, this is an additive linear model with mesocosm as the blocking factor.
Mesocosm_tuss_T4vsT0<-factor(c(1,2,1,2))
Timepoint_tuss_T4vsT0<-factor(c("T0","T0","T4","T4"))
data.frame(Sample_tuss_T4vsT0=colnames(tuss_T4vsT0),Mesocosm_tuss_T4vsT0,Timepoint_tuss_T4vsT0)
tuss_T4vsT0_design<-model.matrix(~Mesocosm_tuss_T4vsT0+Timepoint_tuss_T4vsT0)
rownames(tuss_T4vsT0_design)<-colnames(tuss_T4vsT0)
tuss_T4vsT0_design

## Estimating Dispersion
# We estimate the NB dispersion for the dataset
tuss_T4vsT0<-estimateDisp(tuss_T4vsT0,tuss_T4vsT0_design,robust=TRUE)
tuss_T4vsT0$common.dispersion

# The square root of the common dispersion gives the coefficient of variation of biological variation.
sqrt(tuss_T4vsT0$common.dispersion)

# View the dispersion estimates in a BCV plot
plotBCV(tuss_T4vsT0)

## Differential Expression
# Now we proceed to determine differentially expressed genes by fitting genewise glms
tuss_T4vsT0_fit<-glmFit(tuss_T4vsT0,tuss_T4vsT0_design)

# Conduct likelihood ratio tests for T4 vs T0 differences and show the top genes. The genewise tests are for T4 vs T0 differential expression, adjusting for baseline differences between the two mesocosms.  The tests can be viewed as analogous to paired t-tests. The top DE tags have tiny p-values and FDR values, as well as large fold changes.
tuss_T4vsT0_lrt<-glmLRT(tuss_T4vsT0_fit)
tuss_T4vsT0_topTags<-topTags(tuss_T4vsT0_lrt,n=25)
tuss_T4vsT0_topTags

## Here's the TPM-normalized values in individual samples for all significant genes.
# We see that all the significant genes have consistent T4 vs T0 changes for the two mesocosms.
tuss_T4vsT0_pvalue<-order(tuss_T4vsT0_lrt$table$PValue)
cpm(tuss_T4vsT0)[tuss_T4vsT0_pvalue[1:25],]
```
</div>

Plotting significant KOs in DGE analysis.
```{r}
# The total number of differentially expressed genes at 5% FDR is given by:
tuss_T4vsT0_FDR<-p.adjust(tuss_T4vsT0_lrt$table$PValue, method="BH")
sum(tuss_T4vsT0_FDR<0.05)
summary(decideTests(tuss_T4vsT0_lrt))

# Plot the log-fold change against log-transcripts per million, with DE genes highlighted.
tuss_T4vsT0_MDplot<-plotMD(tuss_T4vsT0_lrt, hl.col=c("green3","palegreen1"), main = NULL, xlab="Average log TPM") + abline(h=0,col="gray", lty="dashed")
tuss_T4vsT0_MDplot
```

```{r echo=FALSE}
# Write the topTags results to .csv table
tuss_T4vsT0_results <- as.data.frame(topTags(tuss_T4vsT0_lrt,n = Inf))
write.csv(tuss_T4vsT0_results,"DE.Results/tuss_T4vsT0_results.csv", row.names = FALSE)
```

##### 7.4.2.2. Tuss T24vsT4

Now analyze the differentially expressed genes in the Tussock dataset between timepoints T24 (mixed redox) and T4 (oxic redox). Click on the **Show/Hide** button to see the steps.

<button class="btn btn-primary" data-toggle="collapse" data-target="#BlockName13"> Show/Hide </button>  
<div id="BlockName13" class="collapse">
```{r}
# Subset data from the full tpm_expressed dataset
tuss_T24vsT4<-tpm_tuss_expressed[,c(4:7,8:14)]

# Aggregate (sum) genecall ID counts by shared KO and Taxonomy to reduce duplicates in the dataset and improve the DE analysis results to be biologically relevant.
tuss_T24vsT4<-aggregate(. ~ KO + Symbol + Function + Tier_II + Tier_III + Tier_IV + Taxonomy, data=tuss_T24vsT4, FUN=sum)

## Reading in the data
# Create DGEList for WS T4 vs T0 subdata
tuss_T24vsT4<-DGEList(counts=tuss_T24vsT4[,8:11], genes=tuss_T24vsT4[,1:7])
rownames(tuss_T24vsT4$counts)<-rownames(tuss_T24vsT4$genes)<-tuss_T24vsT4$genes$ID
tuss_T24vsT4$genes$ID<-NULL

## Normalization
# Tell edgeR that you don't want to normalize the subdata (already TPM normalized)
tuss_T24vsT4<-calcNormFactors(tuss_T24vsT4, method="none")
tuss_T24vsT4$samples

## Data Exploration
# The plotMDS function produces a plot in which distances between samples correspond to leading biological coefficient of variation (BCV) between those samples
plotMDS(tuss_T24vsT4, col=rep(1:2, each=2))

## The Design Matrix
# Before we fit negative binomial GLMs, we need to define our design matrix based on the experimental design. Here, we want to test for differential expression between T4 and T0 timepoints within mesocosms, i.e., adjusting for differences between mesocosms. In statistics, this is an additive linear model with mesocosm as the blocking factor.
Mesocosm_tuss_T24vsT4<-factor(c(1,2,1,2))
Timepoint_tuss_T24vsT4<-factor(c("T4","T4","T24","T24"))
data.frame(Sample_tuss_T24vsT4=colnames(tuss_T24vsT4),Mesocosm_tuss_T24vsT4,Timepoint_tuss_T24vsT4)
tuss_T24vsT4_design<-model.matrix(~Mesocosm_tuss_T24vsT4+Timepoint_tuss_T24vsT4)
rownames(tuss_T24vsT4_design)<-colnames(tuss_T24vsT4)
tuss_T24vsT4_design

## Estimating Dispersion
# We estimate the NB dispersion for the dataset
tuss_T24vsT4<-estimateDisp(tuss_T24vsT4,tuss_T24vsT4_design,robust=TRUE)
tuss_T24vsT4$common.dispersion

# The square root of the common dispersion gives the coefficient of variation of biological variation.
sqrt(tuss_T24vsT4$common.dispersion)

# View the dispersion estimates in a BCV plot
plotBCV(tuss_T24vsT4)

## Differential Expression
# Now we proceed to determine differentially expressed genes by fitting genewise glms
tuss_T24vsT4_fit<-glmFit(tuss_T24vsT4,tuss_T24vsT4_design)

# Conduct likelihood ratio tests for T4 vs T0 differences and show the top genes. The genewise tests are for T4 vs T0 differential expression, adjusting for baseline differences between the two mesocosms.  The tests can be viewed as analogous to paired t-tests. The top DE tags have tiny p-values and FDR values, as well as large fold changes.
tuss_T24vsT4_lrt<-glmLRT(tuss_T24vsT4_fit)
topTags(tuss_T24vsT4_lrt,n=25)

# Here's the TPM-normalized values in individual samples.
# We see that all the genes have consistent changes between replicates within each timepoint.
tuss_T24vsT4_pvalue<-order(tuss_T24vsT4_lrt$table$PValue)
cpm(tuss_T24vsT4)[tuss_T24vsT4_pvalue[1:25],]
```
</div>

Plotting significant KOs in DGE analysis.
```{r}
# The total number of differentially expressed genes at 5% FDR is given by:
tuss_T24vsT4_FDR<-p.adjust(tuss_T24vsT4_lrt$table$PValue, method="BH")
sum(tuss_T24vsT4_FDR<0.05)
summary(decideTests(tuss_T24vsT4_lrt))

# Plot the log-fold change against log-transcripts per million, with DE genes highlighted.
tuss_T24vsT4_MDplot<-plotMD(tuss_T24vsT4_lrt, hl.col=c("darkgreen","green3"), main = NULL, xlab="Average log TPM") + abline(h=0,col="gray", lty="dashed")
tuss_T24vsT4_MDplot
```

```{r echo=FALSE}
# Write the topTags results to .csv table
tuss_T24vsT4_results <- as.data.frame(topTags(tuss_T24vsT4_lrt,n = Inf))
write.csv(tuss_T24vsT4_results,"DE.Results/tuss_T24vsT4_results.csv", row.names = FALSE)
```

##### 7.4.2.3. Tuss T24vsT0

Finally, analyze the differentially expressed genes in the Tussock dataset between timepoints T24 (mixed redox) and T0 (anoxic redox). Click on the **Show/Hide** button to see the steps.

<button class="btn btn-primary" data-toggle="collapse" data-target="#BlockName14"> Show/Hide </button>  
<div id="BlockName14" class="collapse">
```{r}
# Subset data from the full tpm_expressed dataset
tuss_T24vsT0<-tpm_tuss_expressed[,c(2:3,6:14)]

# Aggregate (sum) genecall ID counts by shared KO and Taxonomy to reduce duplicates in the dataset and improve the DE analysis results to be biologically relevant.
tuss_T24vsT0<-aggregate(. ~ KO + Symbol + Function + Tier_II + Tier_III + Tier_IV + Taxonomy, data=tuss_T24vsT0, FUN=sum)

## Reading in the data
# Create DGEList for Tuss T24 vs T0 subdata
tuss_T24vsT0<-DGEList(counts=tuss_T24vsT0[,8:11], genes=tuss_T24vsT0[,1:7])
rownames(tuss_T24vsT0$counts)<-rownames(tuss_T24vsT0$genes)<-tuss_T24vsT0$genes$ID
tuss_T24vsT0$genes$ID<-NULL

## Normalization
# Tell edgeR that you don't want to normalize the subdata (already TPM normalized)
tuss_T24vsT0<-calcNormFactors(tuss_T24vsT0, method="none")
tuss_T24vsT0$samples

## Data Exploration
# The plotMDS function produces a plot in which distances between samples correspond to leading biological coefficient of variation (BCV) between those samples
plotMDS(tuss_T24vsT0, col=rep(1:2, each=2))

## The Design Matrix
# Before we fit negative binomial GLMs, we need to define our design matrix based on the experimental design. Here, we want to test for differential expression between T4 and T0 timepoints within mesocosms, i.e., adjusting for differences between mesocosms. In statistics, this is an additive linear model with mesocosm as the blocking factor.
Mesocosm_tuss_T24vsT0<-factor(c(1,2,1,2))
Timepoint_tuss_T24vsT0<-factor(c("T0","T0","T24","T24"))
data.frame(Sample_tuss_T24vsT0=colnames(tuss_T24vsT0),Mesocosm_tuss_T24vsT0,Timepoint_tuss_T24vsT0)
tuss_T24vsT0_design<-model.matrix(~Mesocosm_tuss_T24vsT0+Timepoint_tuss_T24vsT0)
rownames(tuss_T24vsT0_design)<-colnames(tuss_T24vsT0)
tuss_T24vsT0_design

## Estimating Dispersion
# We estimate the NB dispersion for the dataset
tuss_T24vsT0<-estimateDisp(tuss_T24vsT0,tuss_T24vsT0_design,robust=TRUE)
tuss_T24vsT0$common.dispersion

# The square root of the common dispersion gives the coefficient of variation of biological variation.
sqrt(tuss_T24vsT0$common.dispersion)

# View the dispersion estimates in a BCV plot
plotBCV(tuss_T24vsT0)

## Differential Expression
# Now we proceed to determine differentially expressed genes by fitting genewise glms
tuss_T24vsT0_fit<-glmFit(tuss_T24vsT0,tuss_T24vsT0_design)

# Conduct likelihood ratio tests for T4 vs T0 differences and show the top genes. The genewise tests are for T4 vs T0 differential expression, adjusting for baseline differences between the two mesocosms.  The tests can be viewed as analogous to paired t-tests. The top DE tags have tiny p-values and FDR values, as well as large fold changes.
tuss_T24vsT0_lrt<-glmLRT(tuss_T24vsT0_fit)
topTags(tuss_T24vsT0_lrt,n=25)

## Here's the TPM-normalized values in individual samples.
# We see that all the genes have consistent changes between replicates within each timepoint.
tuss_T24vsT0_pvalue<-order(tuss_T24vsT0_lrt$table$PValue)
cpm(tuss_T24vsT0)[tuss_T24vsT0_pvalue[1:25],]
```
</div>

Plotting significant KOs in DGE analysis.
```{r}
# The total number of differentially expressed genes at 5% FDR is given by:
tuss_T24vsT0_FDR<-p.adjust(tuss_T24vsT0_lrt$table$PValue, method="BH")
sum(tuss_T24vsT0_FDR<0.05)
summary(decideTests(tuss_T24vsT0_lrt))

# Plot the log-fold change against log-transcripts per million, with DE genes highlighted.
tuss_T24vsT0_MDplot<-plotMD(tuss_T24vsT0_lrt, hl.col=c("darkgreen","palegreen1"), main = NULL, xlab="Average log TPM") + abline(h=0,col="gray", lty="dashed")
tuss_T24vsT0_MDplot
```

```{r echo=FALSE}
# Write the topTags results to .csv table
tuss_T24vsT0_results <- as.data.frame(topTags(tuss_T24vsT0_lrt,n = Inf))
write.csv(tuss_T24vsT0_results,"DE.Results/tuss_T24vsT0_results.csv", row.names = FALSE)
```

##### 7.4.2.4. WS T4vsT0

Start by analysing differentially expressed genes in the Wet Sedge dataset between timepoints T4 (oxic redox) and T0 (anoxic redox). Click on the **Show/Hide** button to see the steps.

<button class="btn btn-primary" data-toggle="collapse" data-target="#BlockName15"> Show/Hide </button>  
<div id="BlockName15" class="collapse">
```{r}
# Subset data from the full tpm_expressed dataset
ws_T4vsT0<-tpm_ws_expressed[,c(2:5,8:14)]

# Aggregate (sum) genecall ID counts by shared KO and Taxonomy to reduce duplicates in the dataset and improve the DE analysis results to be biologically relevant.
ws_T4vsT0<-aggregate(. ~ KO + Symbol + Function + Tier_II + Tier_III + Tier_IV + Taxonomy, data=ws_T4vsT0, FUN=sum)

## Reading in the data
# Create DGEList for WS T4 vs T0 subdata
ws_T4vsT0<-DGEList(counts=ws_T4vsT0[,8:11], genes=ws_T4vsT0[,1:7])
rownames(ws_T4vsT0$counts)<-rownames(ws_T4vsT0$genes)<-ws_T4vsT0$genes$ID
ws_T4vsT0$genes$ID<-NULL

## Normalization
# Tell edgeR that you don't want to normalize the subdata (already TPM normalized)
ws_T4vsT0<-calcNormFactors(ws_T4vsT0, method="none")
ws_T4vsT0$samples

## Data Exploration
# The plotMDS function produces a plot in which distances between samples correspond to leading biological coefficient of variation (BCV) between those samples
plotMDS(ws_T4vsT0, col=rep(1:2, each=2))

## The Design Matrix
# Before we fit negative binomial GLMs, we need to define our design matrix based on the experimental design. Here, we want to test for differential expression between T4 and T0 timepoints within mesocosms, i.e., adjusting for differences between mesocosms. In statistics, this is an additive linear model with mesocosm as the blocking factor.
Mesocosm_ws_T4vsT0<-factor(c(1,2,1,2))
Timepoint_ws_T4vsT0<-factor(c("T0","T0","T4","T4"))
data.frame(Sample_ws_T4vsT0=colnames(ws_T4vsT0),Mesocosm_ws_T4vsT0,Timepoint_ws_T4vsT0)
ws_T4vsT0_design<-model.matrix(~Mesocosm_ws_T4vsT0+Timepoint_ws_T4vsT0)
rownames(ws_T4vsT0_design)<-colnames(ws_T4vsT0)
ws_T4vsT0_design

## Estimating Dispersion
# We estimate the NB dispersion for the dataset
ws_T4vsT0<-estimateDisp(ws_T4vsT0,ws_T4vsT0_design,robust=TRUE)
ws_T4vsT0$common.dispersion

# The square root of the common dispersion gives the coefficient of variation of biological variation.
sqrt(ws_T4vsT0$common.dispersion)

# View the dispersion estimates in a BCV plot
plotBCV(ws_T4vsT0)

## Differential Expression
# Now we proceed to determine differentially expressed genes by fitting genewise glms
ws_T4vsT0_fit<-glmFit(ws_T4vsT0,ws_T4vsT0_design)

# Conduct likelihood ratio tests for T4 vs T0 differences and show the top genes. The genewise tests are for T4 vs T0 differential expression, adjusting for baseline differences between the two mesocosms.  The tests can be viewed as analogous to paired t-tests. The top DE tags have tiny p-values and FDR values, as well as large fold changes.
ws_T4vsT0_lrt<-glmLRT(ws_T4vsT0_fit)
ws_T4vsT0_topTags<-topTags(ws_T4vsT0_lrt,n=25)
ws_T4vsT0_topTags

## Here's the TPM-normalized values in individual samples.
# We see that all the significant genes have consistent T4 vs T0 changes for the two mesocosms.
ws_T4vsT0_pvalue<-order(ws_T4vsT0_lrt$table$PValue)
ws_T4vsT0_cpm<-cpm(ws_T4vsT0)[ws_T4vsT0_pvalue[1:25],]
ws_T4vsT0_cpm
```
</div>

Plotting significant KOs in DGE analysis.
```{r}
# The total number of differentially expressed genes at 5% FDR is given by:
ws_T4vsT0_FDR<-p.adjust(ws_T4vsT0_lrt$table$PValue, method="BH")
sum(ws_T4vsT0_FDR<0.05)
summary(decideTests(ws_T4vsT0_lrt))

# Plot the log-fold change against log-transcripts per million, with DE genes highlighted.
ws_T4vsT0_MDplot<-plotMD(ws_T4vsT0_lrt, hl.col=c("dodgerblue1","lightskyblue1"), main = NULL, xlab="Average log TPM") + abline(h=0,col="gray", lty="dashed")
ws_T4vsT0_MDplot
```

```{r echo=FALSE}
# Write the topTags results to .csv table
ws_T4vsT0_results <- as.data.frame(topTags(ws_T4vsT0_lrt,n = Inf))
write.csv(ws_T4vsT0_results,"DE.Results/ws_T4vsT0_results.csv", row.names = FALSE)
```

##### 7.4.2.5. WS T24vsT4

Now analyze the differentially expressed genes in the Wet Sedge dataset between timepoints T24 (mixed redox) and T4 (oxic redox). Click on the **Show/Hide** button to see the steps.

<button class="btn btn-primary" data-toggle="collapse" data-target="#BlockName16"> Show/Hide </button>  
<div id="BlockName16" class="collapse">
```{r}
# Subset data from the full tpm_expressed dataset
ws_T24vsT4<-tpm_ws_expressed[,c(4:7,8:14)]

# Aggregate (sum) genecall ID counts by shared KO and Taxonomy to reduce duplicates in the dataset and improve the DE analysis results to be biologically relevant.
ws_T24vsT4<-aggregate(. ~ KO + Symbol + Function + Tier_II + Tier_III + Tier_IV + Taxonomy, data=ws_T24vsT4, FUN=sum)

## Reading in the data
# Create DGEList for WS T24 vs T4 subdata
ws_T24vsT4<-DGEList(counts=ws_T24vsT4[,8:11], genes=ws_T24vsT4[,1:7])
rownames(ws_T24vsT4$counts)<-rownames(ws_T24vsT4$genes)<-ws_T24vsT4$genes$ID
ws_T24vsT4$genes$ID<-NULL

## Normalization
# Tell edgeR that you don't want to normalize the subdata (already TPM normalized)
ws_T24vsT4<-calcNormFactors(ws_T24vsT4, method="none")
ws_T24vsT4$samples

## Data Exploration
# The plotMDS function produces a plot in which distances between samples correspond to leading biological coefficient of variation (BCV) between those samples
plotMDS(ws_T24vsT4, col=rep(1:2, each=2))

## The Design Matrix
# Before we fit negative binomial GLMs, we need to define our design matrix based on the experimental design. Here, we want to test for differential expression between T4 and T0 timepoints within mesocosms, i.e., adjusting for differences between mesocosms. In statistics, this is an additive linear model with mesocosm as the blocking factor.
Mesocosm_ws_T24vsT4<-factor(c(1,2,1,2))
Timepoint_ws_T24vsT4<-factor(c("T4","T4","T24","T24"))
data.frame(Sample_ws_T24vsT4=colnames(ws_T24vsT4),Mesocosm_ws_T24vsT4,Timepoint_ws_T24vsT4)
ws_T24vsT4_design<-model.matrix(~Mesocosm_ws_T24vsT4+Timepoint_ws_T24vsT4)
rownames(ws_T24vsT4_design)<-colnames(ws_T24vsT4)
ws_T24vsT4_design

## Estimating Dispersion
# We estimate the NB dispersion for the dataset
ws_T24vsT4<-estimateDisp(ws_T24vsT4,ws_T24vsT4_design,robust=TRUE)
ws_T24vsT4$common.dispersion

# The square root of the common dispersion gives the coefficient of variation of biological variation.
sqrt(ws_T24vsT4$common.dispersion)

# View the dispersion estimates in a BCV plot
plotBCV(ws_T24vsT4)

## Differential Expression
# Now we proceed to determine differentially expressed genes by fitting genewise glms
ws_T24vsT4_fit<-glmFit(ws_T24vsT4,ws_T24vsT4_design)

# Conduct likelihood ratio tests for T4 vs T0 differences and show the top genes. The genewise tests are for T4 vs T0 differential expression, adjusting for baseline differences between the two mesocosms.  The tests can be viewed as analogous to paired t-tests. The top DE tags have tiny p-values and FDR values, as well as large fold changes.
ws_T24vsT4_lrt<-glmLRT(ws_T24vsT4_fit)
ws_T24vsT4_topTags<-topTags(ws_T24vsT4_lrt,n=25)
ws_T24vsT4_topTags

## Here's the TPM-normalized values in individual samples.
# We see that all the genes have consistent changes between replicates within each timepoint.
ws_T24vsT4_pvalue<-order(ws_T24vsT4_lrt$table$PValue)
ws_T24vsT4_cpm<-cpm(ws_T24vsT4)[ws_T24vsT4_pvalue[1:25],]
ws_T24vsT4_cpm
```
</div>

Plotting significant KOs in DGE analysis.
```{r}
# The total number of differentially expressed genes at 5% FDR is given by:
ws_T24vsT4_FDR<-p.adjust(ws_T24vsT4_lrt$table$PValue, method="BH")
sum(ws_T24vsT4_FDR<0.05)
summary(decideTests(ws_T24vsT4_lrt))

# Plot the log-fold change against log-transcripts per million, with DE genes highlighted.
ws_T24vsT4_MDplot<-plotMD(ws_T24vsT4_lrt, hl.col=c("midnightblue","dodgerblue1"), main = NULL, xlab="Average log TPM") + abline(h=0,col="gray", lty="dashed")
ws_T24vsT4_MDplot
```

```{r echo=FALSE}
# Write the topTags results to .csv table
ws_T24vsT4_results <- as.data.frame(topTags(ws_T24vsT4_lrt,n = Inf))
write.csv(ws_T24vsT4_results,"DE.Results/ws_T24vsT4_results.csv", row.names = FALSE)
```

##### 7.4.2.6. WS T24vsT0

Finally, analyze the differentially expressed genes in the Wet Sedge dataset between timepoints T24 (mixed redox) and T0 (anoxic redox). Click on the **Show/Hide** button to see the steps.

<button class="btn btn-primary" data-toggle="collapse" data-target="#BlockName17"> Show/Hide </button>  
<div id="BlockName17" class="collapse">
```{r}
# Subset data from the full tpm_expressed dataset
ws_T24vsT0<-tpm_ws_expressed[,c(2:3,6:14)]

# Aggregate (sum) genecall ID counts by shared KO and Taxonomy to reduce duplicates in the dataset and improve the DE analysis results to be biologically relevant.
ws_T24vsT0<-aggregate(. ~ KO + Symbol + Function + Tier_II + Tier_III + Tier_IV + Taxonomy, data=ws_T24vsT0, FUN=sum)

## Reading in the data
# Create DGEList for WS T24 vs T0 subdata
ws_T24vsT0<-DGEList(counts=ws_T24vsT0[,8:11], genes=ws_T24vsT0[,1:7])
rownames(ws_T24vsT0$counts)<-rownames(ws_T24vsT0$genes)<-ws_T24vsT0$genes$ID
ws_T24vsT0$genes$ID<-NULL

## Normalization
# Tell edgeR that you don't want to normalize the subdata (already TPM normalized)
ws_T24vsT0<-calcNormFactors(ws_T24vsT0, method="none")
ws_T24vsT0$samples

## Data Exploration
# The plotMDS function produces a plot in which distances between samples correspond to leading biological coefficient of variation (BCV) between those samples
plotMDS(ws_T24vsT0, col=rep(1:2, each=2))

## The Design Matrix
# Before we fit negative binomial GLMs, we need to define our design matrix based on the experimental design. Here, we want to test for differential expression between T4 and T0 timepoints within mesocosms, i.e., adjusting for differences between mesocosms. In statistics, this is an additive linear model with mesocosm as the blocking factor.
Mesocosm_ws_T24vsT0<-factor(c(1,2,1,2))
Timepoint_ws_T24vsT0<-factor(c("T0","T0","T24","T24"))
data.frame(Sample_ws_T24vsT0=colnames(ws_T24vsT0),Mesocosm_ws_T24vsT0,Timepoint_ws_T24vsT0)
ws_T24vsT0_design<-model.matrix(~Mesocosm_ws_T24vsT0+Timepoint_ws_T24vsT0)
rownames(ws_T24vsT0_design)<-colnames(ws_T24vsT0)
ws_T24vsT0_design

## Estimating Dispersion
# We estimate the NB dispersion for the dataset
ws_T24vsT0<-estimateDisp(ws_T24vsT0,ws_T24vsT0_design,robust=TRUE)
ws_T24vsT0$common.dispersion

# The square root of the common dispersion gives the coefficient of variation of biological variation.
sqrt(ws_T24vsT0$common.dispersion)

# View the dispersion estimates in a BCV plot
plotBCV(ws_T24vsT0)

## Differential Expression
# Now we proceed to determine differentially expressed genes by fitting genewise glms
ws_T24vsT0_fit<-glmFit(ws_T24vsT0,ws_T24vsT0_design)

# Conduct likelihood ratio tests for T4 vs T0 differences and show the top genes. The genewise tests are for T4 vs T0 differential expression, adjusting for baseline differences between the two mesocosms.  The tests can be viewed as analogous to paired t-tests. The top DE tags have tiny p-values and FDR values, as well as large fold changes.
ws_T24vsT0_lrt<-glmLRT(ws_T24vsT0_fit)
topTags(ws_T24vsT0_lrt,n=25)

## Here's the TPM-normalized values in individual samples.
# We see that all the genes have consistent changes between replicates within each timepoint.
ws_T24vsT0_pvalue<-order(ws_T24vsT0_lrt$table$PValue)
cpm(ws_T24vsT0)[ws_T24vsT0_pvalue[1:25],]
```
</div>

Plotting significant KOs in DGE analysis.
```{r}
# The total number of differentially expressed genes at 5% FDR is given by:
ws_T24vsT0_FDR<-p.adjust(ws_T24vsT0_lrt$table$PValue, method="BH")
sum(ws_T24vsT0_FDR<0.05)
summary(decideTests(ws_T24vsT0_lrt))

# Plot the log-fold change against log-transcripts per million, with DE genes highlighted.
ws_T24vsT0_MDplot<-plotMD(ws_T24vsT0_lrt, hl.col=c("midnightblue","lightskyblue1"), main = NULL, xlab="Average log TPM") + abline(h=0,col="gray", lty="dashed")
ws_T24vsT0_MDplot
```

```{r echo=FALSE}
# Write the topTags results to .csv table
ws_T24vsT0_results <- as.data.frame(topTags(ws_T24vsT0_lrt,n = Inf))
write.csv(ws_T24vsT0_results,"DE.Results/ws_T24vsT0_results.csv", row.names = FALSE)
```

#### 7.4.3. DGE Between Tundra

Now switch perspectives and determine which expressed genes were significantly differentially expressed between ecosystems (***i.e., vegetation types***) at the same sampling timepoint.

##### 7.4.3.1. WSvsTuss T0

Compare gene expression under anoxic conditions (T0) between Wet Sedge and Tussocks. Click on the **Show/Hide** button to see the steps.

<button class="btn btn-primary" data-toggle="collapse" data-target="#BlockName18"> Show/Hide </button>  
<div id="BlockName18" class="collapse">
```{r}
# Subset data from the full tpm_expressed dataset
ws_tuss_T0<-tpm_all_exp_ann[,c(2:5,14:20)]

# Aggregate (sum) genecall ID counts by shared KO and Taxonomy to reduce duplicates in the dataset and improve the DE analysis results to be biologically relevant.
ws_tuss_T0<-aggregate(. ~ KO + Symbol + Function + Tier_II + Tier_III + Tier_IV + Taxonomy, data=ws_tuss_T0, FUN=sum)

## Reading in the data
# Create DGEList for Tuss vs WS at T4 subdata
ws_tuss_T0<-DGEList(counts=ws_tuss_T0[,8:11], genes=ws_tuss_T0[,1:7])
rownames(ws_tuss_T0$counts)<-rownames(ws_tuss_T0$genes)<-ws_tuss_T0$genes$ID
ws_tuss_T0$genes$ID<-NULL

## Normalization
# Tell edgeR that you don't want to normalize the subdata (already TPM normalized)
ws_tuss_T0<-calcNormFactors(ws_tuss_T0, method="none")
ws_tuss_T0$samples

## Data Exploration
# The plotMDS function produces a plot in which distances between samples correspond to leading biological coefficient of variation (BCV) between those samples
plotMDS(ws_tuss_T0, col=rep(1:2, each=2))

## The Design Matrix
# Before we fit negative binomial GLMs, we need to define our design matrix based on the experimental design. Here, we want to test for differential expression between T4 and T0 timepoints within mesocosms, i.e., adjusting for differences between mesocosms. In statistics, this is an additive linear model with mesocosm as the blocking factor.
Mesocosm_ws_tuss_T0<-factor(c(1,2,1,2))
Vegetation_ws_tuss_T0<-factor(c("WS","WS","TUSS","TUSS"))
data.frame(Sample_ws_tuss_T0=colnames(ws_tuss_T0),Mesocosm_ws_tuss_T0,Vegetation_ws_tuss_T0)
ws_tuss_T0_design<-model.matrix(~Mesocosm_ws_tuss_T0+Vegetation_ws_tuss_T0)
rownames(ws_tuss_T0_design)<-colnames(ws_tuss_T0)
ws_tuss_T0_design

## Estimating Dispersion
# We estimate the NB dispersion for the dataset
ws_tuss_T0<-estimateDisp(ws_tuss_T0,ws_tuss_T0_design,robust=TRUE)
ws_tuss_T0$common.dispersion

# The square root of the common dispersion gives the coefficient of variation of biological variation.
sqrt(ws_tuss_T0$common.dispersion)

# View the dispersion estimates in a BCV plot
plotBCV(ws_tuss_T0)

## Differential Expression
# Now we proceed to determine differentially expressed genes by fitting genewise glms
ws_tuss_T0_fit<-glmFit(ws_tuss_T0,ws_tuss_T0_design)

# Conduct likelihood ratio tests for WS T0 vs TUSS T0 differences and show the top genes. The genewise tests are for WS T0 vs TUSS T0 differential expression, adjusting for baseline differences between the replicate mesocosms.  The tests can be viewed as analogous to paired t-tests. The top DE tags have tiny p-values and FDR values, as well as large fold changes.
ws_tuss_T0_lrt<-glmLRT(ws_tuss_T0_fit)
ws_tuss_T0_topTags<-topTags(ws_tuss_T0_lrt,n=25)
ws_tuss_T0_topTags

## Here's the TPM-normalized values in individual samples.
# We see that all the genes have consistent changes between replicates within each timepoint.
ws_tuss_T0_pvalue<-order(ws_tuss_T0_lrt$table$PValue)
cpm(ws_tuss_T0)[ws_tuss_T0_pvalue[1:25],]
```
</div>

Plotting significant KOs in DGE analysis.
```{r}
# The total number of differentially expressed genes at 5% FDR is given by:
ws_tuss_T0_FDR<-p.adjust(ws_tuss_T0_lrt$table$PValue, method="BH")
sum(ws_tuss_T0_FDR<0.05)
summary(decideTests(ws_tuss_T0_lrt))

# Plot the log-fold change against log-transcripts per million, with DE genes highlighted.
ws_tuss_T0_MDplot<-plotMD(ws_tuss_T0_lrt, hl.col=c("palegreen2","lightskyblue2"), main = NULL, xlab="Average log TPM") + abline(h=0,col="gray", lty="dashed")
ws_tuss_T0_MDplot
```

```{r echo=FALSE}
# Write the topTags results to .csv table
ws_tuss_T0_results <- as.data.frame(topTags(ws_tuss_T0_lrt,n = Inf))
write.csv(ws_tuss_T0_results,"DE.Results/ws_tuss_T0_results.csv", row.names = FALSE)
```

##### 7.4.3.2. WSvsTuss T4

Next, compare gene expression under oxic conditions (T4) between Wet Sedge and Tussocks. Click on the **Show/Hide** button to see the steps.

<button class="btn btn-primary" data-toggle="collapse" data-target="#BlockName19"> Show/Hide </button>  
<div id="BlockName19" class="collapse">
```{r}
# Subset data from the full tpm_expressed dataset
ws_tuss_T4<-tpm_all_exp_ann[,c(6:7,10:11,14:20)]

# Aggregate (sum) genecall ID counts by shared KO and Taxonomy to reduce duplicates in the dataset and improve the DE analysis results to be biologically relevant.
ws_tuss_T4<-aggregate(. ~ KO + Symbol + Function + Tier_II + Tier_III + Tier_IV + Taxonomy, data=ws_tuss_T4, FUN=sum)

## Reading in the data
# Create DGEList for Tuss vs WS at T4 subdata
ws_tuss_T4<-DGEList(counts=ws_tuss_T4[,8:11], genes=ws_tuss_T4[,1:7])
rownames(ws_tuss_T4$counts)<-rownames(ws_tuss_T4$genes)<-ws_tuss_T4$genes$ID
ws_tuss_T4$genes$ID<-NULL

## Normalization
# Tell edgeR that you don't want to normalize the subdata (already TPM normalized)
ws_tuss_T4<-calcNormFactors(ws_tuss_T4, method="none")
ws_tuss_T4$samples

## Data Exploration
# The plotMDS function produces a plot in which distances between samples correspond to leading biological coefficient of variation (BCV) between those samples
plotMDS(ws_tuss_T4, col=rep(1:2, each=2))

## The Design Matrix
# Before we fit negative binomial GLMs, we need to define our design matrix based on the experimental design. Here, we want to test for differential expression between T4 and T0 timepoints within mesocosms, i.e., adjusting for differences between mesocosms. In statistics, this is an additive linear model with mesocosm as the blocking factor.
Mesocosm_ws_tuss_T4<-factor(c(1,2,1,2))
Vegetation_ws_tuss_T4<-factor(c("WS","WS","TUSS","TUSS"))
data.frame(Sample_ws_tuss_T4=colnames(ws_tuss_T4),Mesocosm_ws_tuss_T4,Vegetation_ws_tuss_T4)
ws_tuss_T4_design<-model.matrix(~Mesocosm_ws_tuss_T4+Vegetation_ws_tuss_T4)
rownames(ws_tuss_T4_design)<-colnames(ws_tuss_T4)
ws_tuss_T4_design

## Estimating Dispersion
# We estimate the NB dispersion for the dataset
ws_tuss_T4<-estimateDisp(ws_tuss_T4,ws_tuss_T4_design,robust=TRUE)
ws_tuss_T4$common.dispersion

# The square root of the common dispersion gives the coefficient of variation of biological variation.
sqrt(ws_tuss_T4$common.dispersion)

# View the dispersion estimates in a BCV plot
plotBCV(ws_tuss_T4)

## Differential Expression
# Now we proceed to determine differentially expressed genes by fitting genewise glms
ws_tuss_T4_fit<-glmFit(ws_tuss_T4,ws_tuss_T4_design)

# Conduct likelihood ratio tests for WS T4 vs TUSS T4 differences and show the top genes. The genewise tests are for WS T4 vs TUSS T4 differential expression, adjusting for baseline differences between the replicate mesocosms.  The tests can be viewed as analogous to paired t-tests. The top DE tags have tiny p-values and FDR values, as well as large fold changes.
ws_tuss_T4_lrt<-glmLRT(ws_tuss_T4_fit)
topTags(ws_tuss_T4_lrt,n=25)

## Here's the TPM-normalized values in individual samples.
# We see that all the genes have consistent changes between replicates within each timepoint.
ws_tuss_T4_pvalue<-order(ws_tuss_T4_lrt$table$PValue)
cpm(ws_tuss_T4)[ws_tuss_T4_pvalue[1:25],]
```
</div>

Plotting significant KOs in DGE analysis.
```{r}
# The total number of differentially expressed genes at 5% FDR is given by:
ws_tuss_T4_FDR<-p.adjust(ws_tuss_T4_lrt$table$PValue, method="BH")
sum(ws_tuss_T4_FDR<0.05)
summary(decideTests(ws_tuss_T4_lrt))

# Plot the log-fold change against log-transcripts per million, with DE genes highlighted.
ws_tuss_T4_MDplot<-plotMD(ws_tuss_T4_lrt, hl.col=c("green3","dodgerblue1"), main = NULL, xlab="Average log TPM") + abline(h=0,col="gray", lty="dashed")
ws_tuss_T4_MDplot
```

```{r echo=FALSE}
# Write the topTags results to .csv table
ws_tuss_T4_results <- as.data.frame(topTags(ws_tuss_T4_lrt,n = Inf))
write.csv(ws_tuss_T4_results,"DE.Results/ws_tuss_T4_results.csv", row.names = FALSE)
```

##### 7.4.3.3. WSvsTuss T24

Finally, compare gene expression under mixed-redox conditions (T24) between Wet Sedge and Tussocks. Click on the **Show/Hide** button to see the steps.

<button class="btn btn-primary" data-toggle="collapse" data-target="#BlockName20"> Show/Hide </button>  
<div id="BlockName20" class="collapse">
```{r}
# Subset data from the full tpm_expressed dataset
ws_tuss_T24<-tpm_all_exp_ann[,c(8:9,12:20)]

# Aggregate (sum) genecall ID counts by shared KO and Taxonomy to reduce duplicates in the dataset and improve the DE analysis results to be biologically relevant.
ws_tuss_T24<-aggregate(. ~ KO + Symbol + Function + Tier_II + Tier_III + Tier_IV + Taxonomy, data=ws_tuss_T24, FUN=sum)

## Reading in the data
# Create DGEList for Tuss vs WS at T24 subdata
ws_tuss_T24<-DGEList(counts=ws_tuss_T24[,8:11], genes=ws_tuss_T24[,1:7])
rownames(ws_tuss_T24$counts)<-rownames(ws_tuss_T24$genes)<-ws_tuss_T24$genes$ID
ws_tuss_T24$genes$ID<-NULL

## Normalization
# Tell edgeR that you don't want to normalize the subdata (already TPM normalized)
ws_tuss_T24<-calcNormFactors(ws_tuss_T24, method="none")
ws_tuss_T24$samples

## Data Exploration
# The plotMDS function produces a plot in which distances between samples correspond to leading biological coefficient of variation (BCV) between those samples
plotMDS(ws_tuss_T24, col=rep(1:2, each=2))

## The Design Matrix
# Before we fit negative binomial GLMs, we need to define our design matrix based on the experimental design. Here, we want to test for differential expression between T4 and T0 timepoints within mesocosms, i.e., adjusting for differences between mesocosms. In statistics, this is an additive linear model with mesocosm as the blocking factor.
Mesocosm_ws_tuss_T24<-factor(c(1,2,1,2))
Vegetation_ws_tuss_T24<-factor(c("WS","WS","TUSS","TUSS"))
data.frame(Sample_ws_tuss_T24=colnames(ws_tuss_T24),Mesocosm_ws_tuss_T24,Vegetation_ws_tuss_T24)
ws_tuss_T24_design<-model.matrix(~Mesocosm_ws_tuss_T24+Vegetation_ws_tuss_T24)
rownames(ws_tuss_T24_design)<-colnames(ws_tuss_T24)
ws_tuss_T24_design

## Estimating Dispersion
# We estimate the NB dispersion for the dataset
ws_tuss_T24<-estimateDisp(ws_tuss_T24,ws_tuss_T24_design,robust=TRUE)
ws_tuss_T24$common.dispersion

# The square root of the common dispersion gives the coefficient of variation of biological variation.
sqrt(ws_tuss_T24$common.dispersion)

# View the dispersion estimates in a BCV plot
plotBCV(ws_tuss_T24)

## Differential Expression
# Now we proceed to determine differentially expressed genes by fitting genewise glms
ws_tuss_T24_fit<-glmFit(ws_tuss_T24,ws_tuss_T24_design)

# Conduct likelihood ratio tests for WS T24 vs TUSS T24 differences and show the top genes. The genewise tests are for WS T24 vs TUSS T24 differential expression, adjusting for baseline differences between the replicate mesocosms.  The tests can be viewed as analogous to paired t-tests. The top DE tags have tiny p-values and FDR values, as well as large fold changes.
ws_tuss_T24_lrt<-glmLRT(ws_tuss_T24_fit)
topTags(ws_tuss_T24_lrt,n=25)

## Here's the TPM-normalized values in individual samples.
# We see that all the genes have consistent changes between replicates within each timepoint.
ws_tuss_T24_pvalue<-order(ws_tuss_T24_lrt$table$PValue)
cpm(ws_tuss_T24)[ws_tuss_T24_pvalue[1:25],]
```
</div>

Plotting significant KOs in DGE analysis.
```{r}
# The total number of differentially expressed genes at 5% FDR is given by:
ws_tuss_T24_FDR<-p.adjust(ws_tuss_T24_lrt$table$PValue, method="BH")
sum(ws_tuss_T24_FDR<0.05)
summary(decideTests(ws_tuss_T24_lrt))

# Plot the log-fold change against log-transcripts per million, with DE genes highlighted.
ws_tuss_T24_MDplot<-plotMD(ws_tuss_T24_lrt, hl.col=c("darkgreen","midnightblue"), main = NULL, xlab="Average log TPM") + abline(h=0,col="gray", lty="dashed")
ws_tuss_T24_MDplot
```

```{r echo=FALSE}
# Write the topTags results to .csv table
ws_tuss_T24_results <- as.data.frame(topTags(ws_tuss_T24_lrt,n = Inf))
write.csv(ws_tuss_T24_results,"DE.Results/ws_tuss_T24_results.csv", row.names = FALSE)
```

### 7.5. Fe Gene Expression

Gene expression related to iron acquisition, iron redox cycling, and iron storage were determined using FeGenie [(Garber et al. 2020)](https://www.frontiersin.org/articles/10.3389/fmicb.2020.00037/full).  FeGenie provides a comprehensive database of hidden Markov models (HMMs) based on genes related to iron acquisition, storage, and oxidation/reduction in Bacteria and Archaea, which are generally not annotated as such by established gene annotation pipelines, such as GhostKOALA, which was used to annotate the metatranscriptomes presented in this study.

Here, the quality-controlled sequencing reads for each metatranscriptome sample were assembled separately using MEGAHIT to generate a sample-specific contigs file.  The metatranscriptome contigs files were used as input for FeGenie, which first predicted open-reading frames (ORFs) using Prodigal and then queried them against a custom library of HMMs using hmmsearch, with custom bit score cutoffs for each HMM.  Results from FeGenie included all identified putative iron-related genes, their functional category, bit score, number of canonical heme-binding motifs, amino acid sequence, and closest homolog.  Counts within each iron gene category were summarized as their relative expression against all identified iron genes.

Assemble each QC'd metatranscriptome sample for use in FeGenie
````markdown
`r ''`{Terminal}
module load Bioinformatics megahit/1.2.8
megahit --k-min 21 --k-max 141 --k-step 12 -1 Sample_108379_fwd.good.fastq -2 Sample_108379_rev.good.fastq -o ./megahit_108379_out
megahit --k-min 21 --k-max 141 --k-step 12 -1 Sample_108381_fwd.good.fastq -2 Sample_108381_rev.good.fastq -o ./megahit_108381_out
megahit --k-min 21 --k-max 141 --k-step 12 -1 Sample_108382_fwd.good.fastq -2 Sample_108382_rev.good.fastq -o ./megahit_108382_out
megahit --k-min 21 --k-max 141 --k-step 12 -1 Sample_108383_fwd.good.fastq -2 Sample_108383_rev.good.fastq -o ./megahit_108383_out
megahit --k-min 21 --k-max 141 --k-step 12 -1 Sample_108385_fwd.good.fastq -2 Sample_108385_rev.good.fastq -o ./megahit_108385_out
megahit --k-min 21 --k-max 141 --k-step 12 -1 Sample_108386_fwd.good.fastq -2 Sample_108386_rev.good.fastq -o ./megahit_108386_out
megahit --k-min 21 --k-max 141 --k-step 12 -1 Sample_108388_fwd.good.fastq -2 Sample_108388_rev.good.fastq -o ./megahit_108388_out
megahit --k-min 21 --k-max 141 --k-step 12 -1 Sample_108390_fwd.good.fastq -2 Sample_108390_rev.good.fastq -o ./megahit_108390_out
megahit --k-min 21 --k-max 141 --k-step 12 -1 Sample_108391_fwd.good.fastq -2 Sample_108391_rev.good.fastq -o ./megahit_108391_out
megahit --k-min 21 --k-max 141 --k-step 12 -1 Sample_108392_fwd.good.fastq -2 Sample_108392_rev.good.fastq -o ./megahit_108392_out
megahit --k-min 21 --k-max 141 --k-step 12 -1 Sample_108394_fwd.good.fastq -2 Sample_108394_rev.good.fastq -o ./megahit_108394_out
megahit --k-min 21 --k-max 141 --k-step 12 -1 Sample_108396_fwd.good.fastq -2 Sample_108396_rev.good.fastq -o ./megahit_108396_out
````

#### 7.5.1. FeGenie Summary

FeGenie is a python script with dependencies that can be downloaded from  [Arkadiy-Garber GitHub](https://github.com/Arkadiy-Garber/FeGenie).

Start by loading prerequisite modules
````markdown
`r ''`{Terminal}
module load python3.6-anaconda Bioinformatics hmmer prodigal ncbi-blast R
````

Run the FeGenie python script
````markdown
`r ''`{Terminal}
FeGenie_rna.sh

# This is what's inside the "FeGenie_rna.sh" script
FeGenie.py -bin_dir /Path/To/assembly_fasta -bin_ext fa -out FeGenie_rna_out
````

Results from FeGenie analysis using Garber et al. 2020 R-scripts.

```{r echo=FALSE, warning=FALSE}
FeGenie_heatmap_data_organized<-read.csv("Plot.Data/FeGenie.dotplot.csv", header=FALSE)
```

```{r warning=FALSE}
l = length(FeGenie_heatmap_data_organized)
fegenie.t <- t(FeGenie_heatmap_data_organized[,3:l-1])
fegenie.matrix = as.matrix(fegenie.t)

colnames(fegenie.t) <- as.vector(FeGenie_heatmap_data_organized$V1)
FeGenie <- fegenie.t

#-------------data preparation
# convert to data frame
FeGenie.data <- as.data.frame(FeGenie)
#FeGenie.data<-FeGenie.data[1,]
#class(FeGenie.data)
#(FeGenie.data)

# ----------- melt data
FeGenie.data.melt <- melt(FeGenie.data, id.vars = 1:1)

# rename columns
colnames(FeGenie.data.melt)[colnames(FeGenie.data.melt)=="variable"] <- "Iron_category"
colnames(FeGenie.data.melt)[colnames(FeGenie.data.melt)=="value"] <- "Normalized_gene_abundance"

# ----------------- output files

FeGenie.data.melt$Normalized_gene_abundance = as.character(FeGenie.data.melt$Normalized_gene_abundance)
FeGenie.data.melt$Normalized_gene_abundance = as.numeric(FeGenie.data.melt$Normalized_gene_abundance)

FeGenie.meta.plot <- ggplot(FeGenie.data.melt, aes(x = X, y = Iron_category, size = Normalized_gene_abundance), alpha=0.7) +
  geom_point(aes(color=Iron_category)) +
  scale_size_area(max_size = 10) +
  labs(x="", y="Iron Category") +
  scale_y_discrete(labels=c("iron_aquisition-iron_transport" = "Iron Transport",
                            "iron_aquisition-heme_transport" = "Heme Transport",
                            "iron_aquisition-heme_oxygenase" = "Heme Oxygenase",
                            "iron_aquisition-siderophore_synthesis" = "Siderophore Synthesis",
                            "iron_aquisition-siderophore_transport" = "Siderophore Transport",
                            "iron_gene_regulation" = "Iron Gene Regulation",
                            "iron_oxidation" = "Iron Oxidation",
                            "possible_iron_oxidation_and_possible_iron_reduction" = "Probable Iron Oxidation",
                            "probable_iron_reduction" = "Probable Iron Reduction",
                            "iron_reduction" = "Iron Reduction",
                            "iron_storage" = "Iron Storage",
                            "magnetosome_formation" = "Magnetosome Formation",
                            "iron_aquisition-iron_uptake" = "Iron Uptake", 
                            "iron_aquisition-heme_uptake" = "Heme Uptake", 
                            "iron_aquisition-heme_lyase" = "Heme Lyase", 
                            "iron_aquisition-siderophore_uptake" = "Siderophore Uptake")) +
  theme(panel.background = element_rect(fill = "white", colour = "black", size = 1, linetype = "solid"),
        panel.border = element_rect(colour="black", size=1, fill=NA),
        strip.background=element_rect(fill='white', colour='white'),
        strip.text = element_text(face="bold", size=10),
        panel.grid.major = element_line(size = 0.1, colour = "gray"),
        panel.grid.minor = element_line(size = 0.1, colour = "gray"),
        axis.text = element_text(size=12, colour="black"),
        axis.title.y = element_blank(),
        axis.text.x = element_text(vjust = 1, angle = 270, color = "black", size = 12, hjust=1), legend.position="right") + guides(color = FALSE) + labs(size="Normalized\nGene Abundance")
```

```{r}
FeGenie.meta.plot
```

#### 7.5.2. Tuss FeGenie

Plot the relative iron gene abundance data from the Metagenomes (MG) and Metatranscriptomes (MT) within Tuss.
```{r, echo=FALSE}
tuss_dna_fegenie_bardata<-read.csv("Plot.Data/tuss.dna.fegenie.barplot.csv")
```

```{r warning=FALSE}
# Place categories in the preferred order for plotting
tuss_dna_fegenie_bardata$Iron <- factor(tuss_dna_fegenie_bardata$Iron,levels = c("Iron Oxidation", "Iron Reduction", "Siderophore Transport", "Iron Storage"))

tuss_dna_fegenie_bardata$Sample <- factor(tuss_dna_fegenie_bardata$Sample,levels=c("Tuss-MG","Tuss-MT-T0","Tuss-MT-T4","Tuss-MT-T24"))

tuss_dna_fegenie_barplot<-ggplot(tuss_dna_fegenie_bardata, aes(x = Iron, y = Mean, fill = Sample)) + geom_bar(stat = "identity", position = "dodge", color="black") + geom_errorbar(aes(ymin = Mean - SD, ymax = Mean + SD), width = 0.2, position = position_dodge(0.9)) + ylab(expression(atop("Microbial Iron Genes", paste("Relative Abundance (%)")))) + theme_classic() + theme(axis.text=element_text(size=10), axis.title=element_text(size=12), axis.title.x=element_blank()) + theme(legend.position = "bottom", legend.title=element_blank(), legend.text=element_text(size=8), panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.border = element_blank()) + scale_size(guide=FALSE) + scale_fill_manual(values=c("black", "palegreen", "green3", "darkgreen")) + scale_x_discrete(labels = function(Iron) str_wrap(Iron, width = 10)) + scale_y_continuous(expand = c(0, 0), limits = c(0, 100)) + annotate(geom="text", x=0.9, y=12, label="a") + annotate(geom="text", x=1.12, y=22, label="b") + annotate(geom="text", x=1.35, y=18, label="ab") + annotate(geom="text", x=2.12, y=12, label="italic(N.S.)", parse=TRUE) + annotate(geom="text", x=3.12, y=70, label="italic(N.S.)", parse=TRUE) + annotate(geom="text", x=4.12, y=50, label="italic(N.S.)", parse=TRUE) + geom_segment(aes(x = 1.12, y = 35, xend = 1.12, yend = 50)) + geom_segment(aes(x = 1.12, y = 50, xend = 2.12, yend = 50)) + geom_segment(aes(x = 2.12, y = 50, xend = 2.12, yend = 20)) + annotate("text", x = 1.635, y = 54, size=3, label = "Paired~t-test~(italic(p) == 0.004)", parse = TRUE) + annotate("text", x = 1.635, y = 61, size=4, label = "Mean MT Difference (8%)")
```

```{r warning=FALSE, message=FALSE}
tuss_dna_fegenie_barplot
```

```{r echo=FALSE}
tuss_dna_fegenie_redox_bardata<-read.csv("Plot.Data/tuss.dna.fegenie.redox.barplot.csv")
```

```{r}
# Place categories in the preferred order for plotting
tuss_dna_fegenie_redox_bardata$Iron <- factor(tuss_dna_fegenie_redox_bardata$Iron,levels = c("Iron Oxidation", "Iron Reduction"))

tuss_dna_fegenie_redox_bardata$Sample <- factor(tuss_dna_fegenie_redox_bardata$Sample,levels=c("Tuss-MG","Tuss-MT-T0","Tuss-MT-T4","Tuss-MT-T24"))

tuss_dna_fegenie_redox_barplot<-ggplot(tuss_dna_fegenie_redox_bardata, aes(x = Iron, y = Mean, fill = Sample)) + geom_bar(stat = "identity", position = "dodge", color="black") + geom_errorbar(aes(ymin = Mean - SD, ymax = Mean + SD), width = 0.2, position = position_dodge(0.9)) + ylab(expression(atop("Iron Redox Cycling Genes", paste("Relative Abundance (%)")))) + theme_classic() + theme(axis.text=element_text(size=10), axis.title=element_text(size=12), axis.title.x=element_blank()) + theme(legend.position = "bottom", legend.title=element_blank(), legend.text=element_text(size=8), panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.border = element_blank()) + scale_size(guide=FALSE) + scale_fill_manual(values=c("gray83", "palegreen", "green3", "darkgreen")) + scale_x_discrete(labels = function(Iron) str_wrap(Iron, width = 10)) + scale_y_continuous(expand = c(0, 0), limits = c(0, 40)) + annotate(geom="text", x=0.89, y=8, label="a") + annotate(geom="text", x=1.12, y=18, label="b") + annotate(geom="text", x=1.35, y=13, label="ab") + annotate(geom="text", x=2.12, y=4, label="italic(N.S.)", parse=TRUE) + geom_segment(aes(x = 1.12, y = 22, xend = 1.12, yend = 30)) + geom_segment(aes(x = 1.12, y = 30, xend = 2.12, yend = 30)) + geom_segment(aes(x = 2.12, y = 30, xend = 2.12, yend = 7)) + annotate("text", x = 1.635, y = 32, size=3, label = "Paired~t-test~(italic(p) == 0.004)", parse = TRUE) + annotate("text", x = 1.635, y = 34, size=4, label = "Mean MT Difference (43x)")
```

```{r}
tuss_dna_fegenie_redox_barplot
```

Run statistical tests on the METATRANSCRIPTOMES to determine if significant differences exist between sampling timepoints for each iron gene category .

```{r echo=FALSE}
tuss_fegenie_stats<-read.csv("Stats.Data/tuss.fegenie.stats.csv")
```

```{r}
# Subset response variables for MANOVA
tuss_fegenie_stats$response <- as.matrix(tuss_fegenie_stats[, 2:5])
```

```{r}
# MANOVA test
tuss_fegenie_manova <- manova(response ~ Timepoint, data=tuss_fegenie_stats)
summary.aov(tuss_fegenie_manova)

## Run ANOVA for each category of interest (significant in MANOVA)

# Iron Oxidation
tuss_fegenie_stat1<-aov(Iron.Oxidation~Timepoint,data=tuss_fegenie_stats)
summary.aov(tuss_fegenie_stat1)
TukeyHSD(tuss_fegenie_stat1)

# t-test for differences between reduction and oxidation averaged across experiment
t.test(tuss_fegenie_stats$Iron.Oxidation,tuss_fegenie_stats$Iron.Reduction,paired=TRUE)
```

#### 7.5.3. WS FeGenie

Plot the relative iron gene abundance data from the Metagenomes (MG) and Metatranscriptomes (MT) within WS. 
```{r, echo=FALSE}
ws_dna_fegenie_bardata<-read.csv("Plot.Data/ws.dna.fegenie.barplot.csv")
```

```{r warning=FALSE, message=FALSE}
# Place categories in the preferred order for plotting
ws_dna_fegenie_bardata$Iron <- factor(ws_dna_fegenie_bardata$Iron,levels = c("Iron Oxidation", "Iron Reduction", "Siderophore Transport", "Iron Storage"))

ws_dna_fegenie_bardata$Sample <- factor(ws_dna_fegenie_bardata$Sample,levels=c("WS-MG","WS-MT-T0","WS-MT-T4","WS-MT-T24"))

ws_dna_fegenie_barplot<-ggplot(ws_dna_fegenie_bardata, aes(x = Iron, y = Mean, fill = Sample)) + geom_bar(stat = "identity", position = "dodge") + geom_errorbar(aes(ymin = Mean - SD, ymax = Mean + SD), width = 0.2, position = position_dodge(0.9)) + ylab(expression(atop("Microbial Iron Genes", paste("Relative Abundance (%)")))) + theme_classic() + theme(axis.text=element_text(size=10), axis.title=element_text(size=12), axis.title.x=element_blank()) + theme(legend.position = "bottom", legend.title=element_blank(), legend.text=element_text(size=8), panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.border = element_blank()) + scale_size(guide=FALSE) + scale_fill_manual(values=c("black","lightskyblue1", "dodgerblue1", "midnightblue")) + scale_x_discrete(labels = function(Iron) str_wrap(Iron, width = 10)) + scale_y_continuous(expand = c(0, 0), limits = c(0, 100)) + annotate(geom="text", x=1.12, y=20, label="italic(N.S.)", parse=TRUE) + annotate(geom="text", x=2.12, y=30, label="italic(N.S.)", parse=TRUE) + annotate(geom="text", x=3.12, y=60, label="italic(N.S.)", parse=TRUE) + annotate(geom="text", x=4.12, y=60, label="italic(N.S.)", parse=TRUE) + geom_segment(aes(x = 1.12, y = 30, xend = 1.12, yend = 50)) + geom_segment(aes(x = 1.12, y = 50, xend = 2.12, yend = 50)) + geom_segment(aes(x = 2.12, y = 50, xend = 2.12, yend = 45)) + annotate("text", x = 1.635, y = 54, size=3, label = "Paired~t-test~(italic(p) == 0.005)", parse = TRUE) + annotate("text", x = 1.635, y = 61, size=4, label = "Mean MT Difference (14%)")
```

```{r echo=FALSE}
ws_dna_fegenie_barplot
```

```{r echo=FALSE}
ws_dna_fegenie_redox_bardata<-read.csv("Plot.Data/ws.dna.fegenie.redox.barplot.csv")
```

```{r}
# Place categories in the preferred order for plotting
ws_dna_fegenie_redox_bardata$Iron <- factor(ws_dna_fegenie_redox_bardata$Iron,levels = c("Iron Oxidation", "Iron Reduction"))

ws_dna_fegenie_redox_bardata$Sample <- factor(ws_dna_fegenie_redox_bardata$Sample,levels=c("WS-MG","WS-MT-T0","WS-MT-T4","WS-MT-T24"))

ws_dna_fegenie_redox_barplot<-ggplot(ws_dna_fegenie_redox_bardata, aes(x = Iron, y = Mean, fill = Sample)) + geom_bar(stat = "identity", position = "dodge", color = "black") + geom_errorbar(aes(ymin = Mean - SD, ymax = Mean + SD), width = 0.2, position = position_dodge(0.9)) + ylab(expression(atop("Iron Redox Cycling Genes", paste("Relative Abundance (%)")))) + theme_classic() + theme(axis.text=element_text(size=10), axis.title=element_text(size=12), axis.title.x=element_blank()) + theme(legend.position = "bottom", legend.title=element_blank(), legend.text=element_text(size=8), panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.border = element_blank()) + scale_size(guide=FALSE) + scale_fill_manual(values=c("black","lightskyblue1", "dodgerblue1", "midnightblue")) + scale_x_discrete(labels = function(Iron) str_wrap(Iron, width = 10)) + scale_y_continuous(expand = c(0, 0), limits = c(0, 40)) + annotate(geom="text", x=1.12, y=8.5, label="italic(N.S.)", parse=TRUE) + annotate(geom="text", x=2.12, y=25, label="italic(N.S.)", parse=TRUE) + geom_segment(aes(x = 1.12, y = 10, xend = 1.12, yend = 30)) + geom_segment(aes(x = 1.12, y = 30, xend = 2.12, yend = 30)) + geom_segment(aes(x = 2.12, y = 30, xend = 2.12, yend = 27)) + annotate("text", x = 1.635, y = 32, size=3, label = "Paired~t-test~(italic(p) == 0.005)", parse = TRUE) + annotate("text", x = 1.635, y = 34, size=4, label = "Mean MT Difference (4x)")
```

```{r}
ws_dna_fegenie_redox_barplot
```

```{r, echo=FALSE, eval=FALSE}
# Save the TUSS and WS FeGenie barplots together
setEPS()
postscript("Fig.4.MT.FeGenie.eps", width=10.0, height = 5.0)
ggarrange(tuss_dna_fegenie_redox_barplot, ws_dna_fegenie_redox_barplot, heights=c(5,5),
          labels = c("A","B"),
          ncol = 2, nrow = 1, align="v")
dev.off()
```

Run statistical tests on the METATRANSCRIPTOMES to determine if significant differences exist between sampling timepoints for each iron gene category.

```{r echo=FALSE}
ws_fegenie_stats<-read.csv("Stats.Data/ws.fegenie.stats.csv")
```

```{r}
# Subset response variables for MANOVA
ws_fegenie_stats$response <- as.matrix(ws_fegenie_stats[, 2:5])

# MANOVA test
ws_fegenie_manova <- manova(response ~ Timepoint, data=ws_fegenie_stats)
summary.aov(ws_fegenie_manova)

# t-test for differences between reduction and oxidation averaged across experiment
t.test(ws_fegenie_stats$Iron.Oxidation,ws_fegenie_stats$Iron.Reduction,paired=TRUE)
```

#### 7.5.4. Ferroptosis

Ferroptosis is a form of regulated cell death that results from the iron-dependent accumulation of lipid peroxides associated with reactive oxygen species such as hydroxyl radical.  Here, it is likely that gene expression for ferroptosis is directly linked with the overproduction of hydroxyl radical given that hydroxyl radical production associated with the geochemical oxidation of Fe(II) has been shown to increase in tussock tundra following rainfall.  However, little is known about the regulation of ferroptosis in soil environments or even how it is regulated within microbial cells regardless of environment.  More studies need to be done to determine any direct link between hydroxyl radical production and ferroptosis gene expression following rainfall.
```{r}
tuss_ferroptosis<-tpm_all_exp_ann

tuss_ferroptosis<-subset(tuss_ferroptosis, Tier_IV=="04216 Ferroptosis [PATH:ko04216]",
select=ID:Taxonomy)
tuss_ferroptosis
```

# Reproducibility

The session information is provided for full reproducibility.
```{r}
devtools::session_info()
```