---
title: "Microbial Response to Rainfall -- Gas Chromatography Analysis"
author: 'Authors: [Karl J. Romanowicz](https://lsa.umich.edu/eeb/people/graduate-students/kjromano.html),
  Byron C. Crump, George W. Kling'
output:
  html_notebook:
    theme: spacelab
    toc: yes
    toc_depth: 5
    toc_float:
      collapsed: yes
      smooth_scroll: yes
  html_document:
    toc: yes
    toc_depth: '5'
    df_print: paged
  pdf_document:
    toc: yes
    toc_depth: '5'
---

__________________________________________________

**R Notebook:** Provides reproducible analysis for gas chromatography (GC) data in the following manuscript:

**Citation:** Romanowicz, KJ, Crump, BC, Kling, GW. (2021) Rainfall alters permafrost soil redox conditions, but meta-omics show divergent microbial community responses by tundra type in the arctic. Soil Systems 5(1): 17. [https://doi.org/10.3390/soilsystems5010017](https://doi.org/10.3390/soilsystems5010017)

**GitHub Repository:** [https://github.com/kromanowicz/2021-Romanowicz-SoilSystems](https://github.com/kromanowicz/2021-Romanowicz-SoilSystems)

**NCBI BioProject:** [https://www.ncbi.nlm.nih.gov/bioproject/PRJNA666429](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA666429)

**Accepted for Publication:** *Soil Systems* 10 March 2021

# Experiment

This R Notebook provides complete reproducibility of the data analysis in ***"Rainfall alters permafrost soil redox conditions, but meta-omics show divergent microbial community responses by tundra type in the Arctic "*** by Romanowicz, Crump, and Kling. In this experiment, mesocosms containing soil from the active layer of two dominant tundra types were subjected to simulated rainfall to alter redox conditions.  The microbial functional potential (metagenomics) and gene expression (metatranscriptomics) patterns were measured during saturated anoxic redox conditions prior to rainfall and at multiple time points following the simulated rainfall event. Other measurements include soil properties as well as microbial respiration (CO~2~) and methane (CH~4~) production from soil subsamples collected at each sampling time point. The purpose was to determine if rainfall, as a form of soil oxidation, is sufficient to alter the anoxic redox conditions in arctic tundra and enhance the microbial degradation of organic carbon and CH~4~ to CO~2~.

<!-- Display Figure 1 -->
![](R.Images/Concept1.png)
**Conceptual Figure.**  A total of 12 tundra mesocosms (3 replicates x 2 tundra types x 2 sets of response cores) were acclimated initially under anoxic redox conditions to mimic field conditions (T0).  Dissolved oxygen was supplied to soils through the downward flow of oxygenated water during a simulated rainfall event.  Dissolved oxygen will likely change the redox gradient directly following rainfall (T4) as a short-term effect.  Anoxic conditions will likely be re-established after 24 hours (T24) as the pulse of oxygen is consumed through abiotic and biotic soil processes.  Under anoxic redox conditions (T0), microorganisms likely degrade organic carbon through anaerobic and fermentation pathways, producing CH~4~ and reducing Fe(III) to Fe(II).  Rainfall-induced soil oxidation (T4) should stimulate heterotrophic microorganisms that degrade organic carbon and CH~4~ through aerobic metabolic pathways, releasing CO~2~. Soil oxidation should also stimulate aerobic autotrophic iron oxidizing bacteria that oxidize Fe(II) to Fe(III) and convert CO~2~ into microbial biomass.  The long-term response (T24) will likely be a combination of aerobic and anaerobic metabolism as well as a combination of reduction and oxidation iron reactions as dissolved oxygen is consumed.  The predicted redox conditions and predicted redox reactions for coupled Fe(II)/Fe(III) cycling, as well as the microbial-induced release of CO~2~ or CH~4~ at each time point are based on the predicted availability of dissolved oxygen entering tundra soils through simulated rainfall.

<font color="blue">**Soil Sampling for Microbial Gene Expression**</font>

An initial soil sampling event for microbial activity was conducted at the end of the anoxic acclimation period (4-7 days) in all mesocosm replicates, representing sampling time point **T0**.  Mesocosms were then flushed to simulate a rainfall event.  Additional soil sampling events were conducted at **T4** (4-hrs) and **T24** (24-hrs) following the rainfall event to determine the temporal extent of microbial gene expression.  Soil cores (2.54 cm diameter, 30 cm length) were extracted in duplicate from each mesocosm replicate at each sampling time point and homogenized by depth in 10-cm increments.  The 10-20 cm soil increment, composed of organic soil in all mesocosm replicates, was chosen for microbial gene expression analysis and preserved in RNAlater Stabilization Reagent in sterile tubes at 4&deg;C for 18 hours and then stored at -80&deg;C until extraction.

![](R.Images/Buckets.png)

**Field Experiment.** Tundra soil cores were collected from field sites in August 2017 (top left) and placed in buckets to establish the mesocosm experiment (bottom left).  Tussock tundra cores were composed of an organic soil layer overlying a mineral soil layer (top middle) while wet sedge tundra cores were composed entirely of organic soil (bottom middle).  Soil subsampling for microbial activity was taken from the 10-20 cm depth of duplicate soil cores in Tussock (top right) and Wet Sedge (bottom right).

```{r echo=FALSE}
# Set global options for notebook
knitr::opts_knit$set(root.dir = normalizePath("~/Desktop/RainSim2017"))
knitr::opts_chunk$set(fig.width=10, fig.height=8, warning=FALSE, message=FALSE)
```

```{r message=FALSE, results='hide', warning=FALSE}
# Make a vector of required packages
required.packages <- c("ape","cowplot","data.table","devtools","dplyr","DT","ggplot2","ggpubr","grid","gridExtra","kableExtra","knitr","pheatmap","png","RColorBrewer","reshape","rstatix","statmod","stringr","tibble","tidyr","tidyverse","vegan","yaml")

# Load required packages
lapply(required.packages, library, character.only = TRUE)
```

# Soil Incubation

A subset of each soil sample collected from the 10-20 cm depth during the mesocosm experiment were sealed in jars for an incubation experiment to measure microbial respiration and methane production.  Jars containing soil from the wet sedge tundra ecosystem were purged with N~2~ to reproduce anaerobic field conditions.  Each jar was sealed with an airtight lid that contained a septum for gas sampling.  Jars were incubated for 5 days in the dark at field temperature (4$^\circ$C).  Following the 5-day incubation, we removed 20 mL of headspace gas in an evacuated syringe and immediately analyzed CO~2~ and CH~4~ concentrations via gas chromotography.

![](R.Images/Gases.png)

## Microbial Respiration

```{r echo=FALSE}
# Load the CO2 production data
co2_gas<-read.csv("Plot.Data/CO2.gas.barplot.csv")
```

Plot the CO~2~ data for microbial respiration.

```{r}
# Place categories in preferred order for plotting
co2_gas$Veg<-factor(co2_gas$Veg, levels=c("Tussock","Wet Sedge"))

co2_gas$Sample <- factor(co2_gas$Sample,levels=c("Tuss-T0","WS-T0","Tuss-T4","WS-T4","Tuss-T24","WS-T24"))

co2_gas_barplot<-ggplot(co2_gas, aes(x = Veg, y = Mean, fill = Sample)) + geom_bar(stat = "identity", position = "dodge", color="black") + geom_errorbar(aes(ymin = Mean - SD, ymax = Mean + SD), width = 0.2, position = position_dodge(0.9)) + ylab(expression(atop("Microbial Respiration", paste(CO[2]~(ng~C~g^{"-1"}~soil~hr^{"-1"}))))) + theme_classic() + theme(axis.text=element_text(size=10), axis.title=element_text(size=12), axis.title.x=element_blank()) + theme(legend.position = "bottom", legend.title=element_blank(), legend.text=element_text(size=8), panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.border = element_blank()) + scale_size(guide=FALSE) + scale_fill_manual(values=c("palegreen","lightskyblue1", "green3", "dodgerblue1", "darkgreen", "midnightblue")) + scale_y_continuous(expand = c(0, 0), limits = c(0, 120)) + annotate(geom="text", x=1.0, y=85, label="italic(N.S.)", parse=TRUE) + annotate(geom="text", x=2.0, y=23, label="italic(N.S.)", parse=TRUE) + geom_segment(aes(x = 1.0, y = 90, xend = 1.0, yend = 100)) + geom_segment(aes(x = 1.0, y = 100, xend = 2.0, yend = 100)) + geom_segment(aes(x = 2.0, y = 100, xend = 2.0, yend = 32)) + annotate("text", x = 1.5, y = 110, label = "Mean Difference (4.5x)") + annotate("text", x = 1.5, y = 103.5, size=3, label = "Paired~t-test~(italic(p) < 0.001)", parse = TRUE)
```

```{r}
co2_gas_barplot
```

Run statistics for differences between sampling time points within each tundra ecosystem.

```{r}
# Load CO2 data
co2_anova<-read.csv("Stats.Data/co2.anova.csv")
```

```{r}
# Run ANOVA for Tuss differences
tuss_co2_anova<-aov(Tuss_CO2~Time, data=co2_anova)
summary.aov(tuss_co2_anova)
TukeyHSD(tuss_co2_anova)

# Run ANOVA for WS differences
ws_co2_anova<-aov(WS_CO2~Time, data=co2_anova)
summary.aov(ws_co2_anova)
TukeyHSD(ws_co2_anova)
```

Run t-test for differences in CO2 production between tundra ecosystems
```{r echo=FALSE, warning=FALSE, message=FALSE}
co2.t.test<-read.csv("Stats.Data/co2.t.test.csv")
```

```{r warning=FALSE, message=FALSE}
t.test(co2.t.test$Tuss,co2.t.test$WS,paired=TRUE)
```

## Methane Production

```{r echo=FALSE}
# Load the CH4 production data
ch4_gas<-read.csv("Plot.Data/CH4.gas.barplot.csv")
```

Plot the CH~4~ data for methane production.

```{r}
# Place categories in preferred order for plotting
ch4_gas$Veg<-factor(ch4_gas$Veg, levels=c("Tussock","Wet Sedge"))

ch4_gas$Sample <- factor(ch4_gas$Sample,levels=c("Tuss-T0","WS-T0","Tuss-T4","WS-T4","Tuss-T24","WS-T24"))

ch4_gas_barplot<-ggplot(ch4_gas, aes(x = Veg, y = Mean, fill = Sample)) + geom_bar(stat = "identity", position = "dodge", color="black") + geom_errorbar(aes(ymin = Mean - SD, ymax = Mean + SD), width = 0.2, position = position_dodge(0.9)) + ylab(expression(atop("Methane Production", paste(CH[4]~(pg~C~g^{"-1"}~soil~hr^{"-1"}))))) + theme_classic() + theme(axis.text=element_text(size=10), axis.title=element_text(size=12), axis.title.x=element_blank()) + theme(legend.position = "bottom", legend.title=element_blank(), legend.text=element_text(size=8), panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.border = element_blank()) + scale_size(guide=FALSE) + scale_fill_manual(values=c("palegreen", "lightskyblue1", "green3", "dodgerblue1", "darkgreen", "midnightblue")) + scale_y_continuous(expand = c(0, 0), limits = c(-10, 40)) + geom_hline(aes(yintercept = 0), color="black") + annotate(geom="text", x=0.7, y=25, label="a", parse=TRUE) + annotate(geom="text", x=1.0, y=-6, label="b", parse=TRUE) + annotate(geom="text", x=1.3, y=-6, label="b", parse=TRUE) + annotate(geom="text", x=1.7, y=13, label="a", parse=TRUE) + annotate(geom="text", x=2.0, y=-8, label="b", parse=TRUE) + annotate(geom="text", x=2.3, y=-8, label="b", parse=TRUE)
```

```{r}
ch4_gas_barplot
```

```{r echo=FALSE, eval=FALSE}
# Save the CO2 and CH4 plots together
setEPS()
postscript("Fig.5.Gases.CO2.CH4.eps", width=10.0, height = 5.0)
ggarrange(co2_gas_barplot, ch4_gas_barplot, heights=c(5,5),
          labels = c("A","B"),
          ncol = 2, nrow = 1, align="v")
dev.off()
```

Run statistics for differences between sampling time points within each tundra ecosystem.

```{r echo=FALSE}
# Load CH4 data
ch4_anova<-read.csv("Stats.Data/ch4.anova.csv")
```

```{r}
# Run ANOVA for Tuss differences
tuss_ch4_anova<-aov(Tuss_CH4~Time, data=ch4_anova)
tuss_ch4_anova
TukeyHSD(tuss_ch4_anova)

# Run ANOVA for WS differences
ws_ch4_anova<-aov(WS_CH4~Time, data=ch4_anova)
ws_ch4_anova
TukeyHSD(ws_ch4_anova)
```

# Reproducibility

The session information is provided for full reproducibility.
```{r}
devtools::session_info()
```